---
title: "DHE_dissertation_16s"
author: "Benjamin Young"
date: "13/08/2021"
output: html_document
---

```{r package installation, include = F}
#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("microbiomeMarker")
# devtools::install_github('ggloor/CoDaSeq/CoDaSeq')
# remotes::install_github("adw96/breakaway")
#remotes::install_github("adw96/DivNet")
```

```{r package loading, include = F}
library(zCompositions)
library(phyloseq)
library(vegan)
library(DESeq2)
library(tidyverse)
library(dendextend)
library(viridis)
library(reshape)
#library(seqRFLP)
library(phangorn)
library(ape)
library(dada2)
library(DECIPHER)
library(ggtree)
library(compare)
library(ALDEx2)
library(CoDaSeq)
library(ggdendro)
library(factoextra)
library(PCAtools)
library(magrittr)
library(breakaway)
library(DivNet) # last checked with version 0.3.5
library(ANCOMBC)
library(ComplexHeatmap)
library(circlize)
library(venn)
library(DEGreport)
library(biomformat)
library(scales)
library(randomcoloR)
library(microbiomeMarker)
```


# Prepping DADA2 Outputs for 16s Analsyis

This is prepping the outputs of DADA2 for 16s analysis (alpha, beta, differntial abundance etc). Overall pipeline of this section is as follows
- *Combining of lanes* - I ran separate DADA2 pipelines for the 2 MiSeq runs due to error models being generated for sequencing lanes. I also ran taxonomy classification separately (although for future projects i will combine and run taxonomy classification so to avoid some of the steps I had to do below). 

- *Fixing ASVs from lanes with different taxonomic classifications* - As I joined after assigning taxonomy, there are some ASVs which have the same ASV sequence, but different taxonomy. Please note that these taxonomies are identical just they assign to different taxonomic levels (i.e. ASVx is same as ASVy in sequence, ASVx annotates to genus, while ASVy annotates to family). I therfore used the highest taxonomic classification ASV and added the counts to this one. I then removed the ASV which had higher lower taxonomy. 

- *Removing Low Count Samples* - From literature I am seeing that samples with < 1000 counts are dropped. This has therfore been incorporated here.  

- *Phylogenetic Tree Construction* - I generated the phylogenetic tree here so it included all ASVs identified from DADA2 (using phangorn). This is very computationally expensive and took ~2 days using the desktop MAC. I did this here so that i could identify unannotated ASVs from SILVA which groupeded with Mitochondira and Chloroplast, as well as allowing identification of intersting ASVs that group with certain phylogentic ranks. 

- *Removal of Mitochondria and Chloroplast ASVs* - As with all coral 16s research, removal of the ASVs annotated to mitochondria and chloroplast are removed. Thus I do that here with ones which annotated with SILVA to these, as well as using the phylogenetic tree to remove NA ASVs which cluster within the branches of the annotated Mitochondria and Chloroplast. 

- *NAs for Blasting* - So the taxonomy with SILVA left a fair amount of the ASVs with NAs, therfore I am blasting them against the nt NCBI database to try and get better taxonomies. Here I used NCBI local on the desktop, and then imported into R and used taxonomizr to get full taxonomies. 

- *Checking Samples* - After the ASV and sample filtering I just wanted to get an idea of samples on sampling dates. Nothing fancy here just an FYI.


## Reading in DADA2 outputs and Generating All Info Files and Combined File
### Lane 921

```{r Reading in counts for lane 921, include=F}
# read.table(
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/dada2_processing/lane1_921/dada2_output/ASVs_counts_921.tsv",
#   header = T,
#   row.names = 1,
#   check.names = F
# ) %>%
#   as.data.frame() %>%
#   rownames_to_column(var = "ASV_number") -> count_921
# # View(count_921)
# # str(count_921)
```

```{r reading in taxonomy lane 921, include=F}
# read.table(
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/dada2_processing/lane1_921/dada2_output/ASVs_taxonomy_921.tsv",
#   header = T,
#   row.names = 1,
#   check.names = F,
#   sep = "\t"
# ) %>%
#   rownames_to_column(var = "ASV_number") -> tax_921
# #View(tax_921)
# #str(tax_921)
```

```{r reading in fasta lane 921, include=F}
# fa_921 <- read.table(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/dada2_processing/lane1_921/dada2_output/ASV_921.fa")
# #View(fa_921)
```

```{r Splitting lane 921 fasta into long dataframe, include=F}
# fa_921 %>% 
#   group_by(grp = str_c('Column', rep(1:2, length.out = n()))) %>% 
#   mutate(rn = row_number()) %>% 
#   ungroup() %>% 
#   pivot_wider(names_from = grp, values_from = V1) %>% 
#   mutate(Column1 = str_remove_all(Column1, ">"),
#          ASV_number = Column1, 
#          Sequence = Column2) %>% 
#   dplyr::select(-rn, -Column1, -Column2) -> fa_921_long
# # View(fa_921_long)
```

```{r joining to form a meta lane 921 file with counts taxonomy and sequences, include = F}
# tax_921 %>% 
#   full_join(fa_921_long, 
#             by = "ASV_number", 
#             copy = T) %>% 
#   full_join(count_921, 
#             by = "ASV_number", 
#             copy = T) %>% 
#   column_to_rownames(var = "ASV_number") -> complete_921
```

```{r saving all info lane 921 as csv, include = F}
# write.csv(complete_921, 
#           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/lane921_16s_allinfo.csv")
```


### Lane 925
Need to do some additional filtering to remove Mike and Brielle samples here that were included in this lane. 
Columns 132 to 176 contain these, include dplyr::select and remove these. 

```{r reading in counts for lane 925, include=F}
# read.table(
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/dada2_processing/lane2_925/dada2_ouput/ASVs_counts_925.tsv",
#   header = T,
#   row.names = 1,
#   check.names = F
# ) %>%
#   rownames_to_column(var = "ASV_number") %>% 
#   dplyr::select(-132:-176) -> count_925
# # View(count_925)
```

```{r reading in taxonomy for lane 925, include=F}
# read.table(
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/dada2_processing/lane2_925/dada2_ouput/ASVs_taxonomy_925.tsv",
#   header = T,
#   row.names = 1,
#   check.names = F,
#   sep = "\t"
# ) %>% 
#   rownames_to_column(var = "ASV_number") -> tax_925
# # View(tax_925)
```

```{r Reading in lane 925 fasta, include=F}
# fa_925 <- read.table(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/dada2_processing/lane2_925/dada2_ouput/ASV_925.fa")
# # View(fa_925)
```

```{r Splitting lane 925 fasta into long dataframe format, include=F}
# fa_925 %>% 
#   group_by(grp = str_c('Column', rep(1:2, length.out = n()))) %>% 
#   mutate(rn = row_number()) %>% 
#   ungroup() %>% 
#   pivot_wider(names_from = grp, values_from = V1) %>% 
#   mutate(Column1 = str_remove_all(Column1, ">"),
#          ASV_number = Column1, 
#          Sequence = Column2) %>% 
#   dplyr::select(-rn, -Column1, -Column2) -> fa_925_long
```

```{r joining to form meta 925 with everything, include = F}
# tax_925 %>% 
#   full_join(fa_925_long, 
#             by = "ASV_number", 
#             copy = T) %>% 
#   full_join(count_925, 
#             by = "ASV_number", 
#             copy = T) %>% 
#   column_to_rownames(var = "ASV_number") -> complete_925
# # View(complete_925)
```

```{r saving 925 lane, include = F}
# write.csv(complete_925, 
#           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/lane925_16s_allinfo.csv")
```


### Combine Lane 921 and Lane 925

```{r Combined the two lanes by taxonomy and generating new ASV numbers, include = F}
# complete_921 %>%
#   rownames_to_column(var = "ASV_number") %>%
#   dplyr::select(-ASV_number) %>%
#   full_join(
#     complete_925 %>%
#       rownames_to_column(var = "ASV_number") %>%
#       dplyr::select(-ASV_number),
#     by = c(
#       "Sequence",
#       "domain",
#       "phylum",
#       "class",
#       "order",
#       "family",
#       "genus",
#       "species"
#     ),
#     copy = T
#   ) %>%
#   mutate(obs = "ASV",
#          observation = 1:n()) %>%
#   unite(ASV_number, obs, observation, sep = "_") %>%
#   dplyr::select(
#     ASV_number,
#     domain,
#     phylum,
#     class,
#     order,
#     family,
#     genus,
#     species,
#     Sequence,
#     everything()
#   ) -> combined_DHE
# # View(combined_DHE)
```

```{r Checking combining step worked with no problems, echo = F}
# venn(list(complete_921$Sequence, complete_925$Sequence))
# sum(2318, 2415, 5226)
# 
# nrow(combined_DHE)
# nrow(complete_921)
# nrow(complete_925)
# ncol(combined_DHE)
# ncol(complete_921)
# ncol(complete_925)
```

*ASVS (rows)*
combined_DHE -> 10151 ASVs
Lane 921 = 4733 ASVs
Lane 925 = 7641 ASVs

*Samples (columns)*
combined_DHE -> 344
Lane 921 = 198
Lane 925 = 153

Combined DHE has *344 samples*

### Fixing Duplicate ASVs which were Assigned Different Taxonomies
made a new file `DHE_duplicated_for_Addition generation.csv` to combine 
1. Ordered by sequence
2. Checked each duplicate sequence identical
3. copied most complete taxonomic classification for each duplicated sequence
4. copied counts from na row to row above (i.e. complete count for the one asv)
5. generated a column with same ASV classification for each duplicated sequence

Filtering in R 
1. filter combined_DHE by all the ASVs present in the ASV_remove column 
2. select (using filter) every odd row (this has completed counts and no NAs in it)
3. combine this with the combined DHE to make the completed combined_DHE

```{r Generating the csv with thew duplicated sequences, include = F}
## combined_DHE %>% 
##   filter(duplicated(Sequence) | duplicated(Sequence, fromLast = T)) %>% 
##   write.csv(., file = "~/Desktop/DHE_dup_seq_combined_all.csv")
```

This file I have then edited to make the file which is read in directly below. 

```{r Reading in corrected taxonomy information and fixing the combin ed metadata file, include = F}
# corrected_taxa <- read.csv(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/dup_ASVs_fixed_for_analysis/DHE_duplicated_for_addition.csv", 
#                            check.names = F)
# 
# ## making filtering object to remove all duplicated sequences
# corrected_taxa %>% 
#   column_to_rownames(var = "ASV_remove") %>% 
#   rownames() -> combDHEfilter
# 
# ## removing duplicated from combined_DHE
# ## doing this removes 384 which is the same nrow of the corrected taxa file
# nrow(combined_DHE)
# combined_DHE %>%
#   filter(!ASV_number %in% combDHEfilter) -> combined_DHE_dupfilt
# nrow(combined_DHE_dupfilt)
# nrow(corrected_taxa)
# 
# ## removing the ASV_Remove and seleccting only odd rows from corrected and then full joining with DHE_combined
# all(colnames(corrected_taxa %>% dplyr::select(-ASV_remove, -1)) == colnames(combined_DHE_dupfilt))
# 
# corrected_taxa %>% 
#   dplyr::select(-ASV_remove, -1) %>%
#   filter(row_number() %% 2 == 1) %>% 
#   rbind(combined_DHE_dupfilt) -> combined_DHE_dupfilt_DONE
# 
# nrow(combined_DHE_dupfilt_DONE)
# ncol(combined_DHE_dupfilt_DONE)
# 9767 + 384/2
# 
# combined_DHE <- combined_DHE_dupfilt_DONE
```

Original combined_DHE -> 10151 ASVs
De-duplicated combined_DHE -> 9959 ASVs

Matches up as well when we do 9767 + (284/2) = 9959 :)

Column names of corrected taxa match the combined_DHE filter which is also great news. 

```{r writing combined file for supplementary, include = F}
# write.csv(combined_DHE, 
#           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/combined_16s_DHE_alldata.csv")
```


## Removing Low Count Samples (<1000)

```{r Splitting metadata files up for analysis and filtering, include = F}
# combined_DHE %>% 
#   dplyr::select(-domain, -phylum, -class, -order, -family, -genus, -species, -Sequence) %>% 
#   column_to_rownames(var = "ASV_number") %>% 
#   replace(is.na(.), 0) %>% 
#   as.data.frame() -> DHE_counts
# # nrow(DHE_counts)
# # ncol(DHE_counts)
# 
# combined_DHE %>%
#   dplyr::select(ASV_number, domain, phylum, class, order, family, genus, species) %>% 
#   column_to_rownames(var = "ASV_number") -> DHE_taxo
# # nrow(DHE_taxo)
# # ncol(DHE_taxo)
# 
# combined_DHE %>% 
#   dplyr::select(ASV_number, Sequence) -> DHE_sequence
# 
# # View(DHE_counts)
# # View(DHE_taxo)
# # View(DHE_sequence)
```

```{r Treatment and ordering counts, include = F}
# hdl <- read.csv(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/physiological_data/metadata/health_long.csv")
# 
# read.csv(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/sequencing_documents/sequenced_samples.csv") %>% 
#   mutate(Frag_number = as.character(Frag_number)) -> samps
# 
# hdl %>%
#   mutate(SAMPLE_TUBE = na_if(SAMPLE_TUBE, c("na"))) %>%
#   filter(hdl$SAMPLE_TUBE %in% samps$sample_tubes) %>%
#   drop_na() %>%
#   mutate(sample_tubes = SAMPLE_TUBE,
#          Frag_number = as.character(Frag_number)) %>%
#   dplyr::select(-sampled,-SAMPLE_TUBE) %>%
#   full_join(samps,
#             by = c("Frag_number",
#                    "Genotype",
#                    "sample_tubes",
#                    "timepoint")) %>%
#   dplyr::select(sample_ID, Genotype, Frag_number, 
#          Recovery_tank, ERL_tank_number, 
#          D_touch_tank, Dtreatment, 
#          timepoint, health_status, 
#          Tank_treatment,
#          sample_tubes, bead_beating_tube_label) %>%
#   column_to_rownames(var = "sample_ID") -> treat_DHE
# # ncol(DHE_counts)
# # nrow(treat_DHE)
# 
# matchup <- match(rownames(treat_DHE), colnames(DHE_counts))
# DHE_counts  <- DHE_counts[,matchup ]
# all(rownames(treat_DHE) == colnames(DHE_counts))
```

Decided to remove samples with less than 2000 reads. This is because there are only four samples with low reads and then everything else is pretty good. 

```{r removing samples with <2000 reads, include = F}
# DHE_counts %>% 
#   t() %>% 
#   as.data.frame() %>% 
#   rownames_to_column(var="Bagnumber") %>% 
#   mutate(count_total = rowSums(DHE_counts %>% 
#                                  t() %>% 
#                                  as.data.frame() %>% 
#                                  dplyr::select(starts_with("ASV")))) %>%
#   dplyr::select(Bagnumber, count_total) %>% 
#   column_to_rownames(var="Bagnumber") %>% 
#   arrange(count_total) %>%
#   filter(count_total < 2000) %>%
#   rownames() -> lowcount_bagnums
# length(lowcount_bagnums)
# 
# DHE_counts %>% 
#   t() %>% 
#   as.data.frame() %>% 
#   rownames_to_column(var="Bagnumber") %>% 
#   filter(!Bagnumber %in% lowcount_bagnums) %>% 
#   column_to_rownames(var="Bagnumber") %>% 
#   t() %>% 
#   as.data.frame() -> DHE_counts
# ncol(DHE_counts)
# lowcount_bagnums
```

4 samples with less than 2000 reads
*APAL-CN295heatsmarcscensinitial - 0
*APAL-ML235heatplaceboslurry17thaug - 16
*APAL-HS1185heatplaceboslurryinitial - 1447
*APAL-CN477ambientplaceboslurry17thaug - 948

Leaves us with **331** samples (335 original)

```{r Matching up counts to treatment file and taxonomy files, echo = F}
# DHE_counts %>% 
#   t() %>%
#   rownames() -> bagnums
# 
# DHE_counts %>% 
#   rownames() -> ASVs
# 
# treat_DHE %>% 
#   rownames_to_column(var="Bag_number") %>%
#   filter(Bag_number %in% bagnums) %>% 
#   column_to_rownames(var = "Bag_number") -> treat_DHE_filt
# 
# matchup <- match(rownames(treat_DHE_filt), colnames(DHE_counts))
# DHE_counts  <- DHE_counts[,matchup]
# all(rownames(treat_DHE_filt) == colnames(DHE_counts))
# 
# DHE_taxo %>% 
#   rownames_to_column(var="ASV_number") %>% 
#   filter(ASV_number %in% ASVs) %>% 
#   column_to_rownames(var="ASV_number") -> DHE_taxo_filt
# 
# all(rownames(DHE_taxo_filt) == rownames(DHE_counts))
# nrow(DHE_taxo_filt)
# nrow(DHE_counts)
# ncol(DHE_counts)
# nrow(treat_DHE_filt)
```

```{r Checking ASVs match between everything, echo = F}
# all(DHE_sequence$ASV_number == rownames(DHE_counts))
# all(DHE_sequence$ASV_number == rownames(DHE_taxo))
# all(rownames(DHE_taxo) == rownames(DHE_counts))
```

Taxo, treatment and counts now all match each other 
ASVs = 9959
Samples = 331


## Phylogenetic Tree Construction
This is an option to include in phyloseq objects and I think important. I have done this but it takes a while to run the secondary fitGTR model for the tree. Run on a supercomputer os very powerful machine

```{r Filtered Sequence tab for phangorn, include = F}
## ASV_filt <- dataframe2fas(DHE_sequence %>%
##                             select(ASV_number, Sequence),
##                           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/phylogenetic_tree_16s/DHE_ASV_all.fasta")
## 
## DHE_seq <- getSequences(object = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/phylogenetic_tree_16s/DHE_ASV_all.fasta")
## 
## align <- AlignSeqs(DNAStringSet(DHE_seq), anchor = NA)
## phang.align <- phyDat(as(align, "matrix"), type="DNA")
## dm <- dist.ml(phang.align)
## treeNJ <- NJ(dm)
## 
## DHE_taxo %>%
##   rownames_to_column(var = "tip.label") %>%
##   write.table(., file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/phylogenetic_tree_16s/DHE_taxo",
##             quote = F, row.names = F, sep = "\t")
```

```{r Nearest Joining and fitGTR phylogenetic trees, include = F}
## Needs to be run on Pegasus otherwise can take a week
## fit = pml(treeNJ, data = phang.align)
## fitGTR <- update(fit, k = 4, inv = 0.2)
## fitGTR <-
##   optim.pml(
##     fitGTR,
##     model = "GTR",
##     optInv = TRUE,
##     optGamma = TRUE,
##     rearrangement = "stochastic",
##     control = pml.control(trace = 0)
##   )
```

```{r Writing tree and tips for figtree, include = F}
## write.tree(treeNJ, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/phylogenetic_tree_16s/DHE_NJ_tree")
## write.tree(fitGTR$tree, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/phylogenetic_tree_16s/DHE_fitgtr_tree")
## 
## DHE_taxo %>%
##   rownames_to_column(var = "tip.label") %>%
##   write.table(., file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/phylogenetic_tree_16s/DHE_taxo",
##             quote = F, row.names = F, sep = "\t")
```

```{r Saving so do not have to run every time want to use, include=F}
## save.image(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/phylogenetic_tree_16s/DHE_phylotree.RData")
## save(align, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/phylogenetic_tree_16s/alignedseqs.RData")
## save(dm, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/phylogenetic_tree_16s/dist.ml_phang.RData")
## save(fitGTR, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/phylogenetic_tree_16s/fitGTR.RData")
## save(phang.align, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/phylogenetic_tree_16s/phangalign.RData")
```

```{r Quick Plot to look at trees, include = F}
## ggtree(fitGTR$tree, layout = "slanted") + geom_tiplab()
## ggtree(treeNJ, layout = "slanted") + geom_tiplab()
```

```{r writing unfiltered ASV taxonomy with sequences for supplementary, include = F}
## DHE_taxo_filt %>% 
##   rownames_to_column(var = "ASV_number") %>% 
##   inner_join(DHE_sequence) %>%
##   write.csv(., file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/DHE_unfilt_ASV_list.csv")
```


## Removal of Mitochondria and Chloroplast ASVs

```{r Objects to work with, echio = F}
# all(DHE_sequence$ASV_number == rownames(DHE_counts))
# all(DHE_sequence$ASV_number == rownames(DHE_taxo))
# all(rownames(DHE_taxo) == rownames(DHE_counts))
```

```{r Reading in chloroplast and mitochondria filter generated from phylogenetic tree for unannotated ASVs, include = F}
# read.delim(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/asv_filtering_files/DHE_chloro_filt.txt", header = F) %>%
#   as.data.frame() %>%
#   filter(V1 %in% combined_DHE$ASV_number) -> chloro_phylo_filt
# 
# read.delim(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/asv_filtering_files/DHE_mito_filt.txt", header = F) %>% 
#   as.data.frame() %>%
#   filter(V1 %in% combined_DHE$ASV_number) -> mito_phylo_filt
```

```{r Filtering out annotated chloroplast and mitochondrial ASVs as well as phylogenetic tree identitifed, include = F}
# DHE_taxo %>% 
#   dplyr::filter_all(any_vars(str_detect(., "Mitochondri|Chloroplas"))) %>%
#   rownames() -> mc_filter
# 
# DHE_taxo %>% 
#   rownames_to_column(var = "ASV_number") %>%
#   filter(!ASV_number %in% chloro_phylo_filt$V1) %>% 
#   filter(!ASV_number %in% mito_phylo_filt$V1) %>%
#   filter(!ASV_number %in% mc_filter) -> DHE_taxo_filt
```

```{r checking length after chloro mito removal, include = F}
# nrow(combined_DHE)
# nrow(DHE_taxo)
# nrow(DHE_taxo_filt)
```

combined_DHE has *9959 ASVs*. 
after mito and chloro removal the DHE_taxo has *9182 ASVs*

179 chloroplast and mitochondria annotated
180 chloro from phylo tree
438 mito from phylo tree

```{r Creating the all info file with the chloplast and mitochondria filtered ASVs, include = F}
# DHE_taxo_filt %>% 
#   inner_join(DHE_counts %>% rownames_to_column(var = "ASV_number")) %>% 
#   inner_join(DHE_sequence) -> combined_DHE
# nrow(combined_DHE)
```

```{r files for analysis after mitochondria and chloroplast filtering, include = F}
# combined_DHE %>% 
#   dplyr::select(-domain, -phylum, -class, -order, -family, -genus, -species, -Sequence) %>% 
#   column_to_rownames(var = "ASV_number") %>% 
#   replace(is.na(.), 0) %>% 
#   as.data.frame() -> DHE_counts
# 
# combined_DHE %>% 
#   dplyr::select(ASV_number, domain, phylum, class, order, family, genus, species) %>% 
#   column_to_rownames(var = "ASV_number") -> DHE_taxo
# 
# combined_DHE %>% 
#   dplyr::select(ASV_number, Sequence) -> DHE_sequence
# 
# nrow(DHE_counts)
# nrow(DHE_taxo)
# nrow(DHE_sequence)
# 
# ncol(DHE_counts)
# nrow(treat_DHE_filt)
```

```{r Checking ASV match, include = F}
# all(DHE_sequence$ASV_number == rownames(DHE_counts))
# all(DHE_sequence$ASV_number == rownames(DHE_taxo))
# all(rownames(DHE_taxo) == rownames(DHE_counts))
# all(rownames(treat_DHE_filt) == colnames(DHE_counts))
```

```{r Saving cleaned up files, include = F}
## save(combined_DHE, file = "/Users/benyoung/Dropbox/PhD/Projects/DHE/NGS/16s/r_saved_objects/DHE_alldata_combined_mitochloro_filt.RData")
## save(DHE_taxo_filt, file = "/Users/benyoung/Dropbox/PhD/Projects/DHE/NGS/16s/r_saved_objects/DHE_taxo_unannotated.RData")
## save(DHE_counts, file = "/Users/benyoung/Dropbox/PhD/Projects/DHE/NGS/16s/r_saved_objects/DHE_counts.RData")
## save(DHE_sequence, file = "/Users/benyoung/Dropbox/PhD/Projects/DHE/NGS/16s/r_saved_objects/DHE_sequence.RData")
## save(treat_DHE_filt, file = "/Users/benyoung/Dropbox/PhD/Projects/DHE/NGS/16s/r_saved_objects/treatment_file.RData")
```

File names which have gone through the 5 steps so far
combined_DHE = 9182 ASVs, 331 samples, sequences and taxonomies
DHE_taxo = 9182 ASVs, 331 samples
DHE_counts = 9182 ASVs, 331 samples
DHE_sequence = 9182 samples
treat_DHE_filt = 331 samples (includes inoculations, initial, day10 and intermediates, no filtering of initial dead/diseased yet, will do after NA section)


## Blasting NA ASvs and assigning taxonomy

```{r Generating filtered name variables, include=F}
# DHE_taxo_mat <- as.matrix(DHE_taxo)
# PCA_asv <- as.matrix(DHE_counts)
# PCA_tax <- DHE_taxo_mat
# PCA_sam <- treat_DHE_filt
```

```{r Generating PhyloSeq Object, include=F}
# ASV <- otu_table(PCA_asv, taxa_are_rows = T)
# TAX <- tax_table(PCA_tax)
# SAMP <- sample_data(PCA_sam)
# 
# physeq_main <- phyloseq(ASV, TAX, SAMP)
# physeq_main
```

```{r getting most abundant taxa and writing na4blast file, include = F}
# taxa_sums(physeq_main) %>% #making the sums of ASV counts across all samples
#   as.data.frame() -> ASV_sum
# 
# names(ASV_sum)[names(ASV_sum) == "."] <- "ASV_counts"
# 
# ASV_sum %>%
#   rownames_to_column(var = "ASV") %>% 
#   inner_join(DHE_taxo %>% 
#                rownames_to_column(var = "ASV")) %>% 
#   inner_join(DHE_sequence, 
#              by = c("ASV" = "ASV_number")) %>% 
#   arrange(desc(ASV_counts)) %>% 
#   dplyr::select(-species) %>% 
#   filter_all(any_vars( is.na(.))) -> na4blast
# # View(na4blast)
# 
# write.csv(na4blast, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/na4blast.csv")
```

```{r Full taxonomy dataframe to be used in organising blast results, include = F}
# ASV_sum %>%
#   rownames_to_column(var = "ASV") %>% 
#   inner_join(DHE_taxo %>% 
#                rownames_to_column(var = "ASV")) %>% 
#   inner_join(DHE_sequence, 
#              by = c("ASV" = "ASV_number")) %>% 
#   arrange(desc(ASV_counts)) -> DHE_taxo_counts_all
# # View(DHE_taxo_counts_all)
# 
# write.csv(na4blast, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/DHE_taxo_counts_all.csv")
```

```{r Fasta for Blast, include = F}
# dataframe2fas(na4blast %>%
#                dplyr::select(ASV, Sequence),
#              file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/na4blast.fa")
```

This is the blast command. Follow instructions from NCBI on making local database with NT. 

```{bash blast command for NAs, include = F}
##blastn -db /Users/benyoung/blast/BLASTDB/nt \
##-task megablast \
##-query /Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/na4blast.fa \
##-out /Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/DHE_all_na.out \
##-max_target_seqs 5 \
##-max_hsps 5 \
##-word_size 100 \
##-outfmt "6 qseqid sseqid evalue pident stitle staxids sskingdoms salltitles sscinames sscomnames sblastnames" \
##-num_threads 4
```

```{r taxonimizr database prep, include = F}
## prepareDatabase('accesionTaxa.sql')
```

```{r Getting just accesion numbers, include = F}
## dhe_16s_na <- read.table(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/DHE_all_na.out", 
##                          header = F, 
##                          sep = "\t", 
##                          stringsAsFactors = F)
## 
## dhe_16s_na %>% 
##   dplyr::select(V1, V2) %>%
##   separate(col = V2, into = c("a", "b", "c", "accesion_number") ,sep = "\\|") %>% 
##   dplyr::select(V1, accesion_number) -> an_4_taxo

## View(dhe_16s_na)
## View(an_4_taxo)
```

```{r Getting taxonomy for NAs, include = F}
## taxaID <- accessionToTaxa(an_4_taxo$accesion_number, "accesionTaxa.sql")
## View(taxaID)
```

```{r Running taxonomiser, include = F}
## taxonomy_4_na <- getTaxonomy(taxaID, "accesionTaxa.sql")
## View(taxonomy_4_na)
## 
## taxonomy_4_na %>% 
##   as.data.frame() %>%
##   rownames_to_column(var = "acc_num_rownames") -> taxonomy_4_na_rn
```

```{r condensing by ASV, include = F}
## condenseTaxa(taxonomy_4_na, groupings = dhe_16s_na$V1) -> DHE_na_condensed
```

```{r Complete DHE na taxo from blast, include = F}
## dhe_16s_na %>%
##   dplyr::select(V1, V2, V3, V4, V5, V6) %>%
##   separate(col = V2, into = c("a", "b", "c", "accesion_number") ,sep = "\\|") %>%
##   dplyr::select(-a, -b, -c) %>%
##   dplyr::rename(ASV_number = V1,
##          e_value = V3,
##          percentage_identity = V4,
##          blast_annotation = V5,
##          ncbi_taxonomic_ID = V6) %>%
##   cbind(taxonomy_4_na) -> dhe_complete_na_taxo
##
```

```{r Viewing taxonomies all and condensed, include = F}
## View(dhe_complete_na_taxo)
## View(DHE_na_condensed)
```

```{r writing out for use, include = F}
## write.csv(DHE_na_condensed, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/dhe_na4blast_taxo_condensed.csv")
## write.csv(dhe_complete_na_taxo, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/blast_16s/dhe_na4blast_taxo_complete.csv")
```

Now I have all the componenets to update the taxonomy file woooooo
- Taxonomy matrix (from DADA2 and SILVA)
- Blast results with taxonomies 
- Phylogenetic tree using figtree to place important ASVs into highest taxonomic rank possible. 

Go through and manually annotate (only way unfortunately) the most abundant ASVs. Especially the Serratia ASVs which only annotated to the family level (weird but thats what SILVA did). 

If ASVs are differentially abundant in differential abundance analysis we can use Blast results and Phylogenetic tree to infer taxonomic rank it is in :). 

```{r loading edited taxonomy file from blast and phylogenetic tree annotations, include = F}
# read.csv(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/16s/blast/DHE_taxo_counts_all_annotated.csv") %>% 
#   dplyr::select(-X, -ASV_counts, -notes, -Sequence) %>%
#   column_to_rownames(var = "ASV") -> edited_taxo
# #View(edited_taxo)
```

```{r saving edited DHE taxonomy as r data object, include = F}
## save(edited_taxo,
##      file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/DHE_taxo_annotated.RData")
```


## Checking Sample Numbers for 16s Analysis

```{r Working out number of samples}
# nrow(treat_DHE)
# nrow(treat_DHE_filt)
# 
# # just initial CORAL samples
# treat_DHE_filt %>%
#   filter(., timepoint %in% c("initial")) %>%
#   filter(., !Tank_treatment %in% c("na")) %>% nrow()
# 
# # just day 10  CORAL samples
# treat_DHE_filt %>%
#   filter(., timepoint %in% c("17th_aug")) %>%
#   filter(., !Tank_treatment %in% c("na")) %>% nrow()
# 
# # just initial and day 10 
# treat_DHE_filt %>% 
#   filter(., timepoint %in% c("initial", "17th_aug")) %>%
#   filter(., !Tank_treatment %in% c("na")) %>% nrow()
# 
# # Intermediate CORAL samples (everything except initial and Day 10)
# treat_DHE_filt %>% 
#   filter(., !timepoint %in% c("initial", "17th_aug")) %>% 
#   filter(., !Tank_treatment %in% c("na")) %>% nrow()
# 
# # initial inoculations
# treat_DHE_filt %>% 
#   filter(., Tank_treatment %in% c("na")) %>% 
#   filter(., timepoint %in% c("initial")) %>% nrow()
# 
# # second dose inoculations
# treat_DHE_filt %>% 
#   filter(., Tank_treatment %in% c("na")) %>% 
#   filter(., !timepoint %in% c("initial")) %>% nrow()
```

Removed 4 samples which had counts *< 2000*
Filtered treatment file has *331* samples (initial and all dates up to day 10, also initial and second inoculations)
- INOCULATION SAMPLES -> *15* (9 x initial, 6 x 10th aug)
- Intermediate coral samples -> 40 - 6 = *34*
- Initial coral samples -> 165 - 9 = *157*
- Day 10 coral samples -> 125 - 0 = *125*
- Initial and Day 10 coral samples -> 290 - 9 = *282*
- CORAL inital and day10, and inoculation samples = 281 + 15 -> *297*
- All coral samples an no inoculation samples = *316*

THIS ABOVE ADDS TO 331 WHICH IS GOOD :) 


## R Data Objects for Analysis
### Intial, intermediate, and Day 10 samples

```{r Count matrix and treatment filtering CORAL samples, include=F}
# treat_DHE_filt %>%
#   dplyr::filter(., !Tank_treatment %in% c("na")) -> tfall_coral
# nrow(tfall_coral) #correct number
# 
# tfall_coral %>% 
#   rownames -> inoc_coral
# 
# DHE_counts %>%
#   t() %>% 
#   as.data.frame() %>%
#   rownames_to_column(var = "samps") %>% 
#   dplyr::filter(samps %in% inoc_coral) %>% 
#   column_to_rownames(var = "samps") %>% 
#   t() %>% 
#   as.data.frame() -> DHE_counts_coral
# 
# ncol(DHE_counts_coral)
# nrow(DHE_counts_coral)
# 
# matchup <- match(rownames(tfall_coral), colnames(DHE_counts_coral))
# DHE_counts_coral <- DHE_counts_coral[,matchup]
# all(rownames(tfall_coral) == colnames(DHE_counts_coral))
```

```{r Generating metadata columns for analysis, include=F}
# tfall_coral %>%
#   mutate(
#     Health = case_when(
#       timepoint == "initial" &
#         health_status == "healthy" &
#         Tank_treatment != "na" ~ "pre_exposure",
#       timepoint == "17th_aug" &
#         health_status == "healthy " ~ "visually_healthy_day10",
#       timepoint == "17th_aug" &
#         health_status == "diseased" ~ "diseased_day10",
#       timepoint == "8th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "8th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "9th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "9th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "9th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "10th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "10th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "11th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "11th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "11th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "12th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "12th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "12th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "13th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "13th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "13th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "14th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "14th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "14th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "15th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "15th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "15th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "16th_aug" &
#         health_status == "diseased" ~ "diseased_intermediate",
#       timepoint == "16th_aug" &
#         health_status == "diseased " ~ "diseased_intermediate",
#       timepoint == "16th_aug" &
#         health_status == "dead" ~ "DEAD_intermediate",
#       timepoint == "initial" &
#         health_status == "diseased" ~ "pre_exposure_DISEASED",
#       timepoint == "initial" &
#         health_status == "dead" ~ "pre_exposure_DEAD",
#       Dtreatment == "smarinoc1"  ~ "SMar Inoculation initial",
#       Dtreatment == "smarinoc2"  ~ "SMar Inoculation 10th Aug",
#       Dtreatment == "diseaseslurry1"  ~ "Disease Slurry Inoculation Initial",
#       Dtreatment == "diseaseslurry2"  ~ "Disease Slurry Inoculation 10th Aug",
#       Dtreatment == "healthyslurry"  ~ "Healthy Slurry Inoculation Initial"
#     )) %>% 
#   filter(
#     .,
#     !grepl(
#       "pre_exposure_DISEASED|pre_exposure_DEAD|post_exposure_DEAD",
#       Health
#     )) %>%
#   mutate(
#     Dif_dis_treat = case_when(
#       Health == "pre_exposure" ~ "baseline",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "17th_aug" ~ "Disease_slurry_day10",
#       Dtreatment == "placeboslurry" &
#         timepoint == "17th_aug" ~ "Healthy_slurry_day10",
#       Dtreatment == "smplacebo" &
#         timepoint == "17th_aug" ~ "SM_placebo_day10",
#       Dtreatment == "smarcscens" &
#         timepoint == "17th_aug" ~ "Serratia_day10",
#       Dtreatment == "diseaseslurry" &
#         Health == "diseased_intermediate" ~ "Disease_slurry_intermediate",
#       Dtreatment == "placeboslurry" &
#         Health == "diseased_intermediate" ~ "Healthy_slurry_intermediate",
#       Dtreatment == "smplacebo" &
#         Health == "diseased_intermediate" ~ "SM_placebo_intermediate",
#       Dtreatment == "smarcscens" &
#         Health == "diseased_intermediate" ~ "Serratia_intermediate",
#       Dtreatment == "smarinoc1"  ~ "SMar Inoculation initial",
#       Dtreatment == "smarinoc2"  ~ "SMar Inoculation 10th Aug",
#       Dtreatment == "diseaseslurry1"  ~ "Disease Slurry Inoculation Initial",
#       Dtreatment == "diseaseslurry2"  ~ "Disease Slurry Inoculation 10th Aug",
#       Dtreatment == "healthyslurry"  ~ "Healthy Slurry Inoculation Initial"
#     )
#   ) %>%
#   mutate(
#     geno_health = factor(
#       paste0(
#         .$Genotype,
#         .$Health
#       )
#     ),
#     Health = factor(
#       Health,
#       levels = c(
#         "pre_exposure",
#         "visually_healthy_day10",
#         "diseased_intermediate",
#         "diseased_day10"
#       )
#     ),
#     correct_treat = case_when(
#       Dtreatment == "smplacebo" &
#         timepoint == "initial" ~ "spla_baseline",
#       Dtreatment == "smarcscens" &
#         timepoint == "initial" ~ "smar_baseline",
#       Dtreatment == "smplacebo" &
#         Dif_dis_treat == "SM_placebo_intermediate" ~ "spla_intermediate",
#       Dtreatment == "smarcscens" &
#         Dif_dis_treat == "SM_placebo_intermediate" ~ "smar_intermediate",
#       Dtreatment == "smarcscens" &
#         timepoint == "17th_aug" ~ "smar_day10",
#       Dtreatment == "smplacebo" &
#         timepoint == "17th_aug" ~ "spla_day10",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "initial" ~ "disslu_baseline",
#       Dtreatment == "placeboslurry" &
#         timepoint == "initial" ~ "helslu_baseline",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "17th_aug" ~ "disslu_day10",
#       Dtreatment == "placeboslurry" &
#         timepoint == "17th_aug" ~ "helslu_day10",
#       Dtreatment == "placeboslurry" &
#         Dif_dis_treat == "SM_placebo_intermediate" ~ "helslu_intermediate",
#       Dtreatment == "diseaseslurry" &
#         Dif_dis_treat == "Disease_slurry_intermediate" ~ "disslu_intermediate"
#     )
#   ) %>% 
#   dplyr::filter(!health_status %in% c("dead")) -> treat_DHE_filt_all
# #View(treat_DHE_filt_all)
```

```{r Making Sure Count and treatment same order, include=F}
# ncol(DHE_counts_coral)
# nrow(treat_DHE_filt_all)
# 
# matchup <- match(rownames(treat_DHE_filt_all), colnames(DHE_counts_coral))
# DHE_counts_coral <- DHE_counts_coral[ ,matchup]
# all(rownames(treat_DHE_filt_all) == colnames(DHE_counts_coral))
# 
# nrow(treat_DHE_filt_all)
# ncol(DHE_counts_coral)
# nrow(DHE_counts_coral)
# nrow(edited_taxo)
```

This leaves the treatment file with *307* (original 316 samples) as we have removed DEAD fragments and inoculation samples :). 

```{r Saving ALL SAMPLES DATA, include = F}
# save(treat_DHE_filt_all, 
#      file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/treatment_file_ALLDATES.RData")
# save(edited_taxo, 
#      file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/DHE_taxo_ALLDATES.RData")
# save(DHE_counts_coral, 
#      file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/DHE_counts_ALLDATES.RData")
# save(DHE_sequence, 
#      file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/DHE_ASV_sequences.RData")
```


### Inoculation Samples

```{r Filtering}
# treat_DHE_filt %>% 
#   filter(., Tank_treatment %in% c("na")) -> tfall_inocs
# 
# tfall_inocs %>% 
#   rownames -> inoc_filt
# 
# DHE_counts %>% 
#   t() %>% 
#   as.data.frame() %>%
#   rownames_to_column(var = "samps") %>% 
#   filter(samps %in% inoc_filt) %>% 
#   column_to_rownames(var = "samps") %>% 
#   t() %>% 
#   as.data.frame() -> DHE_counts_inoc
# 
# nrow(DHE_counts_inoc)
# ncol(DHE_counts_inoc)
```

```{r Matc hing up Inoculation samples information, echo = F}
# matchup <- match(rownames(tfall_inocs), colnames(DHE_counts_inoc))
# DHE_counts_inoc <- DHE_counts_inoc[,matchup]
# all(rownames(tfall_inocs) == colnames(DHE_counts_inoc))
# 
# ncol(DHE_counts_inoc)
# nrow(tfall_inocs)
# nrow(DHE_counts_inoc)
# nrow(edited_taxo)
```

```{r Saving inoc info, include = F}
# save(DHE_counts_inoc, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/inoculations_16s/DHE_counts_inoc.RData")
# save(tfall_inocs, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/inoculations_16s/DHE_treat_inoc.RData")
# save(edited_taxo, file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/inoculations_16s/DHE_taxo_inoc.RData")
```

NB: can use the edited taxo file as that is the same, only count and treatment different. 

----------

# Analysis of Inoculation 16s Samples
## Reading in and Data Prep for Incoulation Data
```{r reading in saved data, include = F}
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/inoculations_16s/DHE_counts_inoc.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/inoculations_16s/DHE_treat_inoc.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/inoculations_16s/DHE_taxo_inoc.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/phylogenetic_tree_16s/fitGTR.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/DHE_ASV_sequences.RData")
```

```{r Checking all lengths and order correct, include = T}
nrow(tfall_inocs)
ncol(DHE_counts_inoc)
nrow(DHE_counts_inoc)
nrow(edited_taxo)
rownames(tfall_inocs) == colnames(DHE_counts_inoc)
```

```{r making new variables in metadata, include = F}
tfall_inocs %>%
  mutate(Dif_dis_treat = 
           case_when(
             Dtreatment == "smarinoc1" ~ "Serratia",
             Dtreatment == "smarinoc2" ~ "Serratia", 
             Dtreatment == "diseaseslurry1" ~ "Disease_slurry",
             Dtreatment == "diseaseslurry2" ~ "Disease_slurry", 
             Dtreatment == "healthyslurry" ~ "Healthy_slurry"
           )
  ) -> tfall_inocs
```

```{r phyloseq of inoc data}
PCA_asv <- as.matrix(DHE_counts_inoc)
PCA_tax <- as.matrix(edited_taxo)
PCA_sam <- tfall_inocs

ASV <- otu_table(PCA_asv, taxa_are_rows = T)
TAX <- tax_table(PCA_tax)
SAMP <- sample_data(PCA_sam)

physeq_main <- phyloseq(ASV, 
                        TAX, 
                        SAMP, 
                        fitGTR$tree)

physeq_main
physeq_main <- prune_taxa(taxa_sums(physeq_main) > 1, physeq_main)
physeq_main

physeq_inoculation <- filter_taxa(physeq_main, function(x) mean(x) > 4, TRUE)

ntaxa(physeq_main) #9182
ntaxa(physeq_inoculation) #445

# View(otu_table(physeq_inoculation))
# View(tax_table(physeq_inoculation))

length(get_taxa_unique(physeq_main, "genus")) # 548
length(get_taxa_unique(physeq_inoculation, "genus")) #106
```

With filtering at 1, leaves 
- 1519 taxa and 15 samples

```{r Taxonomy Lengths, echo = T}
length(get_taxa_unique(physeq_inoculation, "domain")) #3
length(get_taxa_unique(physeq_inoculation, "phylum")) #16
length(get_taxa_unique(physeq_inoculation, "class")) #23
length(get_taxa_unique(physeq_inoculation, "order")) #49
length(get_taxa_unique(physeq_inoculation, "family")) #67
length(get_taxa_unique(physeq_inoculation, "genus")) #106
```


## Principal Component Analysis  

```{r Pruned data objects for PCA analysis, echo = F}
ASV_inoc = as(otu_table(physeq_inoculation), "matrix")
ASV_taxon = as(tax_table(physeq_inoculation), "matrix")
ASV_metadata = as(sample_data(physeq_inoculation), "matrix")

all(colnames(ASV_inoc) == rownames(ASV_metadata))
```

```{r writing ASV for analysis inocualtion, include = F}
# write.csv(ASV_taxon %>% 
#             as.data.frame() %>%
#             rownames_to_column(var = "ASV_number") %>% 
#             inner_join(DHE_sequence), 
#           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/inoc_info/inoc_ASVs_to_Sequence.csv")


```

```{r CLR Transformation, include = F}
cmultRepl(t(ASV_inoc), method="CZM", label=0) -> ASV_mat_inoc
codaSeq.clr(ASV_mat_inoc, 
            samples.by.row = T) -> ASV_mat_CLR_inoc
t(ASV_mat_CLR_inoc) -> ASV_mat_CLR_inoc

allsamps_inoc <- pca(ASV_mat_CLR_inoc, 
                metadata = ASV_metadata, 
                removeVar = 0.1)
```

```{r Plots from PCAtools, echo = F, fig.width=7, fig.height=4}
## Plot showing the genes which drive the principal component variances. 
plotloadings(allsamps_inoc, labSize = 3)
```

```{r Plots from PCAtools, fig.width=8, fig.height=6}
## Scree plot showing amount of variance explained by each PC (bars) and cumulative variance as you progress along bars (line)
screeplot(allsamps_inoc, 
          getComponents(allsamps_inoc, 1:10), 
          axisLabSize = 10, 
          titleLabSize = 10, 
          returnPlot = T, 
          ylim = c(0,100), 
          vline = c(findElbowPoint(allsamps_inoc$variance))) +
  geom_label(aes(x = findElbowPoint(allsamps_inoc$variance) + 1, y = 50,
      label = 'Elbow method', vjust = -1, size = 4))

# Shows the main ASV weights driving the variance
biplot(
  allsamps_inoc,
  colby = "Dtreatment",
  hline = 0,
  vline = 0,
  legendPosition = 'right', 
  labSize = 2,
  pointSize = 2,
  legendLabSize = 4,
  legendTitleSize = 8,
  axisLabSize = 8
)
## Plotting of a large number of PCs
#pairsplot(allsamps)

## Plot showing the samples with the strongest loadings of ASVs for each axes identyfying ones which are driving the variance
#plotloadings(allsamps, labSize = 3)
```

```{r PCA prep for GGPlots, include = F}
d.czm <- cmultRepl(t(ASV_inoc), 
                   method="CZM", 
                   label=0)

d.clr <- codaSeq.clr(d.czm)
pca_samp <- prcomp(d.clr)
sample_loadings <- as.data.frame(pca_samp$x)
summary(pca_samp)
fviz_eig(pca_samp, addlabels = TRUE, ylim = c(0, 70), )
```

```{r Percentage Loadings of PC Axis for ggplot plots, include = F}
pc1 <- round(pca_samp$sdev[1]^2/sum(pca_samp$sdev^2),2) * 100
pc2 <- round(pca_samp$sdev[2]^2/sum(pca_samp$sdev^2),2) * 100
pc3 <- round(pca_samp$sdev[3]^2/sum(pca_samp$sdev^2),2) * 100
pc4 <- round(pca_samp$sdev[4]^2/sum(pca_samp$sdev^2),2) * 100
pc5 <- round(pca_samp$sdev[5]^2/sum(pca_samp$sdev^2),2) * 100
```

```{r PCA Timepoint, fig.width=11, fig.height=7}
ggplot(sample_loadings, aes(PC1, PC2, fill = PCA_sam$Dtreatment, colour = PCA_sam$Dtreatment)) + 
  geom_point(size = 4) + 
  labs(x = paste("PC1: ", pc1, "% variance", sep = ""), 
       y = paste("PC2: ", pc2, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 10)
  ) + 
  scale_color_manual(values = c("navajowhite2", "navajowhite4", "lightseagreen", "violetred1", "violetred3")) +
  geom_polygon(fill = "NA")

# ggplot(sample_loadings, aes(PC2, PC3, fill = PCA_sam$Dtreatment, colour = PCA_sam$Dtreatment)) +
#   geom_point(size = 2) +
#   labs(x = paste("PC2: ", pc2, "% variance", sep = ""),
#        y = paste("PC3: ", pc3, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) +
#   geom_polygon()
# 
# ggplot(sample_loadings, aes(PC3, PC4, fill = PCA_sam$Dtreatment, colour = PCA_sam$Dtreatment)) +
#   geom_point(size = 2) +
#   labs(x = paste("PC3: ", pc3, "% variance", sep = ""),
#        y = paste("PC4: ", pc4, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) +
#   geom_polygon()
# 
# ggplot(sample_loadings, aes(PC4, PC5, fill = PCA_sam$Dtreatment, colour = PCA_sam$Dtreatment)) +
#   geom_point(size = 2) +
#   labs(x = paste("PC4: ", pc4, "% variance", sep = ""),
#        y = paste("PC5: ", pc5, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) +
#   geom_polygon()
```

```{r PCA Dif dis treat}
ggplot(sample_loadings, aes(PC1, PC2, fill = PCA_sam$Dif_dis_treat, colour = PCA_sam$Dif_dis_treat)) + 
  geom_point(size = 3) + 
  labs(x = paste("PC1: ", pc1, "% variance", sep = ""), 
       y = paste("PC2: ", pc2, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + 
  geom_polygon()

# ggplot(sample_loadings, aes(PC2, PC3, fill = PCA_sam$Dif_dis_treat, colour = PCA_sam$Dif_dis_treat)) + 
#   geom_point(size = 2) + 
#   labs(x = paste("PC2: ", pc2, "% variance", sep = ""), 
#        y = paste("PC3: ", pc3, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) + 
#   geom_polygon()
# 
# ggplot(sample_loadings, aes(PC3, PC4, fill = PCA_sam$Dif_dis_treat, colour = PCA_sam$Dif_dis_treat)) + 
#   geom_point(size = 2) + 
#   labs(x = paste("PC3: ", pc3, "% variance", sep = ""), 
#        y = paste("PC4: ", pc4, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) + 
#   geom_polygon()
# 
# ggplot(sample_loadings, aes(PC4, PC5, fill = PCA_sam$Dif_dis_treat, colour = PCA_sam$Dif_dis_treat)) + 
#   geom_point(size = 2) + 
#   labs(x = paste("PC4: ", pc4, "% variance", sep = ""), 
#        y = paste("PC5: ", pc5, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) + 
#   geom_polygon()
```

## Alpha Analysis

```{r divnet-rs prep, include = F}
# otu_table(physeq_main) %>%
#   as.data.frame() %>%
#   rownames_to_column(var = "taxa") %>%
#   write.csv(
#     .,
#     "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_inoculations_counts_4_divnet.csv",
#     row.names = F
#   )
```

```{r Sample data for divnet, include = F}
# all(rownames(tfall_inocs) == colnames(otu_table(physeq_main)))
# 
# # this divent model is looking at Dtreat outcome, as PCA does not really show genet difs its fine.
# model.matrix( ~ Dtreatment - 1, tfall_inocs) %>%
#   as.data.frame() %>%
#   rownames_to_column(var = "sample") %>%
#   write.csv(., 
#             file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_inoculations_samples_4_divnet.csv",
#             row.names = F)
```

```{bash divnet in rust, include = F}
# divnet-rs /Users/benyoung/bin/divnet-rs-0.2.1/DHE/inoculation_analysis/DHE_divnet_inoc.toml
```

```{r Reading in Divnet RS, include = F}
divnet_rs <- read.table("/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_divnet_output_inoc.csv", 
                        sep = ",",
                        header = TRUE)
# View(divnet_rs)

nreplicates <- 5
# Replicate 0 is actually the estimates for the real data.
rep0 <- divnet_rs[divnet_rs$replicate == 0, -1]
rownames(rep0) <- rep0$sample
rep0$sample <- NULL
#View(divnet_rs)
```

```{r Making the Shannon and Simpson results for actual Data, include =F}
rep0_shannon <- apply(rep0, 1, DivNet::shannon_true)
rep0_simpson <- apply(rep0, 1, DivNet::simpson_true)
```

```{r Shannon and Simspon for the Replicates, include = F}
# Now calculate the shannon index for the replicates.
reps_shannon <- sapply(1:nreplicates, function (i) {
  d <- divnet_rs[divnet_rs$replicate == i, -1]
  rownames(d) <- d$sample
  d$sample <- NULL
  apply(d, 1, DivNet::shannon_true)
})

reps_simpson <- sapply(1:nreplicates, function (i) {
  d <- divnet_rs[divnet_rs$replicate == i, -1]
  rownames(d) <- d$sample
  d$sample <- NULL

  apply(d, 1, DivNet::simpson_true)
})
```

```{r Shannon and Simpson Diversity Estimates for replicates, include = F}
# What we want is the variance in the diversity estimates for the replicates.
reps_shannon_error <- t(apply(reps_shannon, 1, function (x) {
  c(var(x), sd(x))
}))
colnames(reps_shannon_error) <- c("variance", "sd")

# What we want is the variance in the diversity estimates for the replicates.
reps_simpson_error <- t(apply(reps_simpson, 1, function (x) {
  c(var(x), sd(x))
}))
colnames(reps_simpson_error) <- c("variance", "sd")
```

```{r Making nice dataframes so we can group and make nice plots, echo=T}
tibble(names = names(rep0_shannon),
       shannon = rep0_shannon) %>%
  left_join(reps_shannon_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = shannon + 2 * sd,
         cilower = shannon - 2 * sd) %>% 
  filter(names %in% rownames(tfall_inocs)) %>% 
  inner_join(tfall_inocs %>% 
              rownames_to_column(var = "names")) %>% 
  column_to_rownames(var = "names") -> alpha_shannon_treat

tibble(names = names(rep0_simpson),
       simpson = rep0_simpson) %>%
  left_join(reps_simpson_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = simpson + 2 * sd,
         cilower = simpson - 2 * sd) %>%
  inner_join(tfall_inocs %>% 
              rownames_to_column(var = "names")) %>%
  column_to_rownames(var = "names") -> alpha_simpson_treat
```

```{r writing alpha analysis to file, include = F}
# write.csv(alpha_simpson_treat, 
#           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/inoc_info/simp.csv")
# 
# write.csv(alpha_shannon_treat, 
#           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/inoc_info/shan.csv")
```


#### Simspon

```{r SIMPSON PLOTTING Dtreatment, echo = T, fig.width=5, fig.height=6}
ggplot(data = alpha_simpson_treat %>% rownames_to_column(var = "samples"), 
       aes(x = Dtreatment, y = simpson, color = Dtreatment, group = Dtreatment)) + 
  geom_point(position = position_dodge(width=0.5), 
             size = 2) + 
  geom_linerange(aes(ymin = cilower, ymax = ciupper), 
                 position = position_dodge(.5), 
                 size = 1) + 
  theme_bw() + 
  scale_color_manual(values = c("navajowhite2", "navajowhite4", "lightseagreen", "violetred1", "violetred3"))

tibble(names = names(rep0_simpson),
       simpson = rep0_simpson) %>%
  left_join(reps_simpson_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = simpson + 2 * sd,
         cilower = simpson - 2 * sd) %>%
  ggplot(aes(x = names, color = alpha_simpson_treat$Dtreatment)) +
  geom_segment(aes(xend = names, y = cilower, yend = ciupper)) +
  geom_point(aes(y = simpson)) +
  xlab("samples") +
  ylab("simpson estimate") +
  ggtitle("divnet-rs") +
  theme_bw() + 
  theme(panel.grid.minor.x = element_blank(),
        axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) + 
  theme_bw()
```

```{r}
alpha_simpson_treat %>%
  mutate(smar1 = relevel(factor(Dtreatment), ref = "smarinoc1"),
         smar2 = relevel(factor(Dtreatment), ref = "smarinoc2"), 
         disslu1 = relevel(factor(Dtreatment), ref = "diseaseslurry1"), 
         disslu2 = relevel(factor(Dtreatment), ref = "diseaseslurry2"), 
         disslu = relevel(factor(Dif_dis_treat), ref = "Disease_slurry"), 
         ser = relevel(factor(Dif_dis_treat), ref = "Serratia")) -> alpha_simpson_treat
```

```{r hypothesis testing for disslu and smar, include = F}
inoculated_smar1 <- betta(
  chats = alpha_simpson_treat$simpson,
  ses = sqrt(alpha_simpson_treat$variance),
  X = model.matrix( ~ smar1, data = alpha_simpson_treat)
)
#View(inoculated_smar1$table)

inoculated_disslu1 <- betta(
  chats = alpha_simpson_treat$simpson,
  ses = sqrt(alpha_simpson_treat$variance),
  X = model.matrix( ~ disslu1, data = alpha_simpson_treat)
)
#View(inoculated_disslu1$table)
```

inoculated_smar1 -> no significant difs between smar inoc 1 and smar inoc 2
inoculated_disslu1 -> no significant difs between disslu1 and disslu2

```{r hypothesis testing, include = F}
inoculated_disslu <- betta(
  chats = alpha_simpson_treat$simpson,
  ses = sqrt(alpha_simpson_treat$variance),
  X = model.matrix( ~ disslu, data = alpha_simpson_treat)
)
#View(inoculated_disslu$table)

inoculated_ser <- betta(
  chats = alpha_simpson_treat$simpson,
  ses = sqrt(alpha_simpson_treat$variance),
  X = model.matrix( ~ ser, data = alpha_simpson_treat)
)
#View(inoculated_ser$table)
```

inoculated_disslu -> signif dif betwen disslu and healthy, and disslu serratia
inoculated_ser -> signif between disslu and ser, not ser and healthy slurry


#### Shannon

```{r Plots for Shannon, echo = F, fig.width=5, fig.height=6}
ggplot(data = alpha_shannon_treat %>% rownames_to_column(var = "samples"), 
       aes(x = Dtreatment, y = shannon, color = Dtreatment, group = Dtreatment)) + 
  geom_point(position = position_dodge(width=0.5), 
             size = 2) + 
  geom_linerange(aes(ymin = cilower, ymax = ciupper), 
                 position = position_dodge(.5), 
                 size = 1) +
  theme_bw() + 
  scale_color_manual(values = c("navajowhite2", "navajowhite4", "lightseagreen", "violetred1", "violetred3"))

tibble(names = names(rep0_shannon),
       shannon = rep0_shannon) %>%
  left_join(reps_shannon_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = shannon + 2 * sd,
         cilower = shannon - 2 * sd) %>%
  ggplot(aes(x = names, color = alpha_shannon_treat$Dtreatment)) +
  geom_segment(aes(xend = names, y = cilower, yend = ciupper)) +
  geom_point(aes(y = shannon)) +
  xlab("samples") +
  ylab("shannon estimate") +
  ggtitle("divnet-rs") +
  theme_bw() +
  theme(panel.grid.minor.x = element_blank(),
        axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
  theme_bw()
```

#### Shannon Hyp Testing

```{r}
alpha_shannon_treat %>%
  mutate(smar1 = relevel(factor(Dtreatment), ref = "smarinoc1"),
         smar2 = relevel(factor(Dtreatment), ref = "smarinoc2"), 
         disslu1 = relevel(factor(Dtreatment), ref = "diseaseslurry1"), 
         disslu2 = relevel(factor(Dtreatment), ref = "diseaseslurry2"), 
         disslu = relevel(factor(Dif_dis_treat), ref = "Disease_slurry"), 
         ser = relevel(factor(Dif_dis_treat), ref = "Serratia")) -> alpha_shannon_treat
```


```{r}
inoculated_smar1 <- betta(
  chats = alpha_shannon_treat$shannon,
  ses = sqrt(alpha_shannon_treat$variance),
  X = model.matrix( ~ smar1, data = alpha_shannon_treat)
)
View(inoculated_smar1$table)

inoculated_disslu1 <- betta(
  chats = alpha_shannon_treat$shannon,
  ses = sqrt(alpha_shannon_treat$variance),
  X = model.matrix( ~ disslu1, data = alpha_shannon_treat)
)
View(inoculated_disslu1$table)
```

inoculated_smar1 -> no significant difs between smar inoc 1 and smar inoc 2
inoculated_disslu1 -> no significant difs between disslu1 and disslu2

```{r}
inoculated_disslu <- betta(
  chats = alpha_shannon_treat$shannon,
  ses = sqrt(alpha_shannon_treat$variance),
  X = model.matrix( ~ disslu, data = alpha_shannon_treat)
)
View(inoculated_disslu$table)

inoculated_ser <- betta(
  chats = alpha_shannon_treat$shannon,
  ses = sqrt(alpha_shannon_treat$variance),
  X = model.matrix( ~ ser, data = alpha_shannon_treat)
)
View(inoculated_ser$table)
```

inoculated_disslu -> signif dif betwen disslu and healthy, and disslu vs serratia
inoculated_ser -> signif between disslu and ser, and ser and healthy slurry



Intercept is the disease slurry diseased

Interpreting results
- For visually healthy, see an increase of richness of *25.97* taxa, with the SE of *15.75*. 
- For visually healthy, this not significant (hypothesis test for change in richness not be rejected) as p-val = *0.099*. 
- For diseased, see an increase of richness of *100.11* taxa, with SE of *33.18*. 
- For diseased, this is significant (hypothesis test for change in richness can be rejected) as p-val = *0.003*. 


### Relative Abundance

```{r Rel Abun Long Format Data Frame, include = F}
physeq_inoculation %>%
  tax_glom(taxrank = "genus") %>% # agglomerate at family level
  transform_sample_counts(function(x) {
    x / sum(x)
  }) %>% # Transform to rel. abundance
  psmelt() %>%  # Melt to long format
  arrange(genus) -> gen_abundance
head(gen_abundance)
```

```{r Filtering at .01, include = F}
gen_abundance %>%
  dplyr::select(genus, Dtreatment, Abundance) %>%
  dplyr::group_by(genus, Dtreatment) %>%
  dplyr::summarize(avg_abundance = mean(Abundance)) %>%
  dplyr::filter(avg_abundance > 0.005) -> filt_gen_abundance
View(filt_gen_abundance)

ASV_taxon %>% 
  as.data.frame() %>%
  dplyr::select(phylum, class, order, genus) %>% 
  distinct() %>% 
  inner_join(filt_gen_abundance) %>%
              as.data.frame() -> filt_gen_abundance
View(filt_gen_abundance)
```

```{r}
ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = Dtreatment, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE,
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  )
```

```{r Creating Plot, echo = F, fig.height=9, fig.width=8}
library(RColorBrewer)
library(randomcoloR)

ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = Dtreatment, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE, 
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  ) + 
  scale_fill_manual(values = c("darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen", "deeppink", "khaki2", "firebrick", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey"))
#  scale_x_discrete(guide = guide_axis(n.dodge = 2))
  
ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = Dtreatment, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE, 
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  ) + 
  scale_fill_manual(values = colorRampPalette(brewer.pal(11, "Dark2"))(25))

set.seed(13)
ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = Dtreatment, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE, 
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  ) + 
  scale_fill_manual(values = distinctColorPalette(25))
```

```{r relative abundance information, include = F}
# write.csv(filt_gen_abundance, 
#           file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/inoc_info/rel_abu.csv")
```




----------

# Analysis of Coral 16s Samples of Inoculated Acropora palmata Samples
## Reading in All Coral 16s Data For analysis and Additional Sample Prep

```{r loading saved data objects needed for all coral samples analysis, echo = F}
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/treatment_file_ALLDATES.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/DHE_taxo_ALLDATES.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/DHE_counts_ALLDATES.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects//phylogenetic_tree_16s/fitGTR.RData")
load(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/coral_samples_all_for_analysis/DHE_ASV_sequences.RData")

nrow(treat_DHE_filt_all)
ncol(DHE_counts_coral)
nrow(DHE_counts_coral)
nrow(edited_taxo)

all(rownames(treat_DHE_filt_all) == colnames(DHE_counts_coral))
```

```{r Writing edited taxonomy file with sequences for supplemental, include = F}
## View(DHE_sequence)
# edited_taxo %>% 
#   rownames_to_column(var = "ASV_number") %>% 
#   inner_join(DHE_sequence) %>% View()
#   write.csv(., file = "~/Dropbox/PhD/Projects/DHE/NGS/16s/dissertation_files/raw_lists/DHE_filt_ASV_list.csv")
```

```{r Removing serratia placebo and healthy slurry samples that became infectd, include = F}
treat_DHE_filt_all %>% 
  dplyr::filter(Dtreatment == c("smplacebo") & 
                  health_status == "diseased") %>%
  rownames() -> smplac_dis_filt
length(smplac_dis_filt)

treat_DHE_filt_all %>% 
  dplyr::filter(Dtreatment == c("placeboslurry") & 
                  health_status == "diseased") %>% 
  rownames() -> helslu_dis_filt
length(helslu_dis_filt)

nrow(treat_DHE_filt_all)
treat_DHE_filt_all %>% 
  rownames_to_column(var = "samples") %>%
  dplyr::filter(!samples %in% helslu_dis_filt) %>% 
  dplyr::filter(!samples %in% smplac_dis_filt) %>% 
  column_to_rownames(var = "samples") -> treat_DHE_filt_all
nrow(treat_DHE_filt_all)
```

```{r Generating additinal metadata variables for all coral sample analysis, include = F}
treat_DHE_filt_all %>%
  mutate(
    geno_health = factor(
      paste0(treat_DHE_filt_all$Genotype, treat_DHE_filt_all$Health)
    ),
    Health = factor(
      Health,
      levels = c(
        "pre_exposure",
        "visually_healthy_day10",
        "diseased_intermediate",
        "diseased_day10"
      )
    ),
    correct_treat = case_when(
      Dtreatment == "smplacebo" &
        timepoint == "initial" ~ "spla_baseline",
      Dtreatment == "smarcscens" &
        timepoint == "initial" ~ "smar_baseline",
      Dtreatment == "smplacebo" &
        Dif_dis_treat == "SM_placebo_intermediate" ~ "spla_intermediate",
      Dtreatment == "smarcscens" &
        Dif_dis_treat == "Serratia_intermediate" ~ "smar_intermediate",
      Dtreatment == "smarcscens" &
        timepoint == "17th_aug" ~ "smar_day10",
      Dtreatment == "smplacebo" &
        timepoint == "17th_aug" ~ "spla_day10",
      Dtreatment == "diseaseslurry" &
        timepoint == "initial" ~ "disslu_baseline",
      Dtreatment == "placeboslurry" &
        timepoint == "initial" ~ "helslu_baseline",
      Dtreatment == "diseaseslurry" &
        timepoint == "17th_aug" ~ "disslu_day10",
      Dtreatment == "placeboslurry" &
        timepoint == "17th_aug" ~ "helslu_day10",
      Dtreatment == "placeboslurry" &
        Dif_dis_treat == "SM_placebo_intermediate" ~ "helslu_intermediate",
      Dtreatment == "diseaseslurry" &
        Dif_dis_treat == "Disease_slurry_intermediate" ~ "disslu_intermediate"
    ),
    Health_split = Health,
    HI = case_when(
      Health == "pre_exposure" ~ "pre_exposure",
      Health == "visually_healthy_day10" ~ "visually_healthy",
      Health == "diseased_intermediate" ~ "diseased",
      Health == "diseased_day10" ~ "diseased"
    ),
    Inoculated = case_when(
      Health == "pre_exposure" ~ "control",
      HI == "diseased" ~ "inoculated",
      HI == "visually_healthy" ~ "inoculated"
    ), 
    basic_health = case_when(
      Health == "visually_healthy_day10" ~ "visually_healthy", 
      Health == "diseased_day10" ~ "diseased", 
      Health == "diseased_intermediate" ~ "diseased", 
      Health == "pre_exposure" ~ "pre_exposure"
    ), 
    sampling_timepoint = case_when(timepoint == "initial" ~ "control", 
                                   timepoint == "17th_aug" ~ "day_10",
                                   timepoint == "8th_aug" ~ "day_1_9",
                                   timepoint == "9th_aug" ~ "day_1_9", 
                                   timepoint == "10th_aug" ~ "day_1_9", 
                                   timepoint == "11th_aug" ~ "day_1_9", 
                                   timepoint == "12th_aug" ~ "day_1_9", 
                                   timepoint == "13th_aug" ~ "day_1_9", 
                                   timepoint == "14th_aug" ~ "day_1_9", 
                                   timepoint == "15th_aug" ~ "day_1_9", 
                                   timepoint == "16th_aug" ~ "day_1_9"), 
    Dif_dis_treat = case_when(
      basic_health == "pre_exposure" ~ "baseline",
      Dtreatment == "diseaseslurry" &
        timepoint == "17th_aug" ~ "Disease_slurry",
      Dtreatment == "smarcscens" & 
        timepoint == "10th_aug" ~ "Serratia", 
      Dtreatment == "diseaseslurry" & 
        timepoint == "8th_aug" ~ "Disease_slurry",
      Dtreatment == "placeboslurry" &
        timepoint == "17th_aug" ~ "Healthy_slurry",
      Dtreatment == "smplacebo" &
        timepoint == "17th_aug" ~ "SM_placebo",
      Dtreatment == "smarcscens" &
        timepoint == "17th_aug" ~ "Serratia",
      Dtreatment == "diseaseslurry" &
        timepoint == "9th_aug" ~ "Disease_slurry",
      Dtreatment == "diseaseslurry" &
        timepoint == "10th_aug" ~ "Disease_slurry",
      Dtreatment == "diseaseslurry" &
        timepoint == "11th_aug" ~ "Disease_slurry",
      Dtreatment == "diseaseslurry" &
        timepoint == "12th_aug" ~ "Disease_slurry",
      Dtreatment == "diseaseslurry" &
        timepoint == "16th_aug" ~ "Disease_slurry",
      Dtreatment == "smarcscens" &
        timepoint == "9th_aug" ~ "Serratia",
      Dtreatment == "smarcscens" &
        timepoint == "11th_aug" ~ "Serratia",
      Dtreatment == "smarcscens" &
        timepoint == "12th_aug" ~ "Serratia",
      Dtreatment == "smarcscens" &
        timepoint == "13th_aug" ~ "Serratia",
      Dtreatment == "smarcscens" &
        timepoint == "15th_aug" ~ "Serratia",
      Dtreatment == "smarcscens" &
        timepoint == "16th_aug" ~ "Serratia")) %>%
  dplyr::mutate(Health = relevel(as.factor(Health), "pre_exposure"), 
                basic_health = as.factor(basic_health)) -> tfall_coral
```

```{r Few more variables for analysis, include = F}
tfall_coral %>% 
  mutate(Health_disease = paste0(basic_health, Dif_dis_treat)) -> tfall_coral
```

```{r Ordering counts and treatment file after removal of samples, echo = F}
ncol(DHE_counts_coral)
nrow(tfall_coral)

matchup <- match(rownames(tfall_coral), colnames(DHE_counts_coral))
DHE_counts_coral <- DHE_counts_coral[ ,matchup]
all(rownames(tfall_coral) == colnames(DHE_counts_coral))

nrow(treat_DHE_filt_all)
ncol(DHE_counts_coral)
nrow(DHE_counts_coral)
nrow(edited_taxo)
```

```{r Summaries of Sample Numbers based on metadata variables, echo = F}
tfall_coral %>%
  group_by(Health) %>%
  summarise(count = n())

# tfall_coral %>% 
#   group_by(Genotype, HI) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files//DHE_16s_sample_sum.csv")
```

All matched up. For all coral samples we have 294 samples.

```{r samples for venn with transcriptomics, include = F}
tfall_coral %>% 
  mutate(microbe_samp_venn = paste0(.$Genotype, .$HI, .$Frag_number)) %>% 
  dplyr::select(microbe_samp_venn) -> microbe_samp_venn
```


## Generating Phyloseq Object for Analysis

```{r Generating filtered name variables, include=F}
DHE_taxo_mat <- as.matrix(edited_taxo)

PCA_asv <- as.matrix(DHE_counts_coral)
PCA_tax <- DHE_taxo_mat
PCA_sam <- tfall_coral
```

```{r Generating PhyloSeq Object, include=F}
ASV <- otu_table(PCA_asv, taxa_are_rows = T)
TAX <- tax_table(PCA_tax)
SAMP <- sample_data(PCA_sam)

physeq_main <- phyloseq(ASV, TAX, SAMP, fitGTR$tree)
physeq_main
```

```{r removing singletons and zeros from physeq object, echo = F}
prune_taxa(taxa_sums(physeq_main) > 0, physeq_main) %>% 
  prune_taxa(taxa_sums(.) > 1, .) -> physeq_sing_zero_removed
physeq_sing_zero_removed
```

```{r additional filtering out of ASVs with less than 7 across all samples , include= = F}
physeq_sing_zero_removed
physeq_main_10<- prune_taxa(taxa_sums(physeq_sing_zero_removed) > 10, physeq_sing_zero_removed)
physeq_main_10       

ntaxa(physeq_main)
ntaxa(physeq_sing_zero_removed)
ntaxa(physeq_main_10)

length(get_taxa_unique(physeq_main, "genus")) # 1254
length(get_taxa_unique(physeq_sing_zero_removed, "genus")) #466
length(get_taxa_unique(physeq_main_10, "genus")) #2413
```

This has left the following

```{r physeq object with 7 filtering, echo = F}
physeq_main_10
ntaxa(physeq_main_10)
```

- Raw ASV file
- ASV after filtering
- Mito filter
- Chloroplast filter

```{r files for supplemental, include = F}

```


## PiCrust Work 
### Making Files for PiCrust

```{r making objects we need for picrust, echo = F}
otu_table(physeq_main_10) %>% 
  as.data.frame() -> ASV_4_picrust

tfall_coral %>% 
  as.data.frame() %>% 
  dplyr::select(1, 7, 9, 13, 19) %>% 
  rownames_to_column(var = "SampleID") -> metadata_4_picrust

metadata_4_picrust$SampleID == colnames(ASV_4_picrust)

DHE_sequence %>% 
  dplyr::filter(ASV_number %in% rownames(ASV_4_picrust)) -> seq_4_picrust
```

```{r writing metadata as tsv, include = F}
# write_tsv(metadata_4_picrust, 
#           file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/picrust_16s/metadata.tsv")
```

```{r writing fna for fasta file for picrust, include = F}
# dataframe2fas(seq_4_picrust %>%
#                dplyr::select(ASV_number, Sequence),
#              file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/picrust_16s/seqs.fna")
```

```{r writing count matrix as biom file for picrust, include = F}
# make_biom(ASV_4_picrust) -> count_4_picrust_biom
# write_biom(count_4_picrust_biom, 
#            biom_file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/picrust_16s/table.biom")
```

 https://nmbu.brage.unit.no/nmbu-xmlui/bitstream/handle/11250/2642359/Thesis_finished_for_real.pdf?isAllowed=y&sequence=1 
### Running PiCrust 

PiCrust is run in conda and unix so all chunks below are bash and have been commented out as I am running them in the terminal. 

```{bash installing and creating a picrust conda environment}
# conda create -n picrust2 -c bioconda -c conda-forge picrust2=2.4.2
```

```{bash activating conda environment for picrust}
# conda activate picrust2
```

```{bash placing ASV sequences into picrust reference tree}
#cd ~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript 
#place_seqs.py -s r_saved_files/picrust_16s/seqs.fna \
#-o picrust_output/out.tre \
#-p 1 \
#--intermediate picrust_output/intermediate/place_seqs
```

This is the set of poorly aligned input sequences to be excluded: ASV_2757, ASV_7880, ASV_3005, ASV_637, ASV_1828

```{bash hidden-state prediciton of gene famalies}
#hsp.py -i 16s \
#-t picrust_output/out.tre \
#-o picrust_output/marker_predicted_and_nsti.tsv.gz \
#-p 2 \
#-n

#hsp.py -i EC \
#-t picrust_output/out.tre \
#-o picrust_output/EC_predicted.tsv.gz \
#-p 2

#hsp.py -i KO \
#-t picrust_output/out.tre \
#-o picrust_output/KO_predicted.tsv.gz \
#-p 2
```

```{bash generating metagenome predictions}
#metagenome_pipeline.py -i r_saved_files/picrust_16s/table.biom \
#-m picrust_output/marker_predicted_and_nsti.tsv.gz \
#-f picrust_output/EC_predicted.tsv.gz \
#-o picrust_output/EC_metagenome_out \
#--strat_out

#metagenome_pipeline.py -i r_saved_files/picrust_16s/table.biom \
#-m picrust_output/marker_predicted_and_nsti.tsv.gz \
#-f picrust_output/KO_predicted.tsv.gz \
#-o picrust_output/KO_metagenome_out \
#--strat_out
```

EC output
76 of 4900 ASVs were above the max NSTI cut-off of 2.0 and were removed from the downstream analyses.

KO output
76 of 4900 ASVs were above the max NSTI cut-off of 2.0 and were removed from the downstream analyses.

```{bash pathway level inference}
#pathway_pipeline.py -i picrust_output/EC_metagenome_out/pred_metagenome_unstrat.tsv.gz \ 
#-o picrust_output/pathways_out \
#-p 2 \
#--intermediate picrust_output/minpath_working
```

N.B. YOU DO NOT PUT KEGG IN HERE
The default pathway and regroup mapfiles are meant for EC numbers. Note that KEGG pathways are not supported since KEGG is a closed-source database, but you can input custom pathway mapfiles if you have access. If you are using a custom function database did you mean to set the --no-regroup flag and/or change the default pathways mapfile used?

```{bash adding functional descriptions}
#add_descriptions.py -i picrust_output/EC_metagenome_out/pred_metagenome_unstrat.tsv.gz \
#-m EC \
#-o picrust_output/EC_metagenome_out/pred_metagenome_unstrat_descrip.tsv.gz

#add_descriptions.py -i picrust_output/KO_metagenome_out/pred_metagenome_unstrat.tsv.gz \
#-m KO \
#-o picrust_output/KO_metagenome_out/pred_metagenome_unstrat_descrip.tsv.gz

#add_descriptions.py -i picrust_output/pathways_out/path_abun_unstrat.tsv.gz \
#-m METACYC \
#-o picrust_output/pathways_out/path_abun_unstrat_descrip.tsv.gz
```

### Stratified PiCrust Pipeline

```{bash}
# picrust2_pipeline.py -s r_saved_files/picrust_16s/seqs.fna \
# -i r_saved_files/picrust_16s/table.biom \
# -o picrust_stratified \
# -p 2 \
# --in_traits EC,KO,PFAM,COG \
# --stratified \
# --verbose
```


## PCA Analysis

```{r CLR transformation and prep for principal component analysis, include = F}
otu_table(physeq_main_10) -> ASV_mat
cmultRepl(t(ASV_mat), 
          method = "CZM", 
          label = 0) -> ASV_mat
codaSeq.clr(ASV_mat, 
            samples.by.row = T) -> ASV_mat_CLR
t(ASV_mat_CLR) -> ASV_mat_CLR_t

# rv <- rowVars(CLR_allsamps)
# select <- order(rv,
#                 decreasing = TRUE)[seq_len(min(500,
#                                                length(rv)))]
# length(select)

allsamps <- pca(ASV_mat_CLR_t, 
                metadata = tfall_coral, 
                removeVar = 0.1)
```

```{r Plots from PCAtools, echo = F, fig.width=7, fig.height=4}
## Plot showing the genes which drive the principal component variances. 
plotloadings(allsamps, labSize = 3)
```

```{r Plots from PCAtools, fig.width=6, fig.height=3}
## Scree plot showing amount of variance explained by each PC (bars) and cumulative variance as you progress along bars (line)
screeplot(allsamps, 
          getComponents(allsamps, 1:15), 
          axisLabSize = 10, 
          titleLabSize = 10, 
          returnPlot = T, 
          ylim = c(0,55), 
          vline = c(findElbowPoint(allsamps$variance))) +
  geom_label(aes(x = findElbowPoint(allsamps$variance) + 1, y = 25,
      label = 'Elbow method', vjust = -1, size = 4))
```

```{r EigenPlots, fig.width=10, fig.height = 3}
# # Plotting of the metadata to the PC axes to see which one has strong significant relationships with axes.
# eigencorplot(
#   allsamps,
#   metavars = c("Dtreatment", "timepoint", "Genotype", "basic_health",
#                "geno_health"),
#   components = getComponents(c, 1:10),
#   col = c('darkblue', 'blue2', 'black', 'red2', 'darkred'),
#   cexCorval = 0.7,
#   colCorval = 'white',
#   fontCorval = 2,
#   posLab = 'bottomleft',
#   rotLabX = 45,
#   posColKey = 'top',
#   cexLabColKey = 1.5,
#   scale = TRUE,
#   main = 'PC1 - 11, Metadata Correlations',
#   colFrame = 'white',
#   plotRsquared = FALSE
# )

eigencorplot(
  allsamps,
  metavars = c("Dtreatment", "timepoint", "Genotype", "basic_health", 
               "geno_health"),
  components = getComponents(allsamps, 1:11),
  col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
  cexCorval = 1.2,
  fontCorval = 2,
  posLab = 'all',
  rotLabX = 45,
  scale = TRUE,
  main = bquote(
    Principal ~ Component ~ Pearson ~ r ^ 2 ~ metadata ~ significant ~ correlation
  ),
  plotRsquared = T,
  corFUN = 'pearson',
  corUSE = 'pairwise.complete.obs',
  corMultipleTestCorrection = 'BH',
  signifSymbols = c('****', '***', '**', '*', ''),
  signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
)
```

```{r Nice correlation matrix plot including genet variance, fig.width=13, fig.height = 4}
eigencorplot(allsamps,
             components = getComponents(allsamps, 1:11),
    metavars = c("basic_health", "correct_treat", "Genotype", "Tank_treatment"), 
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 1.2,
    fontCorval = 2,
    posLab = 'all',
    rotLabX = 45,
    scale = TRUE,
    main = bquote(Principal~Component~Pearson~r^2~metadata~significant~correlation),
    plotRsquared = T,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'BH',
    signifSymbols = c('****', '***', '**', '*', ''),
    signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
    )
```

```{r Genet correlation plot, fig.width=13, fig.height = 4}
eigencorplot(allsamps,
             components = getComponents(allsamps, 1:11),
    metavars = c("Genotype", "HI"), 
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 1.2,
    fontCorval = 2,
    posLab = 'all',
    rotLabX = 45,
    scale = T,
    main = bquote(Principal~Component~Pearson~r^2~metadata~significant~correlation),
    plotRsquared = T,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'BH',
    signifSymbols = c('****', '***', '**', '*', ''),
    signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
    )

View(allsamps$metadata)
```

```{r sample loadings for principal component analysis, echo = F}
pca_samp <- prcomp(ASV_mat_CLR)
sample_loadings <- as.data.frame(pca_samp$x)
summary(pca_samp)
fviz_eig(pca_samp, addlabels = TRUE, ylim = c(0, 30), )
```

```{r Percentage Loadings of PC Axis, include = F}
pc1 <- round(pca_samp$sdev[1]^2/sum(pca_samp$sdev^2),2) * 100
pc2 <- round(pca_samp$sdev[2]^2/sum(pca_samp$sdev^2),2) * 100
pc3 <- round(pca_samp$sdev[3]^2/sum(pca_samp$sdev^2),2) * 100
pc4 <- round(pca_samp$sdev[4]^2/sum(pca_samp$sdev^2),2) * 100
pc5 <- round(pca_samp$sdev[5]^2/sum(pca_samp$sdev^2),2) * 100
```

```{r Tank treatment PCA plot for supplemental, echo = F}
ggplot(sample_loadings, aes(PC1, PC2, color = PCA_sam$Tank_treatment, shape = PCA_sam$HI)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC1: ", pc1, "% variance", sep = ""), 
       y = paste("PC2: ", pc2, "% variance", sep = "")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  ) + 
    theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC1, PC2, group = PCA_sam$Tank_treatment), type = "norm") +
  scale_color_manual(values = c("blue3", "orangered2"), 
                     name = "Tank_treatment")
```

```{r Genet PCA plots}
ggplot(sample_loadings, aes(PC1, PC2, color = PCA_sam$Genotype, shape = PCA_sam$HI)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC1: ", pc1, "% variance", sep = ""), 
       y = paste("PC2: ", pc2, "% variance", sep = "")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  ) + 
    theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC1, PC2, group = PCA_sam$Genotype), type = "norm") +
  scale_color_manual(values = c("wheat3", "navy", "springgreen3", "grey60"), 
                     name = "Genotype")

ggplot(sample_loadings, aes(PC2, PC3, color = PCA_sam$Genotype, shape = PCA_sam$HI)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC2: ", pc2, "% variance", sep = ""), 
       y = paste("PC3: ", pc3, "% variance", sep = "")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  ) + 
    theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC2, PC3, group = PCA_sam$Genotype), type = "norm") +
  scale_color_manual(values = c("wheat3", "navy", "springgreen3", "grey60"), 
                     name = "Genotype")

ggplot(sample_loadings, aes(PC3, PC4, color = PCA_sam$Genotype, shape = PCA_sam$HI)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC3: ", pc3, "% variance", sep = ""), 
       y = paste("PC4: ", pc4, "% variance", sep = "")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  ) + 
    theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC3, PC4, group = PCA_sam$Genotype), type = "norm") +
  scale_color_manual(values = c("wheat3", "navy", "springgreen3", "grey60"), 
                     name = "Genotype")

ggplot(sample_loadings, aes(PC4, PC5, color = PCA_sam$Genotype)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC4: ", pc4, "% variance", sep = ""), 
       y = paste("PC5: ", pc5, "% variance", sep = "")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  ) + 
    theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC4, PC5, group = PCA_sam$Genotype), type = "norm") +
  scale_color_manual(values = c("wheat3", "navy", "springgreen3", "grey60"), 
                     name = "Genotype")
```

```{r PCA Visual Health Status, echo = F}
ggplot(
  sample_loadings,
  aes(
    PC1,
    PC2,
    color = PCA_sam$basic_health,
    shape = PCA_sam$sampling_timepoint
  )
) +
  geom_point(size = 2) +
  labs(
    x = paste("PC1: ", pc1, "% variance", sep = ""),
    y = paste("PC2: ", pc2, "% variance", sep = "")
  ) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  ) +
  theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC1, PC2, group = PCA_sam$basic_health), type = "norm") +
  scale_color_manual(name = "basic_health",
                     values = c("maroon", "dodgerblue3" , "darkgoldenrod2")) +
  scale_shape_manual(values = c(15, 16, 17),
                     name = "timepoint")

# ggplot(sample_loadings, aes(PC2, PC3, color = PCA_sam$Health)) + 
#   geom_point(size = 2) + 
#   labs(x = paste("PC2: ", pc2, "% variance", sep = ""), 
#        y = paste("PC3: ", pc3, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) + 
#   stat_ellipse() +
#     scale_color_manual(name = "Health",
#                      values = c("orangered3", "dodgerblue3" ,"darkgoldenrod2"))
# 
# ggplot(sample_loadings, aes(PC3, PC4, color = PCA_sam$Health)) + 
#   geom_point(size = 2) + 
#   labs(x = paste("PC3: ", pc3, "% variance", sep = ""), 
#        y = paste("PC4: ", pc4, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) + 
#   stat_ellipse() +
#     scale_color_manual(name = "Health",
#                      values = c("orangered3", "dodgerblue3" ,"darkgoldenrod2"))
# 
# ggplot(sample_loadings, aes(PC4, PC5, color = PCA_sam$Health)) + 
#   geom_point(size = 2) + 
#   labs(x = paste("PC4: ", pc4, "% variance", sep = ""), 
#        y = paste("PC5: ", pc5, "% variance", sep = "")) +
#   theme(
#     text = element_text(size = 11, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 8)
#   ) + 
#   stat_ellipse() +
#     scale_color_manual(name = "Health",
#                      values = c("orangered3", "dodgerblue3" ,"darkgoldenrod2"))
```

```{r PCA Dif Dis Treat echo = F}
ggplot(sample_loadings, aes(PC1, PC2, color = PCA_sam$Dif_dis_treat, shape = PCA_sam$HI)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC1: ", pc1, "% variance", sep = ""), 
       y = paste("PC2: ", pc2, "% variance", sep = "")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  ) + 
    theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse() +
  scale_color_manual(values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3"), 
                     name = "dtreat") + 
  scale_shape_manual(name = "time", 
                     values = c(16,17,15))

ggplot(sample_loadings, aes(PC2, PC3, color = PCA_sam$Dif_dis_treat)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC2: ", pc2, "% variance", sep = ""), 
       y = paste("PC3: ", pc3, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + stat_ellipse()

ggplot(sample_loadings, aes(PC3, PC4, color = PCA_sam$Dif_dis_treat)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC3: ", pc3, "% variance", sep = ""), 
       y = paste("PC4: ", pc4, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + stat_ellipse()

ggplot(sample_loadings, aes(PC4, PC5, color = PCA_sam$Dif_dis_treat, shape = PCA_sam$Health)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC4: ", pc4, "% variance", sep = ""), 
       y = paste("PC5: ", pc5, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + stat_ellipse()
```

When splitting by the disease treatments it is interesting. 
- Baseline by its self. 
- All the disease treatments seem to cluster together (i.e. no difs between dis slurry, healthy slurry, serratic and serratia placebo). 

```{r PCA Tank Treatment}
ggplot(sample_loadings, aes(PC1, PC2, color = PCA_sam$Genotype, shape = PCA_sam$Health)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC1: ", pc1, "% variance", sep = ""), 
       y = paste("PC2: ", pc2, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + 
  stat_ellipse(aes(PC1, PC2, group = PCA_sam$Genotype), type = "norm")

ggplot(sample_loadings, aes(PC2, PC3, color = PCA_sam$Genotype)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC2: ", pc2, "% variance", sep = ""), 
       y = paste("PC3: ", pc3, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + 
  stat_ellipse()

ggplot(sample_loadings, aes(PC3, PC4, color = PCA_sam$Genotype)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC3: ", pc3, "% variance", sep = ""), 
       y = paste("PC4: ", pc4, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + 
  stat_ellipse()

ggplot(sample_loadings, aes(PC4, PC5, color = PCA_sam$Genotype)) + 
  geom_point(size = 2) + 
  labs(x = paste("PC4: ", pc4, "% variance", sep = ""), 
       y = paste("PC5: ", pc5, "% variance", sep = "")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  ) + 
  stat_ellipse()
```

Differential Abundance Analysis

## Differential Abundance Analysis
### ANCOM-BC for Genotype

```{r ancombc for genet, echo = F}
genet_ancom = ancombc(
  phyloseq = physeq_main_10, 
  formula = "Genotype", 
  p_adj_method = "holm",
  lib_cut = 0,
  group = "Genotype",
  struc_zero = F,
  neg_lb = F,
  tol = 1e-5,
  max_iter = 100,
  conserve = TRUE,
  alpha = 0.01,
  global = TRUE
)
```

```{r Getting results from genet differential abundance analysis, echo = F}
genet_ancom$res_global %>% 
  as.data.frame() %>% 
  mutate(method = "ancombc") %>% 
  rownames_to_column(var = "ASV")  %>% 
  inner_join(edited_taxo %>% 
               rownames_to_column(var = "ASV"), 
             by = "ASV") %>% 
  column_to_rownames(var = "ASV") -> genet_DA_res

genet_DA_res %>% 
  dplyr::filter(diff_abn %in% "TRUE") -> genet_ancom_sig_da
# View(genet_ancom_sig_da)

# genet_ancom_sig_da %>% 
#   rownames_to_column(var = "ASV_number") %>%
#     inner_join(DHE_sequence) %>% 
#   write.csv(., 
#             file = "~/Dropbox/PhD/Projects/DHE/NGS/16s/dissertation_files/raw_lists/GENET_DA_RES.csv")
```

```{r ANCOM-BC genet singificant genes for heatmap, include = F}
matmatnorder <- ASV_mat_CLR_t[rownames(genet_ancom_sig_da),]

# Colours for the tank treatments
ccann <- data.frame(tfall_coral$Genotype)
colnames(ccann) <- c("Genotype")
colcol <- list("Genotype" = c("ML2" = "#C77CFF", "CN2" = "#F8766D", "HS1" = "#00BFC4", "CN4" = "#7CAE00"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(1, "cm"),
                             annotation_name_gp = gpar(fontsize = 40),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
edited_taxo %>%
  rownames_to_column(var = "ASV") %>% 
  dplyr::select(-species) %>%
  right_join(matmatnorder %>%
               as.data.frame() %>% 
               rownames_to_column(var="ASV"),
             by = "ASV") %>% 
  column_to_rownames(var="ASV") -> matmatnorder

# Changing names to a character
matmatnorder$genus <- as.character(matmatnorder$genus)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(genus)))
```

```{r ANCOM-BC genet significant ASVs heatmap, echo = F, fig.height=15, fig.width=15}
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        1:6
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 4,
    column_gap = unit(0.5, "cm"),
    row_km = 2,
    row_gap = unit(0.5, "cm"),
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$genus,  gp = gpar(fontsize = 20, font = 10))), 
    show_row_names = F, 
    show_column_names = F,
    column_dend_height = unit(5, "cm"))
```


### ANCOM-BC for Pre-exposure versus visually healthy and pre exposure versus diseased

```{r making subset phyloseq objects for health status, include = F}
tfall_coral %>% 
  dplyr::filter(HI %in% c("visually_healthy", "pre_exposure")) %>%
  rownames() -> cvh_prune

tfall_coral %>% 
  dplyr::filter(HI %in% c("diseased", "pre_exposure")) %>%
  rownames() -> cd_prune

# tfall_coral %>% 
#   dplyr::filter(HI %in% c("visually_healthy", "diseased")) %>%
#   rownames() -> vhd_prune

# prune_samples(vhd_prune, physeq_main_5) -> physeq_main_5_vhd
prune_samples(cvh_prune, physeq_main_10) -> physeq_main_10_cvh
prune_samples(cd_prune, physeq_main_10) -> physeq_main_10_cd
```

```{r Checking physeq objects for splits for ANCOM-bc and health status, echo = F}
physeq_main_10_cvh
physeq_main_10_cd
```

```{r ANCOM-BC health differential abundance tests, include = F}
cvh_ancom <- ancombc(
  phyloseq = physeq_main_10_cvh, 
  formula = "HI", 
  p_adj_method = "holm",
  lib_cut = 0,
  group = "HI",
  struc_zero = F,
  neg_lb = F,
  tol = 1e-5,
  max_iter = 100,
  conserve = TRUE,
  alpha = 0.01,
  global = TRUE
)

cd_ancom <- ancombc(
  phyloseq = physeq_main_10_cd, 
  formula = "HI", 
  p_adj_method = "holm",
  lib_cut = 0,
  group = "HI",
  struc_zero = F,
  neg_lb = F,
  tol = 1e-5,
  max_iter = 100,
  conserve = TRUE,
  alpha = 0.01,
  global = TRUE
)
```

```{r Subsetting results to dataframes, include = F}
as.data.frame(cvh_ancom$res) %>% 
  dplyr::rename(., Beta = 1) %>% 
  dplyr::rename(., SE = 2) %>%
  dplyr::rename(., W = 3) %>%
  dplyr::rename(., p_val = 4) %>%
  dplyr::rename(., padj = 5) %>%
  dplyr::rename(., diff_abn = 6) %>%
  mutate(method = "ancombc") %>% 
  rownames_to_column(var = "ASV")  %>% 
  inner_join(edited_taxo %>% 
               rownames_to_column(var = "ASV"), 
             by = "ASV") %>% 
  column_to_rownames(var = "ASV") -> cvh_res

as.data.frame(cd_ancom$res) %>% 
  dplyr::rename(., Beta = 1) %>% 
  dplyr::rename(., SE = 2) %>%
  dplyr::rename(., W = 3) %>%
  dplyr::rename(., p_val = 4) %>%
  dplyr::rename(., padj = 5) %>%
  dplyr::rename(., diff_abn = 6) %>%
  mutate(method = "ancombc") %>% 
  rownames_to_column(var = "ASV")  %>% 
  inner_join(edited_taxo %>% 
               rownames_to_column(var = "ASV"), 
             by = "ASV") %>% 
  column_to_rownames(var = "ASV") -> cd_res

cvh_res %>% 
  dplyr::filter(diff_abn %in% "TRUE") -> cvh_res_true

cd_res %>% 
  dplyr::filter(diff_abn %in% "TRUE") -> cd_res_true

# View(cvh_res_true)
# View(cd_res_true)
```

```{r Checking LFC greater and less than 1 for DAA results for health, echo = F}
cvh_res_true %>% nrow()
cvh_res_true %>% 
  dplyr::filter(Beta >= 1) %>% nrow()
cvh_res_true %>% 
  dplyr::filter(Beta <= -1) %>% nrow()

cd_res_true %>% nrow()
cd_res_true %>% 
  dplyr::filter(Beta >= 1) %>% nrow()
cd_res_true %>% 
  dplyr::filter(Beta <= -1) %>% nrow()
```

```{r writing DAA results for health status, include = F}
cvh_res_true %>%
  dplyr::filter(Beta >= 1 | Beta <= -1) -> sch_Res_true_lfc1
cd_res_true %>%
  dplyr::filter(Beta >= 1 | Beta <= -1) -> cd_Res_true_lfc1

# write.csv(cvh_res_true %>% 
#             dplyr::filter(Beta >=1 | Beta <=-1), 
#           file = "~/Dropbox/PhD/Projects/DHE/NGS/16s/dissertation_files/raw_lists/cvh_daa.lfc1.csv")
# write.csv(cd_res_true %>% 
#             dplyr::filter(Beta >=1 | Beta <=-1), 
#           file = "~/Dropbox/PhD/Projects/DHE/NGS/16s/dissertation_files/raw_lists/cd_daa.lfc1.csv")
```

```{r venn diagram of intersect of DAA for visual health, echo = F}
venn(list(rownames(sch_Res_true_lfc1), rownames(cd_Res_true_lfc1)))
```

```{r making common ASV dataframe for visualisation, include = F}
sch_Res_true_lfc1 %>%
  rownames_to_column(var = "ASV_number") %>% 
  rbind(cd_Res_true_lfc1 %>% 
          rownames_to_column(var = "ASV_number")) %>% 
  group_by(ASV_number, .drop = F) %>% 
  dplyr::summarise(sum = n()) %>% 
  dplyr::filter(sum %in% c("2")) %>% 
  column_to_rownames(var = "ASV_number") %>% 
  rownames() -> disease_common_filt

sch_Res_true_lfc1 %>% 
  mutate(ANCOM_contrast = "c_vs_vh") %>%
  rownames_to_column(var = "ASV_number") %>% 
  rbind(cd_Res_true_lfc1 %>% 
          mutate(ANCOM_contrast = "c_vs_d") %>%
          rownames_to_column(var = "ASV_number")) %>% 
  dplyr::filter(ASV_number %in% disease_common_filt) %>% 
  inner_join(DHE_sequence) -> common_DA_taxa_d_vs

# write.csv(common_DA_taxa_d_vs,
#           file = "~/Dropbox/PhD/Projects/DHE/NGS/16s/dissertation_files/raw_lists/common_da_D_VH.csv")
```


### Heatmaps from DAA ANlaysis for Health
#### Heatmaps C vs VH

```{r Heatmap prep for pre exposure and visually healthy, include = F}
tfall_coral %>% 
  dplyr::filter(!HI %in% c("diseased")) %>%
  rownames() -> hm_filter

hm_filter %>% length

ASV_mat_CLR_t[rownames(sch_Res_true_lfc1),] %>% 
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "sample") %>% 
  dplyr::filter(sample %in% hm_filter) %>% 
  column_to_rownames(var = "sample") %>% 
  t() %>% 
  as.matrix() -> matmatnorder

# Colours for the tank treatments
data.frame(tfall_coral$HI) %>% 
  dplyr::filter(!tfall_coral.HI %in% c("diseased")) -> ccann
colnames(ccann) <- c("HI")
colcol <- list("HI" = c("pre_exposure"="dodgerblue3", "visually_healthy"="darkgoldenrod2"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(0.6, "cm"),
                             annotation_name_gp = gpar(fontsize = 10),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
edited_taxo %>%
  rownames_to_column(var = "ASV") %>% 
  dplyr::select(-species) %>%
  right_join(matmatnorder %>%
               as.data.frame() %>% 
               rownames_to_column(var="ASV"),
             by = "ASV") %>% 
  column_to_rownames(var="ASV") -> matmatnorder
#View

# Changing names to a character
matmatnorder$genus <- as.character(matmatnorder$genus)
matmatnorder$family <- as.character(matmatnorder$family)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(family)))
```

```{r Heatmap for LFC1-1 DAA pre exposure and visually healthy, fig.height=15, fig.width=8, echo = F}
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        1:6
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 2,
    column_gap = unit(0.3, "cm"),
    row_km = 3,
    row_gap = unit(0.3, "cm"),
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$family,  gp = gpar(fontsize = 5, font = 10))), 
    show_row_names = F, 
    show_column_names = F,
    column_dend_height = unit(3, "cm"))
```


#### DAA LFC1-1 Preexposure versus diseased heatmap

```{r DAA pre exposure versus diseased heatmap prep, include = F}
tfall_coral %>% 
  dplyr::filter(!HI %in% c("visually_healthy")) %>%
  rownames() -> hm_filter

hm_filter %>% length

ASV_mat_CLR_t[rownames(cd_Res_true_lfc1),] %>% 
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "sample") %>% 
  dplyr::filter(sample %in% hm_filter) %>% 
  column_to_rownames(var = "sample") %>% 
  t() %>% 
  as.matrix() -> matmatnorder

# Colours for the tank treatments
data.frame(tfall_coral$HI) %>% 
  dplyr::filter(!tfall_coral.HI %in% c("visually_healthy")) -> ccann
colnames(ccann) <- c("HI")
colcol <- list("HI" = c("pre_exposure"="dodgerblue3", "diseased"="orangered3"))


# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(0.6, "cm"),
                             annotation_name_gp = gpar(fontsize = 10),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
edited_taxo %>%
  rownames_to_column(var = "ASV") %>% 
  dplyr::select(-species) %>%
  right_join(matmatnorder %>%
               as.data.frame() %>% 
               rownames_to_column(var="ASV"),
             by = "ASV") %>% 
  column_to_rownames(var="ASV") -> matmatnorder

# Changing names to a character
matmatnorder$genus <- as.character(matmatnorder$genus)
matmatnorder$family <- as.character(matmatnorder$family)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(family)))
```

```{r Heatmap for DAA results pre exposure versus diseased, fig.height=15, fig.width=8}
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        1:6
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 2,
    column_gap = unit(0.3, "cm"),
    row_km = 2,
    row_gap = unit(0.3, "cm"),
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$family,  gp = gpar(fontsize = 5, font = 10))), 
    show_row_names = F, 
    show_column_names = F,
    column_dend_height = unit(3, "cm"))
```


#### Heatmaps Intersect DAA

```{r Heatmap prep for common ASVs between DAA analsyis, include = F}
cd_res_true %>% 
  rownames_to_column(var = "ASVs") %>%
  inner_join(cvh_res_true %>% 
               rownames_to_column(var = "ASVs"), 
             by = "ASVs") %>% 
  column_to_rownames(var = "ASVs") -> ASV_DAA_intersect

matmatnorder <- ASV_mat_CLR_t[rownames(ASV_DAA_intersect),]

# Colours for the tank treatments
ccann <- data.frame(tfall_coral$HI)
colnames(ccann) <- c("HI")
colcol <- list("HI" = c("pre_exposure"="dodgerblue3", "visually_healthy"="darkgoldenrod2", "diseased" = "orangered3"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(1, "cm"),
                             annotation_name_gp = gpar(fontsize = 40),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
edited_taxo %>%
  rownames_to_column(var = "ASV") %>% 
  dplyr::select(-species) %>%
  right_join(matmatnorder %>%
               as.data.frame() %>% 
               rownames_to_column(var="ASV"),
             by = "ASV") %>% 
  column_to_rownames(var="ASV") -> matmatnorder

# Changing names to a character
matmatnorder$genus <- as.character(matmatnorder$genus)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(genus)))
```

```{r Heatmap of common ASVs between DAA analsyis, fig.height=15, fig.width=15}
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        1:6
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 2,
    column_gap = unit(0.5, "cm"),
    row_km = 2,
    row_gap = unit(0.5, "cm"),
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$genus,  gp = gpar(fontsize = 5, font = 10))), 
    show_row_names = F, 
    show_column_names = F,
    column_dend_height = unit(5, "cm"))
```


## Differnetial Abundance Analysis using Global for HEalth and DEGPatterns

I have decoded to do this so to generate the clusters of ASVs that can be used as input for WGCNA in the gene expression analysis so to try and integrate the analyses together rather than having a 16s analysis and a gene expression analysis. 

```{r ancombc for genet, echo = F}
health_global_ancomb = ancombc(
  phyloseq = physeq_main_10,
  formula = "basic_health",
  p_adj_method = "holm",
  lib_cut = 0,
  group = "basic_health",
  struc_zero = F,
  neg_lb = F,
  tol = 1e-5,
  max_iter = 100,
  conserve = TRUE,
  alpha = 0.05,
  global = TRUE
)
```

```{r Getting results from genet differential abundance analysis, echo = F}
health_global_ancomb$res_global %>% 
  as.data.frame() %>% 
  mutate(method = "ancombc") %>% 
  rownames_to_column(var = "ASV")  %>% 
  inner_join(edited_taxo %>% 
               rownames_to_column(var = "ASV"), 
             by = "ASV") %>% 
  column_to_rownames(var = "ASV") -> health_global_DA_res

health_global_DA_res %>% 
  dplyr::filter(diff_abn %in% "TRUE") -> health_global_sig_da
View(health_global_sig_da)

# health_global_sig_da %>% 
#   rownames_to_column(var = "ASV_number") %>%
#     inner_join(DHE_sequence) %>% 
#   write.csv(., 
#             file = "~/Dropbox/PhD/Projects/DHE/NGS/16s/dissertation_files/raw_lists/GENET_DA_RES.csv")
```


### Heatmap to Have a look at clustering 

```{r ANCOM-BC genet singificant genes for heatmap, include = F}
matmatnorder <- ASV_mat_CLR_t[rownames(health_global_sig_da),]

# Colours for the tank treatments
ccann <- data.frame(tfall_coral$HI)
colnames(ccann) <- c("HI")
colcol <- list("HI" = c("pre_exposure"="dodgerblue3", "visually_healthy"="darkgoldenrod2", "diseased" = "orangered3"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(1, "cm"),
                             annotation_name_gp = gpar(fontsize = 40),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
edited_taxo %>%
  rownames_to_column(var = "ASV") %>% 
  dplyr::select(-species) %>%
  right_join(matmatnorder %>%
               as.data.frame() %>% 
               rownames_to_column(var="ASV"),
             by = "ASV") %>% 
  column_to_rownames(var="ASV") -> matmatnorder

# Changing names to a character
matmatnorder$genus <- as.character(matmatnorder$genus)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(genus)))
```

```{r ANCOM-BC genet significant ASVs heatmap, echo = F, fig.height=15, fig.width=15}
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        1:6
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 3,
    column_gap = unit(0.5, "cm"),
    row_km = 6,
    row_gap = unit(0.5, "cm"),
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$genus,  gp = gpar(fontsize = 20, font = 10))), 
    show_row_names = F, 
    show_column_names = F,
    column_dend_height = unit(5, "cm"))
```


### DEGpatterns of ASVs from global ANCOMBC Analysis

```{r saving DEG results, fig.height=5, fig.width=7}
ASV_mat_CLR_t %>%
 as.data.frame() %>%
 rownames_to_column(var = "ASVs") %>%
 dplyr::filter(ASVs %in% rownames(health_global_sig_da)) %>%
 column_to_rownames(var = "ASVs") -> ma
# View(ma)

tfall_coral %>% 
  mutate(samples = row.names(tfall_coral), 
         samples = as.factor(samples)) -> tfall_coral

res_norm_scale <- degPatterns(ma, tfall_coral, time = "basic_health", reduce = T, scale = T)
degPlotCluster(res_norm_scale$normalized, time = "basic_health", points = F) + theme_bw()
```

```{r LRT Cluster for basic health ggplot, echo = F, fig.width=11, fig.height=4}
res_norm_scale$normalized %>%
  as.data.frame() %>%
  mutate(
    basic_health = fct_relevel(basic_health,
                               "pre_exposure",
                               "visually_healthy",
                               "diseased"),
    cluster_pretty = case_when(
      cluster == "1" ~ "Cluster 1",
      cluster == "2" ~ "Cluster 2",
      cluster == "3" ~ "Cluster 3",
      cluster == "4" ~ "Cluster 4"
    )
  ) -> res_norm_scale_ggplot

ggplot(res_norm_scale_ggplot, 
       aes(x=basic_health, y=value, color = basic_health)) +
  geom_boxplot(outlier.size = 0, outlier.shape = NA) +
  stat_smooth(aes(x = basic_health, y = value, 
            group = basic_health, color = basic_health), se = FALSE, method = "lm", 
            formula = y ~ poly(x, splan)) + 
  geom_line(data = res_norm_scale_ggplot %>% 
                group_by(basic_health, cluster_pretty) %>% 
                summarise(average = mean(value)) %>% 
                ungroup(),
            mapping = aes(x=basic_health, y = average, colour = basic_health), 
            color = "black") + 
  scale_color_manual(name = "basic_health",
                     values = c("dodgerblue3", "darkgoldenrod2", "orangered3")) +
  geom_line(data = res_norm_scale_ggplot %>% 
                group_by(basic_health, cluster_pretty) %>% 
                summarise(average = mean(value)) %>% 
                ungroup(),
            mapping = aes(x=basic_health, y = average, group = 0), 
            color = "grey9") +
  facet_wrap(~cluster_pretty, 
             labeller = "label_value", 
             ncol = 2) +
  theme_bw() 
```

```{r saving degpatterns for inclusion in gene expression WGCNA analysis, include = F}
# res_norm_scale$df ## ASVs to clusters
res_norm_scale$normalized %>% View()
res_norm_scale$summarise %>% View()
res_norm_scale$raw %>% View()
str(tfall_coral)
```

```{r Saving modules from LRT and DEGreport, include = F}
# res_norm_scale$df %>%
#   filter(cluster %in% c("1")) %>%
#   write.csv(., file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/LRT_clus1.csv")
# 
# res_norm_scale$df %>%
#   filter(cluster %in% c("2")) %>%
#   write.csv(., file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/LRT_clus2.csv")
# 
# res_norm_scale$df %>%
#   filter(cluster %in% c("3")) %>%
#   write.csv(., file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/LRT_clus3.csv")
# 
# res_norm_scale$df %>%
#   filter(cluster %in% c("4")) %>% 
#   left_join(edited_taxo %>% 
#                rownames_to_column(var = "genes")) %>%
#   write.csv(., file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/LRT_clus4.csv")
```


## Alpha Anlaysis
### Relative Abundance Analysis
#### Genotype 

```{r Rel Abun Long Format Data Frame, include = F}
physeq_main_10 %>%
  tax_glom(taxrank = "genus") %>% # agglomerate at family level
  transform_sample_counts(function(x) {
    x / sum(x)
  }) %>% # Transform to rel. abundance
  psmelt() %>%  # Melt to long format
  arrange(genus) -> gen_abundance
head(gen_abundance)
```

```{r Filtering at .01, include = F}
gen_abundance %>%
  dplyr::select(genus, Genotype, Abundance) %>%
  dplyr::group_by(genus, Genotype) %>%
  dplyr::summarize(avg_abundance = mean(Abundance)) %>%
  dplyr::filter(avg_abundance > 0.02) -> filt_gen_abundance
# View(filt_gen_abundance)
```

```{r Creating Plot, echo = F, fig.height=9, fig.width=8}
ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = Genotype, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE,
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  ) +
  scale_fill_viridis(discrete = T,
                     option = "H")
#  scale_x_discrete(guide = guide_axis(n.dodge = 2))
```

```{r Rel Abun Long Format Data Frame, include = F}
physeq_main_10 %>%
  tax_glom(taxrank = "genus") %>% # agglomerate at family level
  transform_sample_counts(function(x) {
    x / sum(x)
  }) %>% # Transform to rel. abundance
  psmelt() %>%  # Melt to long format
  arrange(genus) -> gen_abundance

head(gen_abundance)
```

```{r Filtering at .01, include = F}
gen_abundance %>%
  dplyr::select(genus, Health, Abundance) %>%
  dplyr::group_by(genus, Health) %>%
  dplyr::summarize(avg_abundance = mean(Abundance)) %>%
  dplyr::filter(avg_abundance > 0.02) -> filt_gen_abundance
# View(filt_gen_abundance)
```

```{r Creating Plot, echo = F, fig.height=10, fig.width=8}
ggplot(filt_gen_abundance %>%
         mutate(Health = fct_relevel(Health, "pre_exposure", "visually_healthy", "diseased"))) +
  geom_col(
    mapping = aes(x = Health, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE,
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  ) +
  scale_fill_viridis(discrete = T,
                     option = "H")
#  scale_x_discrete(guide = guide_axis(n.dodge = 2))
```


#### Visual Health Status

```{r Rel Abun Long Format Data Frame, include = F}
physeq_main_10 %>%
  tax_glom(taxrank = "genus") %>% # agglomerate at family level
  transform_sample_counts(function(x) {
    x / sum(x)
  }) %>% # Transform to rel. abundance
  psmelt() %>%  # Melt to long format
  arrange(genus) -> gen_abundance
```

```{r Filtering at .01, include = F}
gen_abundance %>%
  dplyr::select(genus, basic_health, Abundance) %>%
  dplyr::group_by(genus, basic_health) %>%
  dplyr::summarize(avg_abundance = mean(Abundance)) %>%
  dplyr::filter(avg_abundance > 0.015) %>% 
  dplyr::mutate(percentage_avg_abun = avg_abundance*100, 
         basic_health = fct_relevel(basic_health, c("pre_exposure", "visually_healthy", "diseased"))) -> filt_gen_abundance
View(filt_gen_abundance)
```

```{r Creating Plot, echo = F, fig.height=9, fig.width=8}
ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = basic_health, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE,
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  ) +
  scale_fill_manual(values = c("darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen", "deeppink", "khaki2", "firebrick", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey"))
#  scale_x_discrete(guide = guide_axis(n.dodge = 2))
```

```{r}
# write.csv(filt_gen_abundance, 
#           file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/coral_vhs_RA.csv")
```

#### Visual Health Status by disease inoculation

```{r Rel Abun Long Format Data Frame, include = F}
physeq_main_10 %>%
  tax_glom(taxrank = "genus") %>% # agglomerate at family level
  transform_sample_counts(function(x) {
    x / sum(x)
  }) %>% # Transform to rel. abundance
  psmelt() %>%  # Melt to long format
  arrange(genus) -> gen_abundance
```

```{r Filtering at .01, include = F}
gen_abundance %>%
  dplyr::select(genus, Health_disease, Abundance) %>%
  dplyr::group_by(genus, Health_disease) %>%
  dplyr::summarize(avg_abundance = mean(Abundance)) %>%
  dplyr::filter(avg_abundance > 0.015) %>% 
  ungroup() %>%
  dplyr::mutate(percentage_avg_abun = avg_abundance*100, 
         Health_disease = fct_relevel(Health_disease, c("pre_exposurebaseline", 
                                                            "visually_healthyHealthy_slurry", "visually_healthyDisease_slurry", 
                                                            "visually_healthySM_placebo", "visually_healthySerratia", 
                                                            "diseasedDisease_slurry", "diseasedSerratia"))) -> filt_gen_abundance
View(filt_gen_abundance)
```

```{r Creating Plot, echo = F, fig.height=10, fig.width=8}
ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = Health_disease, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE,
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 8),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10)
  ) +
  scale_fill_manual(values = distinctColorPalette(45)) + 
  scale_x_discrete(labels = c("PE", "HTS", "WBTi DS", "SP", "SM", "WBTi DS", "SM"))
  
#  scale_x_discrete(guide = guide_axis(n.dodge = 2))
```

```{r Rel Abun Long Format Data Frame, include = F}
physeq_main_10 %>%
  tax_glom(taxrank = "genus") %>% # agglomerate at family level
  transform_sample_counts(function(x) {
    x / sum(x)
  }) %>% # Transform to rel. abundance
  psmelt() %>%  # Melt to long format
  arrange(genus) -> gen_abundance
```

```{r Filtering at .01, include = F}
gen_abundance %>%
  dplyr::select(genus, Health_disease, Genotype, Abundance) %>%
  dplyr::group_by(genus, Health_disease, Genotype) %>%
  dplyr::summarize(avg_abundance = mean(Abundance)) %>%
  dplyr::filter(avg_abundance > 0.025) %>% 
  ungroup() %>%
  dplyr::mutate(percentage_avg_abun = avg_abundance*100, 
         Health_disease = fct_relevel(Health_disease, c("pre_exposurebaseline", 
                                                            "visually_healthyHealthy_slurry", "visually_healthyDisease_slurry", 
                                                            "visually_healthySM_placebo", "visually_healthySerratia", 
                                                            "diseasedDisease_slurry", "diseasedSerratia"))) -> filt_gen_abundance
View(filt_gen_abundance)
```

```{r, fig.width = 15, fig.height = 15}
ggplot(filt_gen_abundance) +
  geom_col(
    mapping = aes(x = Health_disease, y = avg_abundance, fill = genus),
    position = "fill",
    show.legend = TRUE,
    color = "black"
  ) +
  ylab("Proportion of Community") +
  xlab(NULL) +
  theme_minimal() +
  theme(
    axis.text.y.left = element_text(size = 10),
    axis.text.x = element_text(size = 10),
    axis.title.y = element_text(size = 10),
    title = element_text(size = 10), 
    legend.key.size = 
  ) +
  scale_fill_manual(values = distinctColorPalette(45)) +
  scale_x_discrete(guide = guide_axis(n.dodge=3)) + 
  facet_wrap(~ Genotype, 
             ncol = 2)
```

### Shannon and Simpson Analysis
#### Visual Health Status

Counts same as last time

```{r divnet-rs prep, include = F}
otu_table(physeq_main_10) %>%
  as.data.frame() %>%
  rownames_to_column(var = "taxa") %>%
  write.csv(., "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_vhs_counts_4_div.csv",
            row.names = F)
```

```{r Sample data for divnet, include = F}
all(rownames(tfall_coral) == colnames(otu_table(physeq_main_10)))

## this divent model is looking at Health, as PCA does not really show genet difs its fine.
model.matrix(~ HI -1, tfall_coral) %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample") %>%
  write.csv(., file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_vhs_treat_4_div.csv",
            row.names = F)

all(rownames(model.matrix(~ HI -1, tfall_coral) %>% as.data.frame()) == rownames(tfall_coral))
```

```{bash divnet in rust, include = F}
# divnet-rs /Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_divnet_vhs.toml
```

```{r Reading in Divnet RS, include = F}
divnet_rs <- read.table("/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_vhs_divout.csv",
                        sep = ",",
                        header = TRUE)

nreplicates <- 5
# Replicate 0 is actually the estimates for the real data.
rep0 <- divnet_rs[divnet_rs$replicate == 0, -1]
rownames(rep0) <- rep0$sample
rep0$sample <- NULL
#View(divnet_rs)
```

```{r Making the Shannon and Simpson results for actual Data, include =F}
rep0_shannon <- apply(rep0, 1, DivNet::shannon_true)
rep0_simpson <- apply(rep0, 1, DivNet::simpson_true)
```

```{r Shannon and Simspon for the Replicates, include = F}
# Now calculate the shannon index for the replicates.
reps_shannon <- sapply(1:nreplicates, function (i) {
  d <- divnet_rs[divnet_rs$replicate == i, -1]
  rownames(d) <- d$sample
  d$sample <- NULL
  apply(d, 1, DivNet::shannon_true)
})

reps_simpson <- sapply(1:nreplicates, function (i) {
  d <- divnet_rs[divnet_rs$replicate == i, -1]
  rownames(d) <- d$sample
  d$sample <- NULL

  apply(d, 1, DivNet::simpson_true)
})
```

```{r Shannon and Simpson Diversity Estimates for replicates, include = F}
# What we want is the variance in the diversity estimates for the replicates.
reps_shannon_error <- t(apply(reps_shannon, 1, function (x) {
  c(var(x), sd(x))
}))
colnames(reps_shannon_error) <- c("variance", "sd")

# What we want is the variance in the diversity estimates for the replicates.
reps_simpson_error <- t(apply(reps_simpson, 1, function (x) {
  c(var(x), sd(x))
}))
colnames(reps_simpson_error) <- c("variance", "sd")
```

```{r Making nice dataframes so we can group and make nice plots, include = F}
tibble(names = names(rep0_shannon),
       shannon = rep0_shannon) %>%
  left_join(reps_shannon_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = shannon + 2 * sd,
         cilower = shannon - 2 * sd) %>%
  filter(names %in% rownames(tfall_coral)) %>%
  inner_join(tfall_coral %>%
              rownames_to_column(var = "names")) %>%
  column_to_rownames(var = "names") -> alpha_shannon_treat

tibble(names = names(rep0_simpson),
       simpson = rep0_simpson) %>%
  left_join(reps_simpson_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = simpson + 2 * sd,
         cilower = simpson - 2 * sd) %>%
  inner_join(tfall_coral %>%
              rownames_to_column(var = "names")) %>%
  column_to_rownames(var = "names") -> alpha_simpson_treat
```

```{r}
View(alpha_simpson_treat)
View(alpha_shannon_treat)
```


##### Simpson 

```{r relevel HI, include = F}
alpha_simpson_treat %>% 
  mutate(HI = factor(HI, levels = c("pre_exposure", "visually_healthy", "diseased"))) -> alpha_simpson_treat
```

```{r SIMPSON PLOTTING Dtreatment, echo = F, fig.width = 3.5, fig.height=5}
ggplot(data = alpha_simpson_treat,
       aes(x = HI, y = simpson, color = HI)) +
  geom_point(position = position_dodge(width=0.5), 
             show.legend = F) +
  geom_linerange(aes(ymin = cilower, ymax = ciupper),
                 position = position_dodge(.5), 
                 show.legend = F) +
  scale_color_manual(name = "HI",
                     values = c("dodgerblue3",  "darkgoldenrod2", "maroon")) +
  theme_bw() + 
  ylim(-0.4, 0.6) + 
  scale_x_discrete(labels = c("PE", "VH", "D")) + 
  theme(axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 12), 
        axis.text.x = element_text(size = 9)) + 
  labs(y = "Simpson")

# tibble(names = names(rep0_simpson),
#        simpson = rep0_simpson) %>%
#   left_join(reps_simpson_error %>%
#               as.data.frame %>%
#               rownames_to_column("names")) %>%
#   mutate(ciupper = simpson + 2 * sd,
#          cilower = simpson - 2 * sd) %>%
#   ggplot(aes(x = names, color = alpha_simpson_treat$HI)) +
#   geom_segment(aes(xend = names, y = cilower, yend = ciupper)) +
#   geom_point(aes(y = simpson)) +
#   xlab("samples") +
#   ylab("simpson estimate") +
#   ggtitle("divnet-rs") +
#   theme_bw() +
#   theme(panel.grid.minor.x = element_blank(),
#         axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
#   scale_color_manual(name = "HI",
#                      values = c("dodgerblue3",  "darkgoldenrod2", "orangered3")) +
#   theme_bw()
```

```{r releveling TT so ambient is base for simpson, include = F}
alpha_simpson_treat %>% 
  mutate(vh = relevel(factor(HI), ref = "visually_healthy"), 
         pe = relevel(factor(HI), ref = "pre_exposure")) -> alpha_simpson_treat
```

```{r}
pe_simpson <- betta(
  chats = alpha_simpson_treat$simpson,
  ses = sqrt(alpha_simpson_treat$sd),
  X = model.matrix( ~ pe, data = alpha_simpson_treat)
)
pe_simpson$table

vh_simpson <- betta(
  chats = alpha_simpson_treat$simpson,
  ses = sqrt(alpha_simpson_treat$sd),
  X = model.matrix( ~ vh, data = alpha_simpson_treat)
)
vh_simpson$table
```


##### Shannon 

```{r relevel HI, include = F}
alpha_shannon_treat %>% 
  mutate(HI = factor(HI, levels = c("pre_exposure", "visually_healthy", "diseased"))) -> alpha_shannon_treat
```

```{r Plots for Shannon, echo = F, fig.width=3.2, fig.height=4.8}
ggplot(data = alpha_shannon_treat,
       aes(x = HI, y = shannon, color = HI, group = HI)) +
  geom_point(position = position_dodge(width=0.5), 
             show.legend = F) +
  geom_linerange(aes(ymin = cilower, ymax = ciupper),
                 position = position_dodge(.5), 
                 show.legend = F) +
  scale_color_manual(name = "HI",
                     values = c("dodgerblue3",  "darkgoldenrod2", "maroon")) +
  theme_bw() + 
  ylim(2, 8) + 
  scale_x_discrete(labels = c("PE", "VH", "D")) +
  theme(axis.title.x = element_blank(), 
        axis.title.y = element_text(size = 12), 
        axis.text.x = element_text(size = 9)) + 
  labs(y = "Shannon")

# tibble(names = names(rep0_shannon),
#        shannon = rep0_shannon) %>%
#   left_join(reps_shannon_error %>%
#               as.data.frame %>%
#               rownames_to_column("names")) %>%
#   mutate(ciupper = shannon + 2 * sd,
#          cilower = shannon - 2 * sd) %>%
#   ggplot(aes(x = names, color = alpha_shannon_treat$HI)) +
#   geom_segment(aes(xend = names, y = cilower, yend = ciupper)) +
#   geom_point(aes(y = shannon)) +
#   xlab("samples") +
#   ylab("shannon estimate") +
#   ggtitle("divnet-rs") +
#   theme_bw() +
#   theme(panel.grid.minor.x = element_blank(),
#         axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
#   scale_color_manual(name = "HI",
#                      values = c("dodgerblue3",  "darkgoldenrod2", "orangered3")) +
#   theme_bw()
```

```{r releveling TT so ambient is base for simpson, include = F}
alpha_shannon_treat %>% 
  mutate(vh = relevel(factor(HI), ref = "visually_healthy"), 
         pe = relevel(factor(HI), ref = "pre_exposure")) -> alpha_shannon_treat
```

```{r}
pe_shannon <- betta(
  chats = alpha_shannon_treat$shannon,
  ses = sqrt(alpha_shannon_treat$sd),
  X = model.matrix( ~ pe, data = alpha_shannon_treat)
)
pe_shannon$table

vh_shannon <- betta(
  chats = alpha_shannon_treat$shannon,
  ses = sqrt(alpha_shannon_treat$sd),
  X = model.matrix( ~ vh, data = alpha_shannon_treat)
)
vh_shannon$table
```


#### Visual Health Status split by Disease inoculations

```{r divnet-rs prep, include = F}
otu_table(physeq_main_10) %>%
  as.data.frame() %>%
  rownames_to_column(var = "taxa") %>%
  write.csv(., "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_vhs_counts_4_div.csv",
            row.names = F)
```

```{r Sample data for divnet, include = F}
all(rownames(tfall_coral) == colnames(otu_table(physeq_main_10)))

## this divent model is looking at Health, as PCA does not really show genet difs its fine.
model.matrix(~ Health_disease -1, tfall_coral) %>%
  as.data.frame() %>%
  rownames_to_column(var = "sample") %>%
  write.csv(., file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_vhs_di_treat_4_div.csv",
            row.names = F)

all(rownames(model.matrix(~ Health_disease -1, tfall_coral) %>% as.data.frame()) == rownames(tfall_coral))
```

```{bash divnet in rust, include = F}
# divnet-rs /Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_divnet_vhs_di.toml
```

```{r Reading in Divnet RS, include = F}
divnet_rs <- read.table("/Users/benjamin.d.young/Documents/projects/DHE/manuscript/divnet/DHE_vhs_di_divout.csv",
                        sep = ",",
                        header = TRUE)

nreplicates <- 5
# Replicate 0 is actually the estimates for the real data.
rep0 <- divnet_rs[divnet_rs$replicate == 0, -1]
rownames(rep0) <- rep0$sample
rep0$sample <- NULL
#View(divnet_rs)
```

```{r Making the Shannon and Simpson results for actual Data, include =F}
rep0_shannon <- apply(rep0, 1, DivNet::shannon_true)
rep0_simpson <- apply(rep0, 1, DivNet::simpson_true)
```

```{r Shannon and Simspon for the Replicates, include = F}
# Now calculate the shannon index for the replicates.
reps_shannon <- sapply(1:nreplicates, function (i) {
  d <- divnet_rs[divnet_rs$replicate == i, -1]
  rownames(d) <- d$sample
  d$sample <- NULL
  apply(d, 1, DivNet::shannon_true)
})

reps_simpson <- sapply(1:nreplicates, function (i) {
  d <- divnet_rs[divnet_rs$replicate == i, -1]
  rownames(d) <- d$sample
  d$sample <- NULL

  apply(d, 1, DivNet::simpson_true)
})
```

```{r Shannon and Simpson Diversity Estimates for replicates, include = F}
# What we want is the variance in the diversity estimates for the replicates.
reps_shannon_error <- t(apply(reps_shannon, 1, function (x) {
  c(var(x), sd(x))
}))
colnames(reps_shannon_error) <- c("variance", "sd")

# What we want is the variance in the diversity estimates for the replicates.
reps_simpson_error <- t(apply(reps_simpson, 1, function (x) {
  c(var(x), sd(x))
}))
colnames(reps_simpson_error) <- c("variance", "sd")
```

```{r Making nice dataframes so we can group and make nice plots, include = F}
tibble(names = names(rep0_shannon),
       shannon = rep0_shannon) %>%
  left_join(reps_shannon_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = shannon + 2 * sd,
         cilower = shannon - 2 * sd) %>%
  filter(names %in% rownames(tfall_coral)) %>%
  inner_join(tfall_coral %>%
              rownames_to_column(var = "names")) %>%
  column_to_rownames(var = "names") -> alpha_shannon_treat_di

tibble(names = names(rep0_simpson),
       simpson = rep0_simpson) %>%
  left_join(reps_simpson_error %>%
              as.data.frame %>%
              rownames_to_column("names")) %>%
  mutate(ciupper = simpson + 2 * sd,
         cilower = simpson - 2 * sd) %>%
  inner_join(tfall_coral %>%
              rownames_to_column(var = "names")) %>%
  column_to_rownames(var = "names") -> alpha_simpson_treat_di
```

```{r relevel shannon and simpson for plotting, include = F}
alpha_shannon_treat_di %>% 
  mutate(Health_disease = factor(Health_disease, levels = c("pre_exposurebaseline", 
                                                            "visually_healthyHealthy_slurry", "visually_healthyDisease_slurry", 
                                                            "visually_healthySM_placebo", "visually_healthySerratia", 
                                                            "diseasedDisease_slurry", "diseasedSerratia"))) -> alpha_shannon_treat_di

alpha_simpson_treat_di %>% 
  mutate(Health_disease = factor(Health_disease, levels = c("pre_exposurebaseline", 
                                                            "visually_healthyHealthy_slurry", "visually_healthyDisease_slurry", 
                                                            "visually_healthySM_placebo", "visually_healthySerratia", 
                                                            "diseasedDisease_slurry", "diseasedSerratia"))) -> alpha_simpson_treat_di
```


##### Simpson 

```{r SIMPSON PLOTTING Dtreatment, echo = F, fig.width=7, fig.height=5}
ggplot(data = alpha_simpson_treat_di,
       aes(x = Health_disease, y = simpson, color = Health_disease)) +
  geom_point(position = position_dodge(width=0.5), 
             show.legend = F) +
  geom_linerange(aes(ymin = cilower, ymax = ciupper),
                 position = position_dodge(.5), 
                 show.legend = F) +
  scale_color_manual(name = "Health_disease",
                     values = c("dodgerblue3", "aquamarine3", "gold3", "lightsteelblue3", "plum1", "aquamarine3", "plum1")) +
  theme_bw() + 
  ylim(-0.4, 0.6) + 
  scale_x_discrete(labels = c("PE", "HTS", "WBTi DS", "SP", "SM", "WBTi DS", "SM")) + 
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 9),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank()) + 
  labs(y = "Simpson")

# tibble(names = names(rep0_simpson),
#        simpson = rep0_simpson) %>%
#   left_join(reps_simpson_error %>%
#               as.data.frame %>%
#               rownames_to_column("names")) %>%
#   mutate(ciupper = simpson + 2 * sd,
#          cilower = simpson - 2 * sd) %>%
#   ggplot(aes(x = names, color = alpha_simpson_treat$Health_disease)) +
#   geom_segment(aes(xend = names, y = cilower, yend = ciupper)) +
#   geom_point(aes(y = simpson)) +
#   xlab("samples") +
#   ylab("simpson estimate") +
#   ggtitle("divnet-rs") +
#   theme_bw() +
#   theme(panel.grid.minor.x = element_blank(),
#         axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
#   scale_color_manual(name = "Health_disease",
#                      values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3", "plum1", "aquamarine3")) +
#   theme_bw()
```

```{r releveling TT so ambient is base for simpson, include = F}
alpha_simpson_treat_di %>% 
  mutate(HTS = relevel(factor(Health_disease), ref = "visually_healthyHealthy_slurry"), 
         WBTi = relevel(factor(Health_disease), ref = "visually_healthyDisease_slurry"),
         SM = relevel(factor(Health_disease), ref = "visually_healthySerratia"), 
         SMDS = relevel(factor(Health_disease), ref = "diseasedDisease_slurry"), 
         WBTID = relevel(factor(Health_disease), ref = "diseasedSerratia")) -> alpha_simpson_treat_di
```

```{r visually healthy disease inoculation betta tests, echo = F}
HTSvh_simp <- betta(
  chats = alpha_simpson_treat_di$simpson,
  ses = sqrt(alpha_simpson_treat_di$sd),
  X = model.matrix( ~ HTS, data = alpha_simpson_treat_di)
)
HTSvh_simp$table

WBTivh_simp <- betta(
  chats = alpha_simpson_treat_di$simpson,
  ses = sqrt(alpha_simpson_treat_di$sd),
  X = model.matrix( ~ WBTi, data = alpha_simpson_treat_di)
)
WBTivh_simp$table

SMvh_simp <- betta(
  chats = alpha_simpson_treat_di$simpson,
  ses = sqrt(alpha_simpson_treat_di$sd),
  X = model.matrix( ~ SM, data = alpha_simpson_treat_di)
)
SMvh_simp$table
```

```{r diseased betta disease inoculations, echo = F}
SMd_simp <- betta(
  chats = alpha_simpson_treat_di$simpson,
  ses = sqrt(alpha_simpson_treat_di$sd),
  X = model.matrix( ~ SMDS, data = alpha_simpson_treat_di)
)
SMd_simp$table

WBDd_simp <- betta(
  chats = alpha_simpson_treat_di$simpson,
  ses = sqrt(alpha_simpson_treat_di$sd),
  X = model.matrix( ~ WBTID, data = alpha_simpson_treat_di)
)
WBDd_simp$table
```


##### Shannon 

```{r Plots for Shannon, echo = F, fig.width=7.2, fig.height=4.5}
ggplot(data = alpha_shannon_treat_di,
       aes(x = Health_disease, y = shannon, color = Health_disease)) +
  geom_point(position = position_dodge(width=0.5), 
             show.legend = F) +
  geom_linerange(aes(ymin = cilower, ymax = ciupper),
                 position = position_dodge(.5), 
                 show.legend = F) +
  scale_color_manual(name = "Health_disease",
                     values = c("dodgerblue3", "aquamarine3", "gold3", "lightsteelblue3", "plum1", "aquamarine3", "plum1")) +
  theme_bw() + 
  ylim(2, 8) + 
  scale_x_discrete(labels = c("PE", "HTS", "WBTi DS", "SP", "SM", "WBTi DS", "SM")) +
  theme(axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 9),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(), 
        axis.ticks.y = element_blank()) + 
  labs(y = "Shannon")

# tibble(names = names(rep0_shannon),
#        shannon = rep0_shannon) %>%
#   left_join(reps_shannon_error %>%
#               as.data.frame %>%
#               rownames_to_column("names")) %>%
#   mutate(ciupper = shannon + 2 * sd,
#          cilower = shannon - 2 * sd) %>%
#   ggplot(aes(x = names, color = alpha_shannon_treat$Health_disease)) +
#   geom_segment(aes(xend = names, y = cilower, yend = ciupper)) +
#   geom_point(aes(y = shannon)) +
#   xlab("samples") +
#   ylab("shannon estimate") +
#   ggtitle("divnet-rs") +
#   theme_bw() +
#   theme(panel.grid.minor.x = element_blank(),
#         axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
#   scale_color_manual(name = "Health_disease",
#                      values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3", "plum1", "aquamarine3")) +
#   theme_bw()
```

```{r releveling TT so ambient is base for simpson, include = F}
alpha_shannon_treat_di %>% 
  mutate(HTS = relevel(factor(Health_disease), ref = "visually_healthyHealthy_slurry"), 
         WBTi = relevel(factor(Health_disease), ref = "visually_healthyDisease_slurry"),
         SM = relevel(factor(Health_disease), ref = "visually_healthySerratia"), 
         SMDS = relevel(factor(Health_disease), ref = "diseasedDisease_slurry"), 
         WBTID = relevel(factor(Health_disease), ref = "diseasedSerratia")) -> alpha_shannon_treat_di
```

```{r visually healthy disease inoculation betta tests, echo = F}
HTSvh_simp <- betta(
  chats = alpha_shannon_treat_di$shannon,
  ses = sqrt(alpha_shannon_treat_di$sd),
  X = model.matrix( ~ HTS, data = alpha_shannon_treat_di)
)
HTSvh_simp$table

WBTivh_simp <- betta(
  chats = alpha_shannon_treat_di$shannon,
  ses = sqrt(alpha_shannon_treat_di$sd),
  X = model.matrix( ~ WBTi, data = alpha_shannon_treat_di)
)
WBTivh_simp$table

SMvh_simp <- betta(
  chats = alpha_shannon_treat_di$shannon,
  ses = sqrt(alpha_shannon_treat_di$sd),
  X = model.matrix( ~ SM, data = alpha_shannon_treat_di)
)
SMvh_simp$table
```

```{r diseased betta disease inoculations, echo = F}
SMd_simp <- betta(
  chats = alpha_shannon_treat_di$shannon,
  ses = sqrt(alpha_shannon_treat_di$sd),
  X = model.matrix( ~ SMDS, data = alpha_shannon_treat_di)
)
SMd_simp$table

WBDd_simp <- betta(
  chats = alpha_shannon_treat_di$shannon,
  ses = sqrt(alpha_shannon_treat_di$sd),
  X = model.matrix( ~ WBTID, data = alpha_shannon_treat_di)
)
WBDd_simp$table
```


#### Combined Figure for Paper

Dataframes to be used for these combined figures

- alpha_shannon_treat
- alpha_shannon_treat_di
- alpha_simpson_treat
- alpha_simpson_treat_di

```{r}
View(alpha_shannon_treat)
View(alpha_shannon_treat_di)
View(alpha_simpson_treat)
View(alpha_simpson_treat_di)
```

##### Simpson for Figure

```{r}
alpha_simpson_treat %>% 
  dplyr::mutate(simp_fig = HI) %>% 
  dplyr::select(1:5, simp_fig) %>% 
  rownames_to_column(var = "sample") %>% 
  rbind(alpha_simpson_treat_di %>% 
          dplyr::mutate(simp_fig = Health_disease) %>%
          dplyr::select(1:5, simp_fig) %>% 
          rownames_to_column(var = "sample")) %>%
  mutate(simp_fig = factor(simp_fig, levels = c("pre_exposure", "visually_healthy", "diseased", 
                                                "pre_exposurebaseline",
                                                "visually_healthyHealthy_slurry", "visually_healthyDisease_slurry", 
                                                "visually_healthySM_placebo", "visually_healthySerratia", 
                                                "diseasedDisease_slurry", "diseasedSerratia"))) -> simp_comb
```

```{r simpson all plotted, echo = F, fig.width = 10, fig.height=5}
ggplot(data = simp_comb,
       aes(x = simp_fig, y = simpson, color = simp_fig)) +
  geom_point(position = position_dodge(width=0.5)) +
  geom_linerange(aes(ymin = cilower, ymax = ciupper),
                 position = position_dodge(.5)) +
  scale_color_manual(name = "Health_disease",
                     values = c("dodgerblue3", "darkgoldenrod2"," maroon", "dodgerblue3", "gold3", "aquamarine3", "lightsteelblue3", "plum1", "aquamarine3", "plum1")) +
  theme_bw() + 
  scale_x_discrete(labels = c("PE", "VH", "D", "PE", "HTS", "WBTi DS", "SP", "SM", "WBTi DS", "SM"))
```


## WGCNA Analysis for ASV Data COUNTS
### WGCNA Analysis with Genet Removed

So this will be an interesting attempt using WGCNA for ASV data. No batch removal needed as disease response is the largest driver of the ASV data (i.e. no genet responses). Input needs to be normalised counts so I think using the CLR transformed counts is a good option. 

Generation of the treatment file is the same as the gene expression WGCNA analysis. 

So lets do that and if a reviewer doesn't like it I will fight them. 

```{r SUPER DUPER IMPORTANT FOR WGCNA TO DO THIS, include = F}
options(stringsAsFactors = FALSE)
library(WGCNA)
```

```{r Making the all trait file for correlatvie analysis, include = F}
tfall_coral %>% 
  rownames_to_column(var = "samps") %>% 
  dplyr::select(samps, Inoculated, basic_health, Health_disease) %>%
  column_to_rownames(var = "samps") -> WGCNA_treat_cato 

model.matrix(~Inoculated-1, WGCNA_treat_cato) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Sample_numbers") %>% 
  full_join(model.matrix(~Health_disease-1, WGCNA_treat_cato) %>% 
              as.data.frame() %>% 
              rownames_to_column(var = "Sample_numbers")) %>% 
  full_join(model.matrix(~basic_health-1, WGCNA_treat_cato) %>% 
              as.data.frame() %>% 
              rownames_to_column(var = "Sample_numbers")) %>%
  dplyr::select(1, 2, 3, 12, 13, 11, 6, 8, 7, 4, 10, 9, 5) -> allTraits
# View(allTraits)
```

```{r Making WGCNA gene objects and checking everything matches up, echo = F}
genes4wgcna <- t(ASV_mat_CLR_t)
genes4wgcna <- as.data.frame(genes4wgcna)
nrow(genes4wgcna)
dim(genes4wgcna)

gsg = goodSamplesGenes(genes4wgcna, verbose = 3)
gsg$allOK
```

```{r Sample clustering to identify outliers, fig.height=20, fig.width=20, echo = F}
sampleTree = hclust(dist(genes4wgcna), method = "ward.D2");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
sizeGrWindow(12,9)
#pdf(file = "Plots/sampleClustering.pdf", width = 12, height = 9);
par(cex = 1);
par(mar = c(0,4,2,0))
plot(sampleTree, 
     main = "Sample clustering to detect outliers", 
     sub="", 
     xlab="", 
     cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
# Plot a line to show the cut
abline(h = 1200, col = "red");
# Determine cluster under the line
clust = cutreeStatic(sampleTree, cutHeight = 800, minSize = 10)
table(clust)
# clust 1 contains the samples we want to keep.
keepSamples = (clust==1)
datExpr = genes4wgcna[keepSamples, ]
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
```

There are no crazy outliers so going to include everything for the downstream WGCNA analysis. 

```{r Correlative data matching with WGCNA gene objects, echo = F}
#4. Create an object with the treatment file
#Create an object called "datTraits" that contains your trait data
head(allTraits)
str(allTraits)
#form a data frame analogous to expression data that will hold the clinical traits.
rownames(allTraits) = allTraits$Sample_numbers
allTraits$Sample_numbers = NULL
table(rownames(allTraits)==rownames(genes4wgcna)) #should return TRUE if datasets align correctly, otherwise your names are out of order
```

Everything matches which is fantastic news between WGCNA gene object and the treatment file. 

```{r Identyfying Soft Power, echo = F}
# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5, dataIsExpr = T)
# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```
Soft power identified as 7, curves look good so no need for additional data filtering. 

```{r Signed adjacency matrix using soft power from last step, include = F}
softPower = 12;
adjacency = adjacency(datExpr, type="signed", power = softPower);
```

```{r Generation of topological overlap, include = F}
# Turn adjacency into topological overlap
TOM = TOMsimilarity(adjacency);
dissTOM = 1-TOM
```

```{r Gene clustering with TOM-based dissimilarity plot, fig.width=10, fig.height=4}
# Call the hierarchical clustering function
geneTree = hclust(as.dist(dissTOM), 
                  method = "average");
# Plot the resulting clustering tree (dendrogram)
sizeGrWindow(12,9)
plot(geneTree, 
     xlab="", 
     sub="", 
     main = "Gene clustering on TOM-based dissimilarity",
labels = FALSE, hang = 0.04);
```

```{r Module identification, echo = F}
# We like large modules, so we set the minimum module size relatively high:
minModuleSize = 10;
# Module identification using dynamic tree cut:
dynamicMods = cutreeDynamic(dendro = geneTree, 
                            distM = dissTOM,
                            deepSplit = 2,
                            pamRespectsDendro = FALSE, 
                            minClusterSize = minModuleSize);
table(dynamicMods)
```

```{r Gene dendogram and module colours plot, echo = F, fig.width = 7, fig.height= 2}
# Convert numeric lables into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
# Plot the dendrogram and colors underneath
sizeGrWindow(8,6)
plotDendroAndColors(geneTree, 
                    dynamicColors, 
                    "Dynamic Tree Cut", 
                    dendroLabels = FALSE, 
                    hang = 0.03,
                    addGuide = TRUE, 
                    guideHang = 0.05, 
                    main = "Gene dendrogram and module colors")
```

```{r Merging similar modules at 0.40, fig.height=6, fig.width=10, echo = F}
# Calculate eigengenes
MEList = moduleEigengenes(datExpr, 
                          colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), 
                method = "average");
# Plot the result
sizeGrWindow(7, 6)

plot(METree, 
     main = "Clustering of module eigengenes",
     xlab = "", 
     sub = "")
MEDissThres = 0.35
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")
```

```{r Merging similar modules, include = F}
# Call an automatic merging function
merge = mergeCloseModules(datExpr, 
                          dynamicColors, 
                          cutHeight = MEDissThres, 
                          verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs;
```

```{r Cluster dendogram with original and merged modules, fig.height=3, fig.width=10, echo = F}
sizeGrWindow(12, 5)
#pdf(file = "Plots/geneDendro-3.pdf", wi = 9, he = 6)
plotDendroAndColors(geneTree, 
                    cbind(dynamicColors, 
                          mergedColors), 
                    c("Dynamic Tree Cut", "Merged dynamic"), 
                    dendroLabels = FALSE, 
                    hang = 0.03, 
                    addGuide = TRUE, 
                    guideHang = 0.05)
#dev.off()
```

```{r Eigengenes for each sample to each module, echo = F}
# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", 
               standardColors(50));
moduleLabels = match(moduleColors, 
                     colorOrder)-1
MEs = mergedMEs
# View(MEs)
```

```{r Significance of eigengenes to correlative data, inclue = F}
# Define numbers of genes and samples
nGenes = ncol(datExpr);
nSamples = nrow(datExpr);
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(datExpr, 
                        moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, 
                     allTraits, 
                     use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, 
                                     nSamples);

```

```{r}
# write.csv(MEs0, 
#           file = "~/Documents/projects/DHE/manuscript/r_saved_files/correlative_analysis/dhe_16s_eigengenes.csv")
# write.csv(moduleTraitCor, 
#           file = "~/Documents/projects/DHE/manuscript/r_saved_files/correlative_analysis/dhe_16s_moduletraitcor.csv")
```


Pipeline all run now for visualization and analysis of the modules. 

```{r summary data for outputs from WGCNA, include = F}
MEs %>% 
  rownames_to_column(var = "ASV_number") %>% 
  inner_join(tfall_coral %>% 
               rownames_to_column(var = "ASV_number"), 
             by = "ASV_number") -> MEs_and_tfall

# moduleTraitCor %>% View()
```

```{r All modules module-trait relationship plot, fig.width=8, fig.height=8}
sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(
  Matrix = moduleTraitCor,
  xLabels = names(allTraits),
  yLabels = names(MEs),
  ySymbols = names(MEs),
  colorLabels = FALSE,
  colors = blueWhiteRed(50),
  textMatrix = textMatrix,
  setStdMargins = FALSE,
  cex.text = 0.5,
  zlim = c(-1, 1),
  cex.lab = 0.5,
  main = paste("Module-trait relationships")
)
```

```{r Hub gene for each module identified form analysis, echo = F}
chooseTopHubInEachModule(genes4wgcna,
                                   moduleColors,
                                   omitColors = "grey",
                                   power = 10) %>% 
  as.data.frame() %>%
  dplyr::rename(., ASV_number = .) %>%
  rownames_to_column(var = "module") %>% 
  inner_join(edited_taxo %>% 
               rownames_to_column(var = "ASV_number"), 
             by = "ASV_number") %>% 
  inner_join(DHE_sequence)-> tophub_annotated
View(tophub_annotated)

# write.csv(tophub_annotated,
#          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/tophub_annotated_16s.csv",
#          row.names = F)
```

Omit the grey as it is the "rubbish bin" of WGCNA. 

```{r dataframes of genes within each module from WGCNA analysis, include = F}
cyan <- names(genes4wgcna)[moduleColors == "cyan"] %>%
  as.data.frame() %>%
  dplyr::rename(., ASV_number = .) %>% 
  mutate(module = "cyan") %>%
  inner_join(edited_taxo %>%
               rownames_to_column(var = "ASV_number")) %>% 
  inner_join(DHE_sequence) -> cyan_annotated

darkgreen <- names(genes4wgcna)[moduleColors == "darkgreen"] %>%
  as.data.frame() %>%
  dplyr::rename(., ASV_number = .) %>%
  mutate(module = "darkgreen") %>%
  inner_join(edited_taxo %>%
               rownames_to_column(var = "ASV_number")) %>% 
  inner_join(DHE_sequence) -> darkgreen_annotated

purple <- names(genes4wgcna)[moduleColors == "purple"] %>%
  as.data.frame() %>%
  dplyr::rename(., ASV_number = .) %>%
  mutate(module = "purple") %>%
  inner_join(edited_taxo %>%
               rownames_to_column(var = "ASV_number")) %>% 
  inner_join(DHE_sequence) -> purple_annotated

green <- names(genes4wgcna)[moduleColors == "green"] %>%
  as.data.frame() %>%
  dplyr::rename(., ASV_number = .) %>%
  mutate(module = "green") %>%
  inner_join(edited_taxo %>%
               rownames_to_column(var = "ASV_number")) %>% 
  inner_join(DHE_sequence) -> green_annotated

orange <- names(genes4wgcna)[moduleColors == "orange"] %>%
  as.data.frame() %>%
  dplyr::rename(., ASV_number = .) %>%
  mutate(module = "orange") %>%
  inner_join(edited_taxo %>%
               rownames_to_column(var = "ASV_number")) %>% 
  inner_join(DHE_sequence) -> orange_annotated

darkred <- names(genes4wgcna)[moduleColors == "darkred"] %>%
  as.data.frame() %>%
  dplyr::rename(., ASV_number = .) %>%
  mutate(module = "darkred") %>%
  inner_join(edited_taxo %>%
               rownames_to_column(var = "ASV_number")) %>% 
  inner_join(DHE_sequence) -> darkred_annotated
```

```{r all modules combined from WGCNA analysis}
darkred_annotated %>% 
  rbind(orange_annotated) %>% 
  rbind(green_annotated) %>% 
  rbind(purple_annotated) %>% 
  rbind(darkgreen_annotated) %>% 
  rbind(cyan_annotated) -> WGCNA_4_picrust
```

```{r Number of genes in each module from WGCNA analysis, echo = F}
nrow(cyan_annotated)
nrow(darkgreen_annotated)
nrow(purple_annotated)
nrow(green_annotated)
nrow(darkred_annotated)
nrow(orange_annotated)
```

```{r}
View(darkgreen_annotated)
View(darkred_annotated)
View(cyan_annotated)
View(orange_annotated)
View(green_annotated)
View(purple)
```


```{r writing annotataed module gene lists, include = F}
# write.csv(
#   cyan_annotated,
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/wgcna_asvlists/cyan_annotated_16s.csv")
# 
# write.csv(
#   darkgreen_annotated,
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/wgcna_asvlists/darkgreen_annotated_16s.csv")
# 
# write.csv(
#   purple_annotated,
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/wgcna_asvlists/purple_annotated_16s.csv")
# 
# write.csv(
#   darkred_annotated,
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/wgcna_asvlists/darkred_annotated_16s.csv")
# 
# write.csv(
#   orange_annotated,
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/wgcna_asvlists/orange_annotated_16s.csv")
# 
# write.csv(
#   green_annotated,
#   file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/wgcna_asvlists/green_annotated_16s.csv")
```


### Sexy Plot

Removing the grey module as this is the rubbish bin of WGCNA. 

```{r complex heatmap dataframes with correct columns, include = F}
textMatrix %>%
  as.data.frame() %>% 
  dplyr::slice(6, 5, 3, 1, 2, 4) %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) %>% 
  dplyr::filter(!row_number() %in% c(7))-> tmatrix_ch

moduleTraitCor %>%
  as.data.frame() %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) %>%
  dplyr::filter(!row_number() %in% c(7)) %>% 
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "remove") %>% 
  dplyr::select(-remove) %>%
  mutate(
    Row_annotations = c(
      "Pre Exposure",
      "Visually Healthy",
      "Diseased",
      "Health Slurry (VH)",
      "Disease Slurry (VH)",
      "Serratia Placebo (VH)",
      "Serratia (VH)",
      "Disease Slurry (D)",
      "Serratia (D)"
    )
  ) %>% 
  column_to_rownames(var = "Row_annotations") %>% 
  dplyr::select(MEpurple, MEgreen, MEcyan, MEdarkgreen, MEdarkred, MEorange) %>%
  t() %>%
  as.data.frame() -> modtraitcor_ch

allTraits %>% 
  as.data.frame() %>%
dplyr::select(3:12) -> alltrait_ch
```

```{r Complex heatmap prep for WGCNA analysis, include =}
ccann <- data.frame(rownames(modtraitcor_ch))
colnames(ccann) <- c("Modules")
colcol <- list("Modules" = c("MEpurple" = "purple",
                             "MEgreen" = "Green",
                             "MEcyan" = "cyan",
                             "MEdarkgreen" = "Dark Green", 
                             "MEdarkred" = "darkred", 
                             "MEorange" = "orange"
                             ))

samname <- HeatmapAnnotation(
  df = ccann,
  col = colcol,
  which = "row",
  simple_anno_size = unit(0.5, "cm"),
  annotation_name_gp = gpar(fontsize = 10), 
  show_annotation_name = F)

col_fun = colorRamp2(c(-1, 0, 1), c("royalblue", "white", "red"))
```

```{r WGCNA complex heatmap, echo = F, fig.width = 9, fig.height=6}
Heatmap(
  modtraitcor_ch %>% as.matrix(),
  cluster_rows = F,
  cluster_columns = F,
  left_annotation = samname, 
  row_names_side = "left",
  column_names_rot = 45,
  col = col_fun,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(sprintf(tmatrix_ch[i, j]), x, y, gp = gpar(fontsize = 10))
  },
  column_split = data.frame(rep(c("a", "b", "c"), c(3, 4, 2))),
  row_split = data.frame(rep(c("1", "2", "3", "4"), c(1, 1, 1, 3))),
  column_gap = unit(0.3, "cm"), 
  row_gap = unit(0.3, "cm"))
```

### Bar Graph with ASV Counts

```{r Number of genes in each module from WGCNA analysis, echo = F}
nrow(cyan_annotated)
nrow(darkgreen_annotated)
nrow(purple_annotated)
nrow(green_annotated)
nrow(darkred_annotated)
nrow(orange_annotated)
# View(green_annotated)
View(purple_annotated)
# View(cyan_annotated)
```

```{r gene count heatmap prep, include = F}
gene_bg <-
  cbind(c(
    nrow(orange_annotated),
    nrow(darkred_annotated),
    nrow(darkgreen_annotated),
    nrow(cyan_annotated),
    nrow(green_annotated),
    nrow(purple_annotated)
  ))
colnames(gene_bg) <- c("ASVs")
rownames(gene_bg) <-
  c(
    "Orange",
    "Dark Red",
    "Dark Green",
    "Cyan",
    "Green",
    "Purple"
  )
gene_bg %>%
  as.data.frame() %>%
  rownames_to_column(var = "module") -> gene_bg
gene_bg$module <-
  factor(
    gene_bg$module,
    c(
    "Orange",
    "Dark Red",
    "Dark Green",
    "Cyan",
    "Green",
    "Purple"
    )
  )
str(gene_bg)
```

```{r bargraph of genes for MM heatmap, fig.height=15, fig.width=5}
ggplot(data = gene_bg, (aes(module, ASVs, fill = module))) +
  geom_bar(
    stat = "identity",
    fill = c("Orange",
    "Dark Red",
    "Dark Green",
    "Cyan",
    "Green",
    "Purple"),
    position = "dodge"
  ) +
  theme_classic() +
  theme(
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    axis.line.x = element_blank(),
    axis.line = element_line(),
    axis.title.y = element_blank(),
    axis.line.y = element_blank(),
    legend.position = "none",
    text = element_text(size = 15)
  ) +
  scale_y_continuous(position = "left",
                     trans = "log10") +
  coord_flip() +
  geom_text(aes(label = ASVs),
            hjust = -1.5,
            label.size = 8)
```

### Cyan Pie Plots

```{r}
cyan_annotated %>%
  drop_na(domain, 
          phylum) -> cyan_na_removed

cyan_na_removed %>% 
  group_by(phylum) %>% 
  summarise(count = n()) %>% 
  dplyr::filter(!count < 5) %>% 
  mutate(Percentage = count/sum(count) * 100) -> cyan_phylum_group
View(cyan_phylum_group)

cyan_na_removed %>% 
  dplyr::filter(phylum %in% "Proteobacteria") -> cyan_proteo

cyan_proteo %>% 
  dplyr::filter(class %in% "Gammaproteobacteria") %>% 
  group_by(order) %>% 
  summarise(count = n()) %>% 
  dplyr::filter(!count <4.5) %>% 
  mutate(Percentage = count/sum(count) * 100) -> cyan_gamma_prot

cyan_proteo %>% 
  dplyr::filter(class %in% "Alphaproteobacteria") %>% 
  group_by(order) %>% 
  summarise(count = n()) %>% 
  dplyr::filter(!count <1.5) %>% 
  mutate(Percentage = count/sum(count) * 100) -> cyan_alpha_prot

View(cyan_alpha_prot)
View(cyan_gamma_prot)
```

```{r piechart cyan phylum, fig.width=7, fig.height=7}
cyan_phylum_group %>% 
  mutate(csum = rev(cumsum(rev(Percentage))), 
         pos = Percentage/2 + lead(csum, 1),
         pos = if_else(is.na(pos), Percentage/2, pos)) -> cyan_phy_posish

ggplot(cyan_phylum_group, aes(x = "phylum", y = Percentage, fill = phylum)) + 
  geom_col(width = 1, color = 1) +
  coord_polar("y", start=0) + 
  geom_label_repel(data = cyan_phy_posish,
                   aes(y = pos, label = paste0(Percentage %>% round(2), "%")),
                   size = 2, nudge_x = 1, show.legend = FALSE) + 
  guides(fill = guide_legend(title = "Phylum")) + 
  theme_void()
```

```{r alpha prot cyan pie charts, fig.width=7, fig.height=7}
cyan_alpha_prot %>% 
  mutate(csum = rev(cumsum(rev(Percentage))), 
         pos = Percentage/2 + lead(csum, 1),
         pos = if_else(is.na(pos), Percentage/2, pos)) -> cyan_alpha_posish

ggplot(cyan_alpha_prot, aes(x = "", y = Percentage, fill = order)) + 
  geom_col(width = 1, color = 1) +
  coord_polar("y", start=0) +
  geom_label_repel(data = cyan_alpha_posish,
                   aes(y = pos, label = paste0(Percentage %>% round(2), "%")),
                   size = 2, nudge_x = 1, show.legend = FALSE) + 
  guides(fill = guide_legend(title = "Order")) + 
  theme_void()

cyan_gamma_prot %>% 
  mutate(csum = rev(cumsum(rev(Percentage))), 
         pos = Percentage/2 + lead(csum, 1),
         pos = if_else(is.na(pos), Percentage/2, pos)) -> cyan_gamma_posish

ggplot(cyan_gamma_prot, aes(x = "", y = Percentage, fill = order)) + 
  geom_col(width = 1, color = 1) +
  coord_polar("y", start=0) +
  geom_label_repel(data = cyan_gamma_posish,
                   aes(y = pos, label = paste0(Percentage %>% round(2), "%")),
                   size = 2, nudge_x = 1, show.legend = FALSE) + 
  guides(fill = guide_legend(title = "Order")) + 
  theme_void()
```





## Picrust2 Analysis of WGCNA Modules

```{r making objects we need for picrust, echo = F}
# otu_table(physeq_main_10) %>% 
#   as.data.frame() %>% 
#   rownames_to_column(var = "ASV_number") %>% 
#   dplyr::filter(ASV_number %in% WGCNA_4_picrust$ASV_number) %>% 
#   column_to_rownames(var = "ASV_number") -> ASV_4_picrust_WGCNA
# 
# tfall_coral %>% 
#   as.data.frame() %>% 
#   dplyr::select(1, 7, 9, 13, 19) %>%
#   rownames_to_column(var = "SampleID") -> metadata_4_picrust_WGCNA
# 
# metadata_4_picrust_WGCNA$SampleID == colnames(ASV_4_picrust_WGCNA)
# 
# DHE_sequence %>% 
#   dplyr::filter(ASV_number %in% WGCNA_4_picrust$ASV_number) -> seq_4_picrust_WGCNA
```

```{r writing metadata as tsv, include = F}
# write_tsv(metadata_4_picrust_WGCNA,
#           file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/picrust_wgcna/metadata.tsv")
```

```{r writing fna for fasta file for picrust, include = F}
# dataframe2fas(seq_4_picrust_WGCNA %>%
#                dplyr::select(ASV_number, Sequence),
#              file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/picrust_wgcna/seqs.fna")
```

```{r writing count matrix as biom file for picrust, include = F}
# make_biom(ASV_4_picrust_WGCNA) -> count_4_picrust_biom
# write_biom(count_4_picrust_biom,
#            biom_file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/picrust_wgcna/table.biom")
```

```{bash picrust command for WGCNA ASVs}
#cd ~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript
#conda activate picrust2
#picrust2_pipeline.py \
#-s r_saved_files/picrust_16s/seqs.fna \
#-i r_saved_files/picrust_16s/table.biom \
#-o picrust_stratified_WGCNA \
#-p 2 \
#--in_traits EC,KO,PFAM,COG \
#--stratified \
#--verbose
```


## WACNA Analysis K Counts

```{r}
read.table(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/picrust_output/KO_metagenome_out/pred_metagenome_unstrat_descrip.tsv", 
           header = T, 
           sep = "\t") -> pi2_k_out
# View(pi2_k_out)
```

```{r SUPER DUPER IMPORTANT FOR WGCNA TO DO THIS, include = F}
options(stringsAsFactors = FALSE)
library(WGCNA)
```

```{r Making the all trait file for correlatvie analysis, include = F}
tfall_coral %>% 
  rownames_to_column(var = "samps") %>% 
  mutate(samps = str_replace_all(samps, "-", ".")) %>%
  dplyr::select(samps, Inoculated, basic_health, Health_disease) %>%
  column_to_rownames(var = "samps") -> WGCNA_treat_cato 

model.matrix(~Inoculated-1, WGCNA_treat_cato) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Sample_numbers") %>% 
  full_join(model.matrix(~Health_disease-1, WGCNA_treat_cato) %>% 
              as.data.frame() %>% 
              rownames_to_column(var = "Sample_numbers")) %>% 
  full_join(model.matrix(~basic_health-1, WGCNA_treat_cato) %>% 
              as.data.frame() %>% 
              rownames_to_column(var = "Sample_numbers")) %>%
  dplyr::select(1, 2, 3, 12, 13, 11, 6, 8, 7, 4, 10, 9, 5) -> allTraits
# View(allTraits)
```

```{r Making WGCNA gene objects and checking everything matches up, echo = F}
pi2_k_out %>% 
  dplyr::select(-2) %>% 
  column_to_rownames(var = "function.") -> pi_4_wgcna

genes4wgcna <- t(pi_4_wgcna)
genes4wgcna <- as.data.frame(genes4wgcna)
nrow(genes4wgcna)
dim(genes4wgcna)

gsg = goodSamplesGenes(genes4wgcna, verbose = 3)
gsg$allOK
```

```{r Sample clustering to identify outliers, fig.height=20, fig.width=20, echo = F}
sampleTree = hclust(dist(genes4wgcna), method = "ward.D2");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
sizeGrWindow(12,9)
#pdf(file = "Plots/sampleClustering.pdf", width = 12, height = 9);
par(cex = 1);
par(mar = c(0,4,2,0))
plot(sampleTree, 
     main = "Sample clustering to detect outliers", 
     sub="", 
     xlab="", 
     cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
# Plot a line to show the cut
abline(h = 8e+06, col = "red");
# Determine cluster under the line
clust = cutreeStatic(sampleTree, cutHeight = 8e+06, minSize = 10)
table(clust)
# clust 1 contains the samples we want to keep.
keepSamples = (clust==1)
datExpr = genes4wgcna[keepSamples, ]
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
```

There are no crazy outliers so going to include everything for the downstream WGCNA analysis. 

```{r Correlative data matching with WGCNA gene objects, echo = F}
#4. Create an object with the treatment file
#Create an object called "datTraits" that contains your trait data
head(allTraits)
str(allTraits)
#form a data frame analogous to expression data that will hold the clinical traits.
rownames(allTraits) = allTraits$Sample_numbers
allTraits$Sample_numbers = NULL
table(rownames(allTraits)==rownames(genes4wgcna)) #should return TRUE if datasets align correctly, otherwise your names are out of order
```

Everything matches which is fantastic news between WGCNA gene object and the treatment file. 

```{r Identyfying Soft Power, echo = F}
# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5, dataIsExpr = T)
# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```
Soft power identified as 7, curves look good so no need for additional data filtering. 

```{r Signed adjacency matrix using soft power from last step, include = F}
softPower = 8;
adjacency = adjacency(datExpr, type="signed", power = softPower);
```

```{r Generation of topological overlap, include = F}
# Turn adjacency into topological overlap
TOM = TOMsimilarity(adjacency);
dissTOM = 1-TOM
```

```{r Gene clustering with TOM-based dissimilarity plot, fig.width=10, fig.height=4}
# Call the hierarchical clustering function
geneTree = hclust(as.dist(dissTOM), 
                  method = "average");
# Plot the resulting clustering tree (dendrogram)
sizeGrWindow(12,9)
plot(geneTree, 
     xlab="", 
     sub="", 
     main = "Gene clustering on TOM-based dissimilarity",
labels = FALSE, hang = 0.04);
```

```{r Module identification, echo = F}
# We like large modules, so we set the minimum module size relatively high:
minModuleSize = 30;
# Module identification using dynamic tree cut:
dynamicMods = cutreeDynamic(dendro = geneTree, 
                            distM = dissTOM,
                            deepSplit = 2,
                            pamRespectsDendro = FALSE, 
                            minClusterSize = minModuleSize);
table(dynamicMods)
```

```{r Gene dendogram and module colours plot, echo = F, fig.width = 7, fig.height= 2}
# Convert numeric lables into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
# Plot the dendrogram and colors underneath
sizeGrWindow(8,6)
plotDendroAndColors(geneTree, 
                    dynamicColors, 
                    "Dynamic Tree Cut", 
                    dendroLabels = FALSE, 
                    hang = 0.03,
                    addGuide = TRUE, 
                    guideHang = 0.05, 
                    main = "Gene dendrogram and module colors")
```

```{r Merging similar modules at 0.40, fig.height=6, fig.width=10, echo = F}
# Calculate eigengenes
MEList = moduleEigengenes(datExpr, 
                          colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), 
                method = "average");
# Plot the result
sizeGrWindow(7, 6)

plot(METree, 
     main = "Clustering of module eigengenes",
     xlab = "", 
     sub = "")
MEDissThres = 0.25
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")
```

```{r Merging similar modules, include = F}
# Call an automatic merging function
merge = mergeCloseModules(datExpr, 
                          dynamicColors, 
                          cutHeight = MEDissThres, 
                          verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs;
```

```{r Cluster dendogram with original and merged modules, fig.height=3, fig.width=10, echo = F}
sizeGrWindow(12, 5)
#pdf(file = "Plots/geneDendro-3.pdf", wi = 9, he = 6)
plotDendroAndColors(geneTree, 
                    cbind(dynamicColors, 
                          mergedColors), 
                    c("Dynamic Tree Cut", "Merged dynamic"), 
                    dendroLabels = FALSE, 
                    hang = 0.03, 
                    addGuide = TRUE, 
                    guideHang = 0.05)
#dev.off()
```

```{r Eigengenes for each sample to each module, echo = F}
# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", 
               standardColors(50));
moduleLabels = match(moduleColors, 
                     colorOrder)-1
MEs = mergedMEs
# View(MEs)
```

```{r Significance of eigengenes to correlative data, inclue = F}
# Define numbers of genes and samples
nGenes = ncol(datExpr);
nSamples = nrow(datExpr);
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(datExpr, 
                        moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, 
                     allTraits, 
                     use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, 
                                     nSamples);
```

```{r writing for correlation analysis, include = F}
# write.csv(MEs0, 
#           file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/correlative_analysis/dhe_16s_eigengenes_picrust.csv")
```


Pipeline all run now for visualization and analysis of the modules. 

```{r summary data for outputs from WGCNA, include = F}
MEs %>% 
  rownames_to_column(var = "ASV_number") %>% 
  inner_join(tfall_coral %>% 
               rownames_to_column(var = "ASV_number"), 
             by = "ASV_number") -> MEs_and_tfall
```

```{r All modules module-trait relationship plot, fig.height=6, fig.width=8}
sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(
  Matrix = moduleTraitCor,
  xLabels = names(allTraits),
  yLabels = names(MEs),
  ySymbols = names(MEs),
  colorLabels = FALSE,
  colors = blueWhiteRed(50),
  textMatrix = textMatrix,
  setStdMargins = FALSE,
  cex.text = 0.5,
  zlim = c(-1, 1),
  cex.lab = 0.5,
  main = paste("Module-trait relationships")
)
```

```{r Hub gene for each module identified form analysis, echo = F}
chooseTopHubInEachModule(genes4wgcna,
                                   moduleColors,
                                   omitColors = "grey",
                                   power = 10) %>% 
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>%
  rownames_to_column(var = "module") %>%
  dplyr::inner_join(pi2_k_out %>% 
               dplyr::select(1,2),
             by = "function.") -> tophub_annotated
View(tophub_annotated)

# write.csv(tophub_annotated,
#          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files//tophub_annotated_16s_k.csv",
#          row.names = F)
```

```{r dataframes of genes within each module from WGCNA analysis, include = F}
names(genes4wgcna)[moduleColors == "lightcyan"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "lightcyan") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> lightcyan_annotated
# View(lightcyan_annotated)

names(genes4wgcna)[moduleColors == "green"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "green") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> green_annotated
# View(green_annotated)

names(genes4wgcna)[moduleColors == "salmon"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "salmon") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> salmon_annotated
# View(salmon_annotated)

names(genes4wgcna)[moduleColors == "darkred"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "darkred") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> darkred_annotated
# View(darkred_annotated)

names(genes4wgcna)[moduleColors == "tan"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "tan") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> tan_annotated
# View(tan_annotated)

names(genes4wgcna)[moduleColors == "midnightblue"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "midnightblue") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> midnightblue_annotated
# View(midnightblue_annotated)

names(genes4wgcna)[moduleColors == "black"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "black") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> black_annotated
# View(black_annotated)

names(genes4wgcna)[moduleColors == "magenta"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "magenta") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> magenta_annotated
# View(magenta_annotated)

names(genes4wgcna)[moduleColors == "lightgreen"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "lightgreen") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> lightgreen_annotated
# View(lightgreen_annotated)

names(genes4wgcna)[moduleColors == "blue"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "blue") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> blue_annotated
# View(blue_annotated)

names(genes4wgcna)[moduleColors == "red"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "red") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> red_annotated
# View(red_annotated)

names(genes4wgcna)[moduleColors == "yellow"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "yellow") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> yellow_annotated
# View(yellow_annotated)

names(genes4wgcna)[moduleColors == "lightyellow"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "lightyellow") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> lightyellow_annotated
# View(lightyellow_annotated)

names(genes4wgcna)[moduleColors == "purple"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "purple") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> purple_annotated
# View(purple_annotated)

names(genes4wgcna)[moduleColors == "greenyellow"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "greenyellow") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> greenyellow_annotated
# View(greenyellow_annotated)

names(genes4wgcna)[moduleColors == "cyan"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "cyan") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> cyan_annotated
# View(cyan_annotated)

names(genes4wgcna)[moduleColors == "grey60"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "grey60") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> grey60_annotated
# View(grey60_annotated)

names(genes4wgcna)[moduleColors == "royalblue"] %>%
  as.data.frame() %>%
  dplyr::rename(., function. = .) %>% 
  mutate(module = "royalblue") %>%
  inner_join(pi2_k_out %>% 
               dplyr::select(1,2)) -> royalblue_annotated
View(royalblue_annotated)
```

```{r gene lists from picrsut wgcna, include = F}
write.csv(lightcyan_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/lightcyan.csv")

write.csv(green_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/green.csv")

write.csv(salmon_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/salmon.csv")

write.csv(darkred_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/darkred.csv")

write.csv(tan_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/tan.csv")

write.csv(midnightblue_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/midnightblue.csv")

write.csv(royalblue_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/royalblue.csv")

write.csv(black_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/black.csv")

write.csv(magenta_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/magenta.csv")

write.csv(lightgreen_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/lightgreen.csv")

write.csv(blue_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/blue.csv")

write.csv(red_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/red.csv")

write.csv(yellow_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/yellow.csv")

write.csv(lightyellow_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/lightyellow.csv")

write.csv(purple_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/purple.csv")

write.csv(greenyellow_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/greenyellow.csv")

write.csv(cyan_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/cyan.csv")

write.csv(grey60_annotated, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_lists/grey60.csv")
```


### Sexy Plot

```{r complex heatmap dataframes with correct columns, include = F}
textMatrix %>%
  as.data.frame() %>% 
  dplyr::slice(4, 12, 14, 15, 18, 2, 3, 1, 11, 13) %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) -> tmatrix_ch

moduleTraitCor %>%
  as.data.frame() %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) %>% 
  t() %>%
  as.data.frame() %>% 
  rownames_to_column(var = "remove") %>% 
  dplyr::select(-remove) %>%
  mutate(
    Row_annotations = c(
      "Pre Exposure",
      "Visually Healthy",
      "Diseased", 
      "Health Slurry (VH)",
      "Disease Slurry (VH)",
      "Serratia Placebo (VH)",
      "Serratia (VH)",
      "Disease Slurry (D)",
      "Serratia (D)"
    )
  ) %>% 
  column_to_rownames(var = "Row_annotations") %>% 
  dplyr::select(MEdarkred, MEred, MElightyellow, MEpurple, MEgrey60, MEgreen, MEsalmon, MElightcyan, MEblue, MEyellow) %>%
  t() %>%
  as.data.frame() -> modtraitcor_ch

allTraits %>% 
  as.data.frame() %>%
dplyr::select(3:12) -> alltrait_ch
```

```{r Complex heatmap prep for WGCNA analysis, include =F}
ccann <- data.frame(rownames(modtraitcor_ch))
colnames(ccann) <- c("Modules")
colcol <-
  list(
    "Modules" = c(
      "MEdarkred" = "Dark Red",
      "MEred" = "Red",
      "MElightyellow" = "Light Yellow",
      "MEpurple" = "Purple",
      "MEgrey60" = "Grey 60",
      "MEgreen" = "Green",
      "MEsalmon" = "Salmon",
      "MElightcyan" = "Light Cyan",
      "MEblue" = "Blue",
      "MEyellow" = "Yellow"))

samname <- HeatmapAnnotation(
  df = ccann,
  col = colcol,
  which = "column",
  simple_anno_size = unit(0.5, "cm"),
  annotation_name_gp = gpar(fontsize = 10), 
  show_annotation_name = F)

#col_fun = blueWhiteRed(50)
col_fun = colorRamp2(c(-1, 0, 1), c("royalblue", "white", "red"))
```

```{r WGCNA complex heatmap, echo = F, fig.width = 6.5, fig.height=3}
Heatmap(
  modtraitcor_ch %>% t(),
  cluster_rows = F,
  cluster_columns = F,
  show_row_names = T,
  top_annotation = samname, 
  show_column_names = F,
  row_names_side = "left",
  column_names_rot = 45,
  col = col_fun,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(sprintf(tmatrix_ch[j, i]), x, y, gp = gpar(fontsize = 10))
  },
  row_split = data.frame(rep(c("a", "b", "c"), c(3, 4, 2))),
  column_split = data.frame(rep(c("1", "2","3", "4", "5"), c(2, 2, 1, 2, 3))),
  column_gap = unit(0.3, "cm"), 
  row_gap = unit(0.3, "cm"))
```

### Sexy Plot

```{r complex heatmap dataframes with correct columns, include = F}
textMatrix %>%
  as.data.frame() %>% 
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) -> tmatrix_ch

moduleTraitCor %>%
  as.data.frame() %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) %>% 
  t() %>%
  as.data.frame() %>% 
  rownames_to_column(var = "remove") %>% 
  dplyr::select(-remove) %>%
  mutate(
    Row_annotations = c(
      "Pre Exposure",
      "Visually Healthy",
      "Diseased", 
      "Health Slurry (VH)",
      "Disease Slurry (VH)",
      "Serratia Placebo (VH)",
      "Serratia (VH)",
      "Disease Slurry (D)",
      "Serratia (D)"
    )
  ) %>%
  column_to_rownames(var = "Row_annotations") %>% 
  dplyr::select(MElightcyan, MEgreen, MEsalmon, MEdarkred, MEtan, MEmidnightblue, MEroyalblue, MEblack, MEmagenta, MElightgreen, MEblue, MEred, MEyellow, MElightyellow, MEpurple, MEgreenyellow, MEcyan, MEgrey60) %>%
  t() %>%
  as.data.frame() -> modtraitcor_ch

allTraits %>% 
  as.data.frame() %>%
dplyr::select(3:12) -> alltrait_ch
```

```{r Complex heatmap prep for WGCNA analysis, include =F}
ccann <- data.frame(rownames(modtraitcor_ch))
colnames(ccann) <- c("Modules")
colcol <-
  list(
    "Modules" = c(
      "MElightcyan" = "Light Cyan",
      "MEgreen" = "Green",
      "MEsalmon" = "Salmon",
      "MEdarkred" = "Dark Red",
      "MEtan" = "Tan",
      "MEmidnightblue" = "Midnight Blue",
      "MEroyalblue" = "Royal Blue",
      "MEblack" = "Black",
      "MEmagenta" = "Magenta",
      "MElightgreen" = "Light Green",
      "MEblue" = "Blue",
      "MEred" = "Red",
      "MEyellow" = "Yellow",
      "MElightyellow" = "Light Yellow",
      "MEpurple" = "Purple",
      "MEgreenyellow" = "Green Yellow", 
      "MEcyan" = "Cyan", 
      "MEgrey60" = "Grey 60"))

samname <- HeatmapAnnotation(
  df = ccann,
  col = colcol,
  which = "row",
  simple_anno_size = unit(0.5, "cm"),
  annotation_name_gp = gpar(fontsize = 10), 
  show_annotation_name = F)

#col_fun = blueWhiteRed(50)
col_fun = colorRamp2(c(-1, 0, 1), c("royalblue", "white", "red"))
```

```{r WGCNA complex heatmap, echo = F, fig.width = 9.5, fig.height=8}
Heatmap(
  modtraitcor_ch %>% as.matrix(),
  cluster_rows = F,
  cluster_columns = F,
  show_row_names = T,
  left_annotation = samname,
  row_names_side = "left",
  column_names_rot = 45,
  col = col_fun,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(sprintf(tmatrix_ch[i, j]), x, y, gp = gpar(fontsize = 10))
  },
  column_split = data.frame(rep(c("a", "b", "c"), c(3, 4, 2))),
  column_gap = unit(0.3, "cm"))
```


### Bar Graph with ASV Counts

```{r Number of genes in each module from WGCNA analysis, echo = F}
nrow(cyan_annotated)
nrow(darkgreen_annotated)
nrow(purple_annotated)
nrow(green_annotated)
nrow(darkred_annotated)
nrow(orange_annotated)
```

```{r gene count heatmap prep, include = F}
gene_bg <-
  cbind(c(
    nrow(yellow_annotated),
    nrow(blue_annotated),
    nrow(lightcyan_annotated),
    nrow(salmon_annotated),
    nrow(green_annotated),
    nrow(grey60_annotated), 
    nrow(purple_annotated), 
    nrow(lightyellow_annotated), 
    nrow(red_annotated), 
    nrow(darkred_annotated)
  ))
colnames(gene_bg) <- c("ASVs")
rownames(gene_bg) <-
  c(
    "Yellow",
    "Blue",
    "Light Cyan",
    "Salmon",
    "Green",
    "Grey 60", 
    "Purple", 
    "Light Yellow", 
    "Red", 
    "Dark Red"
  )
gene_bg %>%
  as.data.frame() %>%
  rownames_to_column(var = "module") -> gene_bg
gene_bg$module <-
  factor(
    gene_bg$module,
    c(
    "Yellow",
    "Blue",
    "Light Cyan",
    "Salmon",
    "Green",
    "Grey 60", 
    "Purple", 
    "Light Yellow", 
    "Red", 
    "Dark Red"
    )
  )
str(gene_bg)
```

```{r}
nrow(blue_annotated)
```

```{r bargraph of genes for MM heatmap, fig.height=4, fig.width=15}
ggplot(data = gene_bg, (aes(module, ASVs, fill = module))) +
  geom_bar(
    stat = "identity",
    fill = c("Yellow",
    "Blue",
    "Light Cyan",
    "Salmon",
    "Green",
    "Grey 60", 
    "Purple", 
    "Light Yellow", 
    "Red", 
    "Dark Red"),
    position = "dodge"
  ) +
  theme_classic() +
  theme(
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    axis.line.x = element_blank(),
    axis.line = element_line(),
    axis.title.y = element_blank(),
    axis.line.y = element_blank(),
    legend.position = "none",
    text = element_text(size = 15)
  ) +
  scale_y_continuous(position = "left",
                     trans = "log10") +
  geom_text(aes(label = ASVs),
            hjust = -1.5,
            label.size = 8)
```

### ClusterProfiler of Modules

```{r clusterprofiler loading, include = F}
library(clusterProfiler)
```

```{r running KEGG analysis, include = T}
kegg_lightcyan <- enrichKEGG(
  gene = lightcyan_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_green <- enrichKEGG(
  gene = green_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_salmon <- enrichKEGG(
  gene = salmon_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_darkred <- enrichKEGG(
  gene = darkred_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_tan <- enrichKEGG(
  gene = tan_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_midnightblue <- enrichKEGG(
  gene = midnightblue_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_black <- enrichKEGG(
  gene = black_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_magenta <- enrichKEGG(
  gene = magenta_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_lightgreen <- enrichKEGG(
  gene = lightgreen_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_blue <- enrichKEGG(
  gene = blue_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_red <- enrichKEGG(
  gene = red_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_yellow <- enrichKEGG(
  gene = yellow_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_lightyellow <- enrichKEGG(
  gene = lightyellow_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_purple <- enrichKEGG(
  gene = purple_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_greenyellow <- enrichKEGG(
  gene = greenyellow_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_cyan <- enrichKEGG(
  gene = cyan_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_grey60 <- enrichKEGG(
  gene = grey60_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_royalblue <- enrichKEGG(
  gene = royalblue_annotated$function.,
  organism = 'ko',
  pvalueCutoff = 0.05
)
```

```{r}
kegg_lightcyan %>% 
  as.data.frame() -> kegg_lightcyan_res

kegg_green %>% 
  as.data.frame() -> kegg_green_res

kegg_salmon %>% 
  as.data.frame() -> kegg_salmon_res

kegg_darkred %>% 
  as.data.frame() -> kegg_darkred_res

kegg_tan %>% 
  as.data.frame() -> kegg_tan_res

kegg_midnightblue %>% 
  as.data.frame() -> kegg_midnightblue_res

kegg_black %>% 
  as.data.frame() -> kegg_black_res

kegg_magenta %>% 
  as.data.frame() -> kegg_magenta_res

kegg_lightgreen %>% 
  as.data.frame() -> kegg_lightgreen_res

kegg_blue %>% 
  as.data.frame() -> kegg_blue_res

kegg_red %>% 
  as.data.frame() -> kegg_red_res

kegg_yellow %>% 
  as.data.frame() -> kegg_yellow_res

kegg_lightyellow %>% 
  as.data.frame() -> kegg_lightyellow_res

kegg_purple %>% 
  as.data.frame() -> kegg_purple_res

kegg_greenyellow %>% 
  as.data.frame() -> kegg_greenyellow_res

kegg_cyan %>% 
  as.data.frame() -> kegg_cyan_res

kegg_grey60 %>% 
  as.data.frame() -> kegg_grey60_res

kegg_royalblue%>% 
  as.data.frame() -> kegg_royalblue_res
```

```{r writing IMF kegg module results to csv, include = F}
write.csv(kegg_black_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_black.csv")

write.csv(kegg_lightcyan_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_lightcyan.csv")

write.csv(kegg_green_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_green.csv")

write.csv(kegg_salmon_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_salmon.csv")

write.csv(kegg_darkred_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_darkred.csv")

write.csv(kegg_tan_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_tan.csv")

write.csv(kegg_midnightblue_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_midnightblue.csv")

write.csv(kegg_royalblue_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_royalblue.csv")

write.csv(kegg_magenta_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_magenta.csv")

write.csv(kegg_lightgreen_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_lightgreen.csv")

write.csv(kegg_blue_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_blue.csv")

write.csv(kegg_red_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_red.csv")

write.csv(kegg_yellow_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_yellow.csv")

write.csv(kegg_lightyellow_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_lightyellow.csv")

write.csv(kegg_purple_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_purple.csv")

write.csv(kegg_greenyellow_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_greenyellow.csv")

write.csv(kegg_cyan_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_cyan.csv")

write.csv(kegg_grey60_res, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/picrust_wgcna_module_kegg_res/kegg_res_grey60.csv")
```

```{r kegg barplots from k 16s enrichment, fig.height = 10}
#barplot(kegg_lightcyan, showCategory = 100)
barplot(kegg_green, showCategory = 100)
barplot(kegg_salmon, showCategory = 100)
barplot(kegg_darkred, showCategory = 100)
barplot(kegg_tan, showCategory = 100)
barplot(kegg_midnightblue, showCategory = 100)
barplot(kegg_black, showCategory = 100)
barplot(kegg_magenta, showCategory = 100)
barplot(kegg_lightgreen, showCategory = 100)
barplot(kegg_blue, showCategory = 100)
barplot(kegg_red, showCategory = 100)
barplot(kegg_yellow, showCategory = 100)
barplot(kegg_lightyellow, showCategory = 100)
barplot(kegg_purple, showCategory = 100)
barplot(kegg_greenyellow, showCategory = 100)
barplot(kegg_cyan, showCategory = 100)
barplot(kegg_grey60, showCategory = 100)
```

```{r Darkred KEGG viewing genes}
kegg_darkred_res
browseKEGG(kegg_darkred, "ko02020")
browseKEGG(kegg_darkred, "ko02025")
browseKEGG(kegg_darkred, "ko00500")
```

```{r red kegg geneview}
kegg_red_res
browseKEGG(kegg_red, "ko02026")
browseKEGG(kegg_red, "ko00190")
browseKEGG(kegg_red, "ko02020")
View(kegg_red_res)
```

```{r lightyellow kegg overview}
kegg_lightyellow_res
browseKEGG(kegg_lightyellow, "ko00470")
browseKEGG(kegg_lightyellow, "ko00430")
```

```{r}
kegg_purple_res
browseKEGG(kegg_purple, 'ko00640')
browseKEGG(kegg_purple, "ko00450")
browseKEGG(kegg_purple, "ko02026")
```

```{r}
kegg_grey60_res
browseKEGG(kegg_grey60, 'ko05150')
browseKEGG(kegg_grey60, "ko00550")
browseKEGG(kegg_grey60, "ko00900")
```

Genes involved in bacterial colonisation. 
Both bioysnthesis are important in cell walls of prokaryotes

```{r Green KEGG viewing genes}
kegg_green_res
browseKEGG(kegg_green, "ko00190")
```

```{r salmon kegg viewing genes}
kegg_salmon_res
browseKEGG(kegg_salmon, "ko00720")
browseKEGG(kegg_salmon, "ko00430")
```

```{r blue kegg viewing genes}
View(kegg_blue_res)
browseKEGG(kegg_salmon, "ko00720")
browseKEGG(kegg_salmon, "ko00430")
```

```{r}
kegg_yellow_res
browseKEGG(kegg_yellow, "ko01053")
browseKEGG(kegg_yellow, "ko02025")
browseKEGG(kegg_yellow, "ko02040")
browseKEGG(kegg_yellow, "ko00571")
browseKEGG(kegg_yellow, "ko00627")
```


Lipoarabinomannan (LAM) biosynthesis	 - innate mimune system recognise and increase expression


Control, VH and Diseased significant correlation modules
*Dark Red* 
- Two component system - stim response coupling mechanism to allow organisms to respond to changes in environmental conditions
- Bio film, formation - irreversibly attach and grown on a surface (alters phenotype growth and transcription)
https://www.google.com/search?q=biofilm+formation&oq=biofilm+formation+&aqs=chrome..69i57j0i433i512j0i512l5j69i61.2260j0j7&sourceid=chrome&ie=UTF-8 
- Starch and sucrose metabolism - increased can cause increases in carcinogenic potential. Change in gene expression
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4922512/ 

```{r}
kegg_darkred_res
browseKEGG(kegg_darkred, "ko02020")
browseKEGG(kegg_darkred, "ko02025")
```

#### Lists for Writing Discussion 

```{r}
View(darkred_annotated)
View(kegg_darkred_res)

View(red_annotated)
View(kegg_red_res)
```
