---
title: "DHE_manuscript_analysis"
author: "Benjamin Young"
date: "16/02/2022"
output: html_document
---

```{r package installing chunk, include = F}
## BiocManager::install("thacklr")
```

```{r library loading, include = F}
library(tximport)
library(tidyverse)
library(DESeq2)
library(edgeR)
library(ggrepel)
library(WGCNA)
library(DEGreport)
library(PCAtools)
library(venn)
library(clusterProfiler)
library(ComplexHeatmap)
library(circlize)
library(ape)
library(dendextend)
library(enrichplot)
library(scales)
```

Removed thacklr and evemodel

```{r Package Versions for manuscript, include = F}
packageVersion("clusterProfiler")
package.version("enrichplot")
citation("enrichplot")
citation("clusterProfiler")
citation("venn")
```

# Organising Sequenced Samples from Original Lane 1 and Lane 2 and Redo Samples in Lane 3

Sequencing was run on 3 lanes.
- Lane 1
- Lane 2
- Lane 3 (redo of failed samples from lane 1 and lane 2)

I needed to 
1. remove failed reads from lane 1 and lane 2

## Lane 1 and Lane 2

```{r Lane 1 and 2 samples succesfully quanted using Salmon, include=FALSE}
## Loading quantification data output from the slippery Salmon
# setwd("~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/analysis_prep/DHE_salmon_l1l2/")
# list.files(path = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/analysis_prep/DHE_salmon_l1l2//", 
#            full.names = F, 
#            pattern = "\\_salmon$") -> PPall
#View(PPall)
# length(PPall)
```

290 samples in lane 1 and lane 2

```{r Sequencing metadata file with all files sent for sequencing, include = F}
# read.csv("~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/sequencing_documents/sequenced_samples.csv") %>% 
#   mutate(SAL = c("salmon")) %>% 
#   unite(., ID_salmon, c(sample_ID_2, SAL), sep = "_", remove = F) %>%
#   unite(., ID_salmon_2, c(sample_ID, SAL), sep = "_", remove = F) -> seq_tfall
# View(seq_tfall)
```

335 rows here in the sequencing metadata. Need to minus 15 as these are the disease inoculations that only had 16s sequencing performed on them. This leaves 320 tagseq samples in the sequencing metadata file. 

Sequencing metadata = 320 sequenced
Lane 1 and 2 = 290 sequenced succesfully

```{r making yes no presence of salmon files in the sequencing metadata, include = F}
# seq_tfall %>% 
#   filter(ID_salmon %in% PPall) %>%
#   mutate(Ln12_pres = c("yes")) %>% 
#   full_join(seq_tfall) %>% 
#   replace_na(list(Ln12_pres="no")) %>% 
#   select(Original_order ,ID_salmon, ID_salmon_2, Ln12_pres) -> lane12_samples
# View(lane12_samples)
```

```{r Checking lane 1 and 2 samples presentin sequencing metadata and quant files, echo = F}
# lane12_samples %>%
#  filter(Ln12_pres == "yes") %>% nrow()
# 
# lane12_samples %>% 
#    filter(!ID_salmon %in% PPall) %>% nrow()
```

290 of these samples match between lane 1 and lane 2 sequencing and sequencing metadata file. 

## Lane 3 (redo samples)

```{r length of salmon files in lane 3, include = F}
# setwd("~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/analysis_prep/DHE_salmon_l3/")
# list.files(path = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/analysis_prep/DHE_salmon_l3//", 
#            full.names = F, 
#            pattern = "\\_salmon$") -> PPall
# length(PPall)
#View(PPall)
```

There are 52 salmon quant files in lane 3. 

```{r maing yes no presence with sequencing metadata for lane 3, include = F}
# seq_tfall %>%
#   filter(ID_salmon_2 %in% PPall) %>%
#   mutate(Ln3_pres = c("yes")) %>%
#   full_join(seq_tfall) %>% 
#   replace_na(list(Ln3_pres = "no")) %>%
#   select(Original_order, ID_salmon, ID_salmon_2, Ln3_pres) -> lane3samples
# View(lane3samples)
```

```{r Checking lane 1 and 2 samples presentin sequencing metadata and quant files, echo = F}
# lane3samples %>%
#  filter(Ln3_pres == "yes") %>% nrow()
# 
# lane3samples %>% 
#    filter(!ID_salmon %in% PPall) %>% nrow()
```

All 52 of the lane 3 samples match files in the sequencing metadata, yay. 


## Combining Lane 1, Lane 2 and Lane 3

```{r combining 3 sequencing lanes, include = F}
# lane12_samples %>% 
#   full_join(lane3samples, by = c("ID_salmon", "ID_salmon_2")) -> ln123_samples
# View(ln123_samples)
# write.csv(ln123_samples, 
#           file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/analysis_prep/ln123_Samples_DHE.csv")
```

All redone samples in Lane 3 had better read depth than there successfully sequenced counterparts in Lane 1 and Lane 2. As such I am using the lane 3 samples for analysis and removed the duplicates in Lane 1 and Lane 2. The samples which completely failed in Lane 1 and Lane 2, but worked in Lane 3, have obviously been kept for donwstream analysis. 

I was lazy and did this by hand, generated a new folder called `remove_ln1_salmon` and moved them manually. I then made a new folder with all Lane 1, Lane 2 and Lane 3 samples to be used in salmon to generate the count table for gene expression analysis. This folder is called `salmon_quant`. 

These removed files from Lane 1 and Lane 2 are as follows. 

- APAL_CN2_85_heat_placebo_slurry_initial_salmon
- APAL_ML2_170_heat_disease_slurry_initial_salmon
- APAL_CN2_88_heat_placebo_slurry_initial_salmon
- APAL_ML2_11_ambient_disease_slurry_initial_salmon
- APAL_ML2_174_heat_s_m_placebo_initial_salmon
- APAL_ML2_38_ambient_placebo_slurry_initial_salmon
- APAL_CN2_93_heat_s_marcscens_initial_salmon
- APAL_ML2_37_heat_placebo_slurry_initial_salmon
- APAL_CN2_111_ambient_s_marcscens_initial_salmon
- APAL_ML2_22_ambient_s_marcscens_initial_salmon
- APAL_ML2_21_heat_disease_slurry_initial_salmon
- APAL_ML2_24_heat_s_m_placebo_initial_salmon
- APAL_CN2_90_ambient_disease_slurry_initial_salmon
- APAL_ML2_12_heat_s_marcscens_initial_salmon
- APAL_HS1_149_heat_s_marcscens_initial_salmon
- APAL_ML2_169_ambient_placebo_slurry_initial_salmon
- APAL_HS1_153_heat_s_marcscens_initial_salmon
- APAL_HS1_162_ambient_placebo_slurry_9th_aug_salmon
- APAL_CN2_99_heat_s_marcscens_11th_aug_salmon
- APAL_ML2_173_heat_placebo_slurry_17th_aug_salmon
- APAL_ML2_19_ambient_s_marcscens_17th_aug_salmon
- APAL_ML2_26_ambient_s_m_placebo_17th_aug_salmon

Due to different naming schemes for between Lane 1 + Lane 2 and Lane 3, I generated an additional column to make sure the correct salmon file names are present in the merged salmon quant file which will be used in downstream analysis. I could probably very easily do this in r but I cannot be bothered to edit this right now. 

In total 
- 320 tagseq samples in sequencing treatment file
- 0 samples failed completely 
- 52 samples in ln3
- 290 samples in Ln12

I removed the 52 duplicate samples in lane 1 and lane 2 that were redone, therfore we have the following for files to be used in gene expression analysis. 
Lane 1 and Lane 2 = 268 samples
Lane 3 = 52 samples

Total number of samples for gene expression analysis = 320

This 320 matches the total number submitted for sequencing so we have succesfully sequenced all samples. The quality and read counts may cause samples to drop. This is the next step. 

```{r checking salmon quant folder with Lane 1, Lane 2 and Lane 3 samples, echo = F}
# setwd("~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/salmon_quant/")
# list.files(path = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/salmon_quant//", 
#            full.names = F, 
#            pattern = "\\_salmon$") -> PPall
# length(PPall)
```

Have 320 salmon quant folders whhich is what we want after all the sample pre processing done above. 

```{r read in edited treatment file, include = F}
# read.csv("~/Dropbox/PhD/Projects/DHE/NGS/tagseq/sequenced_samples.csv") -> tfall
# str(tfall)
# View(tfall)
```


# Generating Count Matrix and Treatment File with Correct Variables needed for Analysis

The steps in this section are as follows.  
1. Making sure all salmon quant files are present in the `salmon_quant` folder
2. Generating a transcript to gene file to be used in salmon (need this as salmon wuants to transcript level but it is not possible to look at transcripts for 3 prime RNA-seq data).  

```{r Making Salmon Vectors Names, include=FALSE}
## Loading quantification data output from the slippery Salmon
# setwd("~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/salmon_quant/")
# PPall <- list.files(path = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/salmon_quant//", 
#                     full.names = F,
#                     pattern = "\\_salmon$")
# 
# length(PPall)
# FILESall <- file.path(PPall, "quant.sf")
# 
# names(FILESall) <- PPall
# head(FILESall)
# all(file.exists(FILESall))
```

## Reading in tx2gene
Column 1 must equal transcript ID
Column 2 must equal Gene ID

```{bash generating transcript to gene file, include = F}
#cat Apalm_assembly_notrna_v3.1_200911.gff3 \
#| grep -v "#" \
#| awk '$3=="mRNA"' \
#| cut -f9 | tr -s ";" " " \
#| awk '{print$1"\t"$2}' \
#| sort \
#| uniq \
#| sed 's/ID=//g' \
#| sed 's/Parent=//g' > tx2gene_apal1.tsv
```

```{r reading in transcript ot gene file generated from gff3 file in genome, echo = F}
# read.table(file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/genome_resources/tx2gene_apal.tsv") %>%
#   mutate(APAL = "Apalm_v3") %>% 
#   unite(V1, c(APAL, V1), sep = "_") %>% 
#   mutate(Transcript_ID = V1, 
#          Gene_ID = V2) %>% 
#   select(-V1, -V2) -> tx2gene
# 
# tx2gene$Transcript_ID %>% unique() %>% length()
# tx2gene$Gene_ID %>% unique() %>% length()
# View(tx2gene)
```

There are 31,827 transcripts in the Acropora palmata genome
There are 29,258 genes in the Acropora palmata genome

```{r Reading in Salmon Files with tyranscript to gene to quant to gene level, include=F}
## Importing the count data using salmon quant.sf files and the text to gene file
# setwd("~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/tagseq/salmon_quant/")
# txi.salmon.countall <- tximport(FILESall, 
#                                 type = "salmon", 
#                                 tx2gene = tx2gene)
# 
# txi.salmon.countall$counts -> DHE_counts
# View(DHE_counts)
```

Here we have 29,181 genes. This is okay and below total genes, some may have been lost filtering.  

Important to note, need to use the txi$counts for DeSeq2 as recommended in vignette. Done as  correcting the counts for gene length will induce a bias in your analysis, because the counts do not have length bias as they are only the 3 prime end of the mRNA molecules. 

```{r Reading in  sequencing treatment file, include = F}
# DHE_counts %>%
#   t() %>% 
#   rownames() -> DHE_samp
# length(DHE_samp)
# 
# seq_tfall %>% 
#   dplyr::select(-1, -2, -3, -5) %>%
#   dplyr::filter(ID_salmon %in% DHE_samp) %>% # 268 samples for lane 1 and 2 which is correct
#   rbind(seq_tfall %>% 
#                dplyr::select(-1, -3, -4, -5) %>% 
#                rename(., ID_salmon = ID_salmon_2) %>% 
#                dplyr::filter(ID_salmon %in% DHE_samp)) %>% # this filter gives 52 samples which is correct for lane 3
#   mutate(
#     Dtreatment = as.factor(Dtreatment),
#     timepoint = as.factor(timepoint),
#     Tank_treatment = as.factor(Tank_treatment),
#     Genotype = as.factor(Genotype), 
#     Frag_number = as.character(Frag_number), 
#     Disease_tank = as.factor(D_touch_tank)) -> seq_tfall_final
# 
# nrow(seq_tfall_final) #320 samples with the correct names for the columns of the count file 
```

```{r Generating additional treatment variables and removing placebo diseased and dead samples, include = F}
# read.csv("/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/physiological_data/metadata/health_long.csv") %>% 
#   mutate(Genotype = as.factor(Genotype)) %>%
#   mutate(timepoint = as.factor(timepoint), 
#          Frag_number = as.character(Frag_number)) -> health_filt
# 
# ## Getting health data into the treatment file 
# ## Removing the pre exposure disease, pre exposure dead and post exposure dead
# seq_tfall_final %>% 
#   inner_join(health_filt, 
#              by = c("Genotype", "timepoint", "Frag_number")) %>% 
#   mutate(
#     Health = case_when(
#       timepoint == "initial" & health_status == "healthy" ~ "pre_exposure",
#       timepoint == "17th_aug" & health_status == "healthy " ~ "visually_healthy",
#       timepoint == "17th_aug" & health_status == "diseased" ~ "diseased",
#       timepoint == "initial" & health_status == "diseased" ~ "pre_exposure_DISEASED",
#       timepoint == "initial" & health_status == "dead" ~ "pre_exposure_DEAD",
#       timepoint == "8th_aug" & health_status == "diseased" ~ "diseased",
#       timepoint == "9th_aug" & health_status == "diseased" ~ "diseased",
#       timepoint == "9th_aug" & health_status == "diseased " ~ "diseased", 
#       timepoint == "10th_aug" & health_status == "diseased" ~ "diseased", 
#       timepoint == "11th_aug" & health_status == "diseased" ~ "diseased", 
#       timepoint == "12th_aug" & health_status == "diseased" ~ "diseased", 
#       timepoint == "13th_aug" & health_status == "diseased" ~ "diseased", 
#       timepoint == "14th_aug" & health_status == "diseased" ~ "diseased", 
#       timepoint == "15th_aug" & health_status == "diseased" ~ "diseased", 
#       timepoint == "16th_aug" & health_status == "diseased" ~ "diseased",
#       timepoint == "9th_aug" & health_status == "dead" ~ "post_exposure_DEAD",
#       timepoint == "11th_aug" & health_status == "dead" ~ "post_exposure_DEAD",
#       timepoint == "14th_aug" & health_status == "dead" ~ "post_exposure_DEAD",
#       timepoint == "15th_aug" & health_status == "dead" ~ "post_exposure_DEAD", 
#       timepoint == "10th_aug" & health_status == "dead" ~ "post_exposure_DEAD"
#     )) %>% 
#    filter(.,
#          !grepl("pre_exposure_DISEASED|pre_exposure_DEAD|post_exposure_DEAD", Health)) -> seq_tfall_final

## Removing the placebo and healthy slurry disease samples after inoculation
# seq_tfall_final %>% 
#   filter(., grepl("placebo_slurry|s_m_placebo", D_treatment.y)) %>% 
#   filter(Health %in% "diseased") %>% 
#   column_to_rownames(var = "ID_salmon") %>%
#   rownames() -> placebo_filter
# length(placebo_filter) ## 13 samples in the placebo and healthy slurry that got diseased
# 
# seq_tfall_final %>% 
#   filter(!ID_salmon %in% placebo_filter) -> seq_tfall_final
# nrow(seq_tfall_final) ## leaves us with 297 samples
```

Removing pre exposure DEAD and post exposure DEAD leaves us with *310* samples in treatment file (10 bad samples)
Removing healthy slurry and placebo slurry diseased leaves with *297* samples in treatment file (13 bad samples)

```{r Making additional metadata variables based on health and treatment, include = F}
# seq_tfall_final %>% 
#   mutate(
#     Dif_dis_treat = case_when(
#       Health == "pre_exposure" ~ "baseline",
#       Dtreatment == "diseaseslurry" & timepoint == "17th_aug" ~ "Disease_slurry",
#       Dtreatment == "placeboslurry" & timepoint == "17th_aug" ~ "Healthy_slurry",
#       Dtreatment == "smplacebo" & timepoint == "17th_aug" ~ "SM_placebo",
#       Dtreatment == "smarcscens" & timepoint == "17th_aug" ~"Serratia",
#       Dtreatment == "diseaseslurry" & timepoint == "9th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" & timepoint == "10th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" & timepoint == "11th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" & timepoint == "12th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" & timepoint == "16th_aug" ~ "Disease_slurry",
#       Dtreatment == "smarcscens" & timepoint == "9th_aug" ~"Serratia",
#       Dtreatment == "smarcscens" & timepoint == "11th_aug" ~"Serratia",
#       Dtreatment == "smarcscens" & timepoint == "12th_aug" ~"Serratia",
#       Dtreatment == "smarcscens" & timepoint == "13th_aug" ~"Serratia",
#       Dtreatment == "smarcscens" & timepoint == "15th_aug" ~"Serratia",
#       Dtreatment == "smarcscens" & timepoint == "16th_aug" ~"Serratia",
#     )
#   ) %>% 
#   column_to_rownames(var = "ID_salmon") -> seq_tfall_final

## list to filter count matrix
# rownames(seq_tfall_final) -> treat_filt
# length(treat_filt)

## Filtering Count Matrix for filtering done to treatment file 
# DHE_counts %>% 
#   t() %>% 
#   as.data.frame() %>% 
#   rownames_to_column(var = "Samp") %>%
#   filter(Samp %in% treat_filt) %>%
#   column_to_rownames(var = "Samp") %>% 
#   t() %>% 
#   as.matrix() -> DHE_counts
# 
# ncol(DHE_counts)
# nrow(DHE_counts)
#View(DHE_counts)
```

This now gives us the count matrix with 
- 297 columns (samples)
- 29,181 rows (genes)

```{r Filtering samples with less than 1 million reads, include = F}
# DHE_counts %>%
#   t() %>%
#   as.data.frame() %>%
#   mutate(gene_counts = rowSums(DHE_counts %>%
#                                  t() %>%
#                                  as.data.frame() %>%
#                                  select(starts_with("evm")))) %>%
#   rownames_to_column(var = "Sample_number") %>%
#   select(Sample_number, gene_counts) %>%
#   arrange(gene_counts) %>%
#   filter(gene_counts < 1000000) %>% ## 15 samples have less than 1 million reads
#   column_to_rownames(var = "Sample_number") %>%
#   rownames() -> lowcount_samps
# 
# DHE_counts %>% 
#   t() %>% 
#   as.data.frame() %>% 
#   rownames_to_column(var = "Samp") %>%
#   filter(!Samp %in% lowcount_samps) %>%
#   column_to_rownames(var = "Samp") %>% 
#   t() %>% 
#   as.matrix() -> DHE_counts
# 
# ncol(DHE_counts)
# nrow(DHE_counts)
# View(DHE_counts)
```

Filtering out samples with counts less than 1,000,000. There are *15* low count samples
This leaves the count matrix with 
- 282 columns (samples)
- 29,181 genes (rows)

```{r adding variables to treatment file, include = F}
# seq_tfall_final %>% 
#   rownames_to_column(var = "ID_salmon") %>%
#   filter(!ID_salmon %in% lowcount_samps) %>% 
#   column_to_rownames(var = "ID_salmon") %>%
#   mutate(
#     Health = case_when(
#       timepoint == "initial" &
#         health_status == "healthy" ~ "pre_exposure",
#       timepoint == "17th_aug" &
#         health_status == "healthy " ~ "visually_healthy",
#       timepoint == "17th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "8th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "9th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "9th_aug" &
#         health_status == "diseased " ~ "diseased",
#       timepoint == "10th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "11th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "12th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "13th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "14th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "15th_aug" &
#         health_status == "diseased" ~ "diseased",
#       timepoint == "16th_aug" &
#         health_status == "diseased" ~ "diseased"
#     ),
#     master_time = case_when(
#       timepoint == "initial" ~ "pre_exposure",
#       timepoint == "17th_aug" ~ "Day_10",
#       timepoint == "8th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "9th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "9th_aug" &
#         health_status == "diseased " ~ "Day_1-9",
#       timepoint == "10th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "11th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "12th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "13th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "14th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "15th_aug" &
#         health_status == "diseased" ~ "Day_1-9",
#       timepoint == "16th_aug" &
#         health_status == "diseased" ~ "Day_1-9"
#     ),
#     Dif_dis_treat = case_when(
#       Health == "pre_exposure" ~ "baseline",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "17th_aug" ~ "Disease_slurry",
#       Dtreatment == "placeboslurry" &
#         timepoint == "17th_aug" ~ "Healthy_slurry",
#       Dtreatment == "smplacebo" &
#         timepoint == "17th_aug" ~ "SM_placebo",
#       Dtreatment == "smarcscens" &
#         timepoint == "17th_aug" ~ "Serratia",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "9th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "10th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "11th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "12th_aug" ~ "Disease_slurry",
#       Dtreatment == "diseaseslurry" &
#         timepoint == "16th_aug" ~ "Disease_slurry",
#       Dtreatment == "smarcscens" &
#         timepoint == "9th_aug" ~ "Serratia",
#       Dtreatment == "smarcscens" &
#         timepoint == "11th_aug" ~ "Serratia",
#       Dtreatment == "smarcscens" &
#         timepoint == "12th_aug" ~ "Serratia",
#       Dtreatment == "smarcscens" &
#         timepoint == "13th_aug" ~ "Serratia",
#       Dtreatment == "smarcscens" &
#         timepoint == "15th_aug" ~ "Serratia",
#       Dtreatment == "smarcscens" &
#         timepoint == "16th_aug" ~ "Serratia"),
#     Health_disease = paste0(Dif_dis_treat, "_", Health), 
#     geno_tank = factor(paste0(.$Genotype, .$Tank_treatment)), 
#     geno_health = factor(paste0(.$Genotype, .$Health))) -> seq_tfall_final
```

Filtering treatment file after low count samples removed, equals same as count matrix. Now need to order everything correct for actual analysis. 

```{r ordering treatment file with count table, echo = F}
# matchup <- match(rownames(seq_tfall_final), colnames(DHE_counts))
# DHE_counts  <- DHE_counts[,matchup ]
# all(rownames(seq_tfall_final) == colnames(DHE_counts))
```

```{r saving r data objects for analysis, include = F}
# save(DHE_counts, 
#      file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/DHE_counts.RData")
# save(seq_tfall_final, 
#      file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_data_objects/treatment_file.RData")
```


----------

# Gene Expression Analysis of Acropora palmata Fragment inoculated with Difffernt Disease Vectors

```{r Data object loading, include = F}
load("/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/DHE_counts.RData")
load("/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/treatment_file.RData")
load("/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_data_objects/annot_4_analysis.RData")
```

```{r PCA axes modified functions loading, include = F}
load("/Users/benjamin.d.young/Documents/projects/bioinformatc_resources/pca_axes_saves/pca23.RData")
load("/Users/benjamin.d.young/Documents/projects/bioinformatc_resources/pca_axes_saves/pca34.RData")
load("/Users/benjamin.d.young/Documents/projects/bioinformatc_resources/pca_axes_saves/pca45.RData")
load("/Users/benjamin.d.young/Documents/projects/bioinformatc_resources/pca_axes_saves/pca56.RData")
```

```{r function for fixing go terms when reading in bingo output, include = F}
pad27 <- function(x){
  str_x = as.character(x)
  digits = nchar(str_x)
  toAdd = 7-digits
  name = paste0('GO:',
                strrep(0,toAdd), 
                x)
  name
}
```

Treatment file has 282 samples 
 - Filtered out samples with < 1,000,000 reads (15 samples)
 - Filtered out pre exposure dead/diseased (xx samples)
 - Filtered out placebo samples that are diseased (xx samples)
 
This leaves for analysis 
- 282 coral samples
- 29,181 genes

```{r New Variable Columns, include = F}
## adding in grouping variable
## NB the 17th_aug healthy has a whitespace at the end, need "healthy " to make work
seq_tfall_final %>%
  mutate(
    Overall_treat = case_when(
      Health == "pre_exposure" ~ "control",
      Health == "visually_healthy" ~ "inoculated",
      Health == "diseased" ~ "inoculated"
    )
  ) -> seq_tfall_final
# View(seq_tfall_final)
```

```{r ordering tfall with count table, echo = F}
matchup <- match(rownames(seq_tfall_final), colnames(DHE_counts))
DHE_counts  <- DHE_counts[,matchup ]
all(rownames(seq_tfall_final) == colnames(DHE_counts))
```

```{r Sample summaries grouped bu different variables, echo = F}
seq_tfall_final %>% 
  group_by(Tank_treatment, D_treatment.x, Health) %>% 
  summarise(Count = n())

seq_tfall_final %>% 
  group_by(D_treatment.x, Health) %>% 
  summarise(Count = n())

# seq_tfall_final %>% 
#   group_by(Tank_treatment, Health) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/DHE_T_sample_sum.csv")

seq_tfall_final %>% 
  group_by(Genotype, D_treatment.x, Health) %>% 
  summarise(Count = n())

seq_tfall_final %>% 
  group_by(master_time) %>% 
  summarise(Count = n())
```

```{r T samples for venn with 16s samples}
seq_tfall_final %>% 
  mutate(t_samp_venn = paste0(.$Genotype, .$Health, .$Frag_number)) %>% 
  dplyr::select(t_samp_venn) -> t_sample_venn
```

```{r Venn of T samples for analysis and 16s samples for analysis, include = F}
## NB need to load in the 16s data, I have both run so it was possible, hasehed out for this reason
# venn(list(t_sample_venn$t_samp_venn, microbe_samp_venn$microbe_samp_venn))
```

```{r writing samples summaries to files, include = FALSE}
# seq_tfall_final %>% 
#   group_by(Tank_treatment, D_treatment.x, Health) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., 
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/rnaseq_sample_summaries/temp_dtreat_health.csv")
# 
# seq_tfall_final %>% 
#   group_by(D_treatment.x, Health) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., 
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/rnaseq_sample_summaries/dtreat_health.csv")
# 
# seq_tfall_final %>% 
#   group_by(Tank_treatment, Health) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., 
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/rnaseq_sample_summaries/temp_health.csv")
# 
# seq_tfall_final %>% 
#   group_by(Genotype, Health) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., 
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/rnaseq_sample_summaries/geno_health.csv")
# 
# seq_tfall_final %>% 
#   group_by(Genotype, D_treatment.x, Health) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., 
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/rnaseq_sample_summaries/geno_dtreat_health.csv")
# 
# seq_tfall_final %>% 
#   group_by(master_time) %>% 
#   summarise(Count = n()) %>% 
#   write.csv(., 
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/rnaseq_sample_summaries/pre_post_exposure.csv")
```

```{r Making DeSeq object for low gene count filtering, include=FALSE}
#making the dds model to use in deseq2
ddsall = DESeqDataSetFromMatrix(countData =  round(DHE_counts), 
                                seq_tfall_final, 
                                ~ Genotype + Dtreatment + Health)
```

```{r CPM filtering and DDS object creation}
nrow(ddsall)
ncol(ddsall)

# cpm filtering step and seeing what original VS filtered gene number left is
cccall <- counts(ddsall)
keep <- rowSums(cpm(cccall)>=2) >= 15
cccall <- cccall[keep, ]

nrow(ddsall)
ncol(ddsall)
nrow(cccall)
ncol(cccall)

cccall %>%
  as.data.frame()  -> cccall
  
ddsall <- DESeqDataSetFromMatrix(countData = cccall, 
                                 colData = seq_tfall_final, 
                                 design = ~ Genotype + Health)
```

This filtering leaves with 
- 16,950 genes

```{r custom annotation file for bingo GO analysis using cccall and annot file, include = F}
annot_4_analysis %>%
  filter(Count_ID %in% rownames(cccall)) -> annot_filtered_2_15
# View(annot_filtered_2_15)

annot_filtered_2_15 %>%
  dplyr::select(Count_ID, GO_Terms_Swiss.Prot) %>%
  tidyr::drop_na() %>%
  dplyr::filter(!GO_Terms_Swiss.Prot %in% c("0")) %>%
  tidyr::separate_rows(GO_Terms_Swiss.Prot) %>%
  filter(!str_detect(GO_Terms_Swiss.Prot, 'GO')) %>%
  unite(x, c(Count_ID, GO_Terms_Swiss.Prot), sep = " = ", remove = T) -> DHE_CAF_GO
# View(DHE_CAF_GO)
# write.table(DHE_CAF_GO,
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/bingo_cafgo_apal_v3.1_filt_all_samps.txt",
#             quote=F,
#             row.names=F,
#             col.names=F)


annot_4_analysis %>%
  dplyr::select(Count_ID, GO_Terms_Swiss.Prot) %>%
  tidyr::drop_na() %>%
  dplyr::filter(!GO_Terms_Swiss.Prot %in% c("0")) %>%
  tidyr::separate_rows(GO_Terms_Swiss.Prot) %>%
  filter(!str_detect(GO_Terms_Swiss.Prot, 'GO')) %>%
  unite(x, c(Count_ID, GO_Terms_Swiss.Prot), sep = " = ", remove = T) -> DHE_CAF_GO_all
# write.table(DHE_CAF_GO_all,
#             file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/bingo_cafgo_apal_v3.1_all.txt",
#             quote=F,
#             row.names=F,
#             col.names=F)

```

N.B. Need to add in the top line manually for the CAFGO file to work in Bingo. 

In total we have 
 - 282 samples
 - 16,950 genes 


## Principal Component Analysis of Genet
### With genet variacne included

```{r VST, include=FALSE}
vsdall <- vst(ddsall, blind=FALSE)
```

```{r Formating VST for PCAtools so it works the same as plotPCA function, include = F}
PCA_tools_all <- assay(vsdall)
rv <- rowVars(PCA_tools_all)
select <- order(rv,
                decreasing = TRUE)[seq_len(min(500,
                                               length(rv)))]
allsamps <- pca(PCA_tools_all[select, ],
                metadata = seq_tfall_final, 
                removeVar = 0.10)
```

```{r Plots from PCAtools, echo = F, fig.width=7, fig.height=4}
## Plot showing the genes which drive the principal component variances. 
plotloadings(allsamps, labSize = 3)
```

```{r EigenPlots, echo = F, fig.width=13, fig.height = 7}
# Plotting of the metadata to the PC axes to see which one has strong significant relationships with axes. 
eigencorplot(allsamps,
             components = getComponents(allsamps, 1:10),
    metavars = c("Health", "Dif_dis_treat", "Genotype", "Tank_treatment", "Dtreatment", 
                 "Recovery_tank", "ERL_tank_number", "D_touch_tank", "timepoint", "geno_tank", "geno_health"), 
    col = c('darkblue', 'blue2', 'black', 'red2', 'darkred'),
    cexCorval = 0.7,
    colCorval = 'white',
    fontCorval = 2,
    posLab = 'bottomleft',
    rotLabX = 45,
    posColKey = 'top',
    cexLabColKey = 1.5,
    scale = TRUE,
    main = 'PC1 - 11, Metadata Correlations',
    colFrame = 'white',
    plotRsquared = FALSE)

eigencorplot(allsamps,
             components = getComponents(allsamps, 1:11),
    metavars = c("Health", "Dif_dis_treat", "Genotype", "Tank_treatment", "Dtreatment", 
                 "Recovery_tank", "ERL_tank_number", "D_touch_tank", "timepoint", "geno_tank", "geno_health"), 
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 1.2,
    fontCorval = 2,
    posLab = 'all',
    rotLabX = 45,
    scale = TRUE,
    main = bquote(Principal~Component~Pearson~r^2~metadata~significant~correlation),
    plotRsquared = T,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'BH',
    signifSymbols = c('****', '***', '**', '*', ''),
    signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
    )
```

```{r Nice correlation matrix plot including genet variance, fig.width=13, fig.height = 4}
eigencorplot(allsamps,
             components = getComponents(allsamps, 1:11),
    metavars = c("Health", "Health_disease", "Genotype", "Tank_treatment"), 
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 1.2,
    fontCorval = 2,
    posLab = 'all',
    rotLabX = 45,
    scale = TRUE,
    main = bquote(Principal~Component~Pearson~r^2~metadata~significant~correlation),
    plotRsquared = T,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'BH',
    signifSymbols = c('****', '***', '**', '*', ''),
    signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
    )

eigencorplot(allsamps,
             components = getComponents(allsamps, 1:11),
    metavars = c("Genotype", "Health"), 
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 1.2,
    fontCorval = 2,
    posLab = 'all',
    rotLabX = 45,
    scale = TRUE,
    main = bquote(Principal~Component~Pearson~r^2~metadata~significant~correlation),
    plotRsquared = T,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'BH',
    signifSymbols = c('****', '***', '**', '*', ''),
    signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
    )
```

```{r PCA of the Samples, echo = F}
## not transposing the CLR for the PCA
pca_samp <- prcomp(t(PCA_tools_all)[,select])
sample_loadings <- as.data.frame(pca_samp$x)
summary(pca_samp)
```

```{r PC axis dataframes for plotting in ggplot, include = F}
pca12 <-
  plotPCA(
    vsdall,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Health_disease",
      "Genotype",
      "Dif_dis_treat",
      "timepoint",
      "master_time"
    ),
    returnData = TRUE
  )
pca23 <-
  pcaaxes23(
    vsdall,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Health_disease",
      "Genotype",
      "Dif_dis_treat",
      "timepoint",
      "master_time"
    ),
    returnData = TRUE
  )
pca34 <-
  pcaaxes34(
    vsdall,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Health_disease",
      "Genotype",
      "Dif_dis_treat",
      "timepoint",
      "master_time"
    ),
    returnData = TRUE
  )
pca45 <-
  pcaaxes45(
    vsdall,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Health_disease",
      "Genotype",
      "Dif_dis_treat",
      "timepoint",
      "master_time"
    ),
    returnData = TRUE
  )
pca56 <-
  pcaaxes56(
    vsdall,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Health_disease",
      "Genotype",
      "Dif_dis_treat",
      "timepoint",
      "master_time"
    ),
    returnData = TRUE
  )
```

From the correlation analysis what I should look at is 
Health - PC1, PC3, PC4
Genet - PC2 - PC6
Health and Disease inoculation - PC3 and PC4

```{r DeSeq2 PCAs for % loadings, echo = F}
plotPCA(vsdall, intgroup=c("Tank_treatment"), returnData = F)
pcaaxes23(vsdall, intgroup=c("Tank_treatment"), returnData = F)
pcaaxes34(vsdall, intgroup=c("Tank_treatment"), returnData = F)
pcaaxes45(vsdall, intgroup=c("Tank_treatment"), returnData = F)
pcaaxes56(vsdall, intgroup=c("Tank_treatment"), returnData = F)
```

```{r Principal component 1 plots, echo = F}
ggplot(pca12, aes(PC1, PC2, color = Genotype)) +
  geom_point(size = 3) +  xlab(paste0("PC1 27% variance")) +
  ylab(paste0("PC2 13% variance")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  )  +
  theme(legend.key.size = unit(0.7, "cm")) +
  stat_ellipse()

ggplot(pca12, aes(PC1, PC2, color = Health)) +
  geom_point(size = 3) +  xlab(paste0("PC1 27% variance")) +
  ylab(paste0("PC2 13% variance")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  )  +
  theme(legend.key.size = unit(0.7, "cm")) +
  stat_ellipse()
```

```{r Princiapl component 2 to 6 for genet, echo = F}
ggplot(pca23, aes(PC2, PC3, colour=Genotype)) + 
  geom_point(size=3) +  
  xlab(paste0("PC2 13% variance")) + 
  ylab(paste0("PC3 8% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse() + 
  scale_color_manual(name = "Genotype",
                     values = c("wheat3", "navy" ,"springgreen3", "grey60"))

ggplot(pca34, aes(PC3, PC4, colour=Genotype)) + 
  geom_point(size=3) +  
  xlab(paste0("PC3 8% variance")) + 
  ylab(paste0("PC4 8% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()

ggplot(pca45, aes(PC4, PC5, colour=Genotype)) + 
  geom_point(size=3) +  
  xlab(paste0("PC4 8% variance")) + 
  ylab(paste0("PC5 6% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()

ggplot(pca56, aes(PC5, PC6, colour=Genotype)) + 
  geom_point(size=3) +  
  xlab(paste0("PC5 6% variance")) + 
  ylab(paste0("PC6 4% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()
```

```{r BL and Dtreats PCA, echo = F}
ggplot(pca12, aes(PC1, PC2, color = Health_disease)) +
  geom_point(size = 3) +  
  xlab(paste0("PC1 28% variance")) +
  ylab(paste0("PC2 13% variance")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8)
  )  +
  theme(legend.key.size = unit(0.7, "cm")) +
  stat_ellipse()

ggplot(pca23, aes(PC2, PC3, colour = Health_disease)) + 
  geom_point(size=3) +  
  xlab(paste0("PC2 13% variance")) + 
  ylab(paste0("PC3 8% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()

ggplot(pca34, aes(PC3, PC4, colour = Health_disease)) + 
  geom_point(size=3) +  
  xlab(paste0("PC3 8% variance")) + 
  ylab(paste0("PC4 8% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()

ggplot(pca45, aes(PC4, PC5, colour = Health_disease)) + 
  geom_point(size=3) +  
  xlab(paste0("PC4 8% variance")) + 
  ylab(paste0("PC5 6% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()

ggplot(pca56, aes(PC5, PC6, colour = Health_disease)) + 
  geom_point(size=3) +  
  xlab(paste0("PC5 6% variance")) + 
  ylab(paste0("PC6 4% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()
```

### Principal Component Analysis Removing Genet Variance

```{r VST with limma removing genet variance, include = F}
vsdgeno <- vst(ddsall)
assay(vsdgeno) <- limma::removeBatchEffect(assay(vsdgeno), 
                                           vsdgeno$Genotype)
```

```{r using VST and making it behave the same as plotPCA function, include = F}
PCA_tools_all <- assay(vsdgeno)
rv <- rowVars(PCA_tools_all)
select <- order(rv, decreasing = TRUE)[seq_len(min(500, length(rv)))]
allsamps <- pca(PCA_tools_all[select,], 
                metadata = seq_tfall_final, 
                removeVar = 0.10)
```

```{r Plots from PCAtools, echo = F, fig.width=12, fig.height=6}
## Plot showing the samples with the genes which drive principal components
plotloadings(allsamps, labSize = 3)
```

```{r EigenPlots, echo = F, fig.width=13, fig.height = 7}
# Plotting of the metadata to the PC axes to see which one has strong significant relationships with axes. 
eigencorplot(allsamps,
             components = getComponents(allsamps, 1:10),
    metavars = c("Health", "Dif_dis_treat", "Genotype", "Tank_treatment", "Dtreatment", "Genotype", 
                 "Recovery_tank", "ERL_tank_number", "D_touch_tank", "timepoint", "geno_tank", "geno_health", 
      "master_time"), 
    col = c('darkblue', 'blue2', 'black', 'red2', 'darkred'),
    cexCorval = 0.7,
    colCorval = 'white',
    fontCorval = 2,
    posLab = 'bottomleft',
    rotLabX = 45,
    posColKey = 'top',
    cexLabColKey = 1.5,
    scale = TRUE,
    main = 'PC1 - 11, Metadata Correlations',
    colFrame = 'white',
    plotRsquared = FALSE)

eigencorplot(allsamps,
             components = getComponents(allsamps, 1:11),
    metavars = c("Health", "Dif_dis_treat", "Genotype", "Tank_treatment", "Dtreatment", "Genotype", 
                 "Recovery_tank", "ERL_tank_number", "D_touch_tank", "timepoint", "geno_tank", "geno_health", 
      "master_time"), 
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 1.2,
    fontCorval = 2,
    posLab = 'all',
    rotLabX = 45,
    scale = TRUE,
    main = bquote(Principal~Component~Pearson~r^2~metadata~significant~correlation),
    plotRsquared = T,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'BH',
    signifSymbols = c('****', '***', '**', '*', ''),
    signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
    )
```

```{r Nice correlation matrix plot with genet variacne removed, echo = F, fig.width=13, fig.height = 4}
eigencorplot(allsamps,
             components = getComponents(allsamps, 1:11),
    metavars = c("Health", "Health_disease", "Genotype", "Tank_treatment"), 
    col = c('white', 'cornsilk1', 'gold', 'forestgreen', 'darkgreen'),
    cexCorval = 1.2,
    fontCorval = 2,
    posLab = 'all',
    rotLabX = 45,
    scale = TRUE,
    main = bquote(Principal~Component~Pearson~r^2~metadata~significant~correlation),
    plotRsquared = T,
    corFUN = 'pearson',
    corUSE = 'pairwise.complete.obs',
    corMultipleTestCorrection = 'BH',
    signifSymbols = c('****', '***', '**', '*', ''),
    signifCutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1)
    )
```

```{r PCA of the Samples, include = F}
## not transposing the CLR for the PCA
pca_samp <- prcomp(t(PCA_tools_all)[,select])
sample_loadings <- as.data.frame(pca_samp$x)
summary(pca_samp)
```

```{r PC axis dataframes with genet removed, include = F}
pca12_m3 <-
  plotPCA(
    vsdgeno,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Genotype",
      "Disease_tank",
      "Dif_dis_treat",
      "timepoint", 
      "master_time", 
      "Overall_treat", 
      "Health_disease"
    ),
    returnData = TRUE
  )
pca23_m3 <-
  pcaaxes23(
    vsdgeno,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Genotype",
      "Disease_tank",
      "Dif_dis_treat",
      "timepoint", 
      "master_time", 
      "Overall_treat", 
      "Health_disease"
    ),
    returnData = TRUE
  )
pca34_m3 <-
  pcaaxes34(
    vsdgeno,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Genotype",
      "Disease_tank",
      "Dif_dis_treat",
      "timepoint", 
      "master_time", 
      "Overall_treat", 
      "Health_disease"
    ),
    returnData = TRUE
  )
pca45_m3 <-
  pcaaxes45(
    vsdgeno,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Genotype",
      "Disease_tank",
      "Dif_dis_treat",
      "timepoint", 
      "master_time", 
      "Overall_treat", 
      "Health_disease"
    ),
    returnData = TRUE
  )
pca56_m3 <-
  pcaaxes56(
    vsdgeno,
    intgroup = c(
      "Dtreatment",
      "Tank_treatment",
      "Health",
      "Genotype",
      "Disease_tank",
      "Dif_dis_treat",
      "timepoint", 
      "master_time", 
      "Overall_treat", 
      "Health_disease"
    ),
    returnData = TRUE
  )
```

```{r DeSeq2 PCA for axes loadings, echo = F}
plotPCA(vsdgeno, intgroup=c("Health"), returnData = F)
pcaaxes23(vsdgeno, intgroup=c("Health"), returnData = F)
pcaaxes34(vsdgeno, intgroup=c("Health"), returnData = F)
pcaaxes45(vsdgeno, intgroup=c("Health"), returnData = F)
pcaaxes56(vsdgeno, intgroup=c("Health"), returnData = F)
```

```{r Temperature treatment PCAs, fig.height=4, fig.width=7, echo = F}
ggplot(pca12_m3, aes(PC1, PC2, color = Tank_treatment, shape = Health)) +
  geom_point(size = 2) +  
  xlab(paste0("PC1 33% variance")) +
  ylab(paste0("PC2 10% variance")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  )  +
  theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC1, PC2, group = Tank_treatment), type = "norm") +
    scale_color_manual(name = "Tank_treatment",
                     values = c("blue3", "orangered2"))
```

```{r Health PCA, echo = F}
cbPalette <- c("dodgerblue3" ,"darkgoldenrod2", "orangered3")

ggplot(pca12_m3, aes(PC1, PC2, color = Health, shape = master_time)) +
  geom_point(size = 2) +  
  xlab(paste0("PC1 33% variance")) +
  ylab(paste0("PC2 10% variance")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  )  +
  theme(legend.key.size = unit(0.8, "cm")) +
  stat_ellipse(aes(PC1, PC2, group = Health), type = "norm") +
    scale_color_manual(name = "Health",
                     values = c("orangered3", "dodgerblue3" ,"darkgoldenrod2"))

ggplot(pca23_m3, aes(PC2, PC3, colour=Health, shape = master_time)) + 
  geom_point(size=3) +  
  xlab(paste0("PC2 10% variance")) + 
  ylab(paste0("PC3 6% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse(aes(PC2, PC3, group = Health), type = "norm") +
    scale_color_manual(name = "Health",
                     values = c("orangered3", "dodgerblue3" ,"darkgoldenrod2"))

# ggplot(pca34_m3, aes(PC3, PC4, colour=Health)) + 
#   geom_point(size=3) +  
#   xlab(paste0("PC3 7% variance")) + 
#   ylab(paste0("PC4 6% variance")) + 
#   theme(text = element_text(size=11, family = "Arial"), 
#         legend.position = "right", 
#         panel.background = element_rect(fill = "transparent"), 
#         axis.line = element_line(colour = "black"), 
#         legend.key = element_rect(fill = "transparent"), 
#         axis.text = element_text(size = 8)) + 
#   stat_ellipse()
# 
# ggplot(pca45_m3, aes(PC4, PC5, colour=Health)) + 
#   geom_point(size=3) +  
#   xlab(paste0("PC4 6% variance")) + 
#   ylab(paste0("PC5 4% variance")) + 
#   theme(text = element_text(size=11, family = "Arial"), 
#         legend.position = "right", 
#         panel.background = element_rect(fill = "transparent"), 
#         axis.line = element_line(colour = "black"), 
#         legend.key = element_rect(fill = "transparent"), 
#         axis.text = element_text(size = 8)) + 
#   stat_ellipse()
# 
# ggplot(pca56_m3, aes(PC5, PC6, colour=Health)) + 
#   geom_point(size=3) +  
#   xlab(paste0("PC5 4% variance")) + 
#   ylab(paste0("PC6 3% variance")) + 
#   theme(text = element_text(size=11, family = "Arial"), 
#         legend.position = "right", 
#         panel.background = element_rect(fill = "transparent"), 
#         axis.line = element_line(colour = "black"), 
#         legend.key = element_rect(fill = "transparent"), 
#         axis.text = element_text(size = 8)) + 
#   stat_ellipse()
```

```{r health and disease inoc PCA with genet variance removed, fig.width=7, fig.height=4}
# ggplot(pca12_m3, aes(PC1, PC2, color = Health_disease)) +
#   geom_point(size = 3) +  
#   xlab(paste0("PC1 33% variance")) +
#   ylab(paste0("PC2 10% variance")) +
#   theme(
#     text = element_text(size = 12, family = "Arial"),
#     legend.position = "right",
#     panel.background = element_rect(fill = "transparent"),
#     axis.line = element_line(colour = "black"),
#     legend.key = element_rect(fill = "transparent"),
#     axis.text = element_text(size = 9)
#   )  +
#   theme(legend.key.size = unit(0.9, "cm")) + 
#   stat_ellipse()
# 
# ggplot(pca23_m3, aes(PC2, PC3, colour = Health_disease)) + 
#   geom_point(size=3) +  
#   xlab(paste0("PC2 10% variance")) + 
#   ylab(paste0("PC3 6% variance")) + 
#   theme(text = element_text(size=11, family = "Arial"), 
#         legend.position = "right", 
#         panel.background = element_rect(fill = "transparent"), 
#         axis.line = element_line(colour = "black"), 
#         legend.key = element_rect(fill = "transparent"), 
#         axis.text = element_text(size = 8)) + 
#   stat_ellipse()
# 
# ggplot(pca34_m3, aes(PC3, PC4, colour=Health_disease)) + 
#   geom_point(size=3) +  
#   xlab(paste0("PC3 6% variance")) + 
#   ylab(paste0("PC4 4% variance")) + 
#   theme(text = element_text(size=11, family = "Arial"), 
#         legend.position = "right", 
#         panel.background = element_rect(fill = "transparent"), 
#         axis.line = element_line(colour = "black"), 
#         legend.key = element_rect(fill = "transparent"), 
#         axis.text = element_text(size = 8)) + 
#   stat_ellipse()
# 
# ggplot(pca45_m3, aes(PC4, PC5, colour=Health_disease)) + 
#   geom_point(size=3) +  
#   xlab(paste0("PC4 4% variance")) + 
#   ylab(paste0("PC5 4% variance")) + 
#   theme(text = element_text(size=11, family = "Arial"), 
#         legend.position = "right", 
#         panel.background = element_rect(fill = "transparent"), 
#         axis.line = element_line(colour = "black"), 
#         legend.key = element_rect(fill = "transparent"), 
#         axis.text = element_text(size = 8)) + 
#   stat_ellipse()
# 
# ggplot(pca56_m3, aes(PC5, PC6, colour=Health_disease)) + 
#   geom_point(size=3) +  
#   xlab(paste0("PC5 4% variance")) + 
#   ylab(paste0("PC6 3% variance")) + 
#   theme(text = element_text(size=11, family = "Arial"), 
#         legend.position = "right", 
#         panel.background = element_rect(fill = "transparent"), 
#         axis.line = element_line(colour = "black"), 
#         legend.key = element_rect(fill = "transparent"), 
#         axis.text = element_text(size = 8)) + 
#   stat_ellipse()
```

```{r BL and Dtreats PCA, echo = F}
ggplot(pca12_m3, aes(PC1, PC2, color = Dif_dis_treat, shape = Health)) +
  geom_point(size = 2) +  xlab(paste0("PC1 33% variance")) +
  ylab(paste0("PC2 10% variance")) +
  theme(
    text = element_text(size = 12, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(colour = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 9)
  )  +
  theme(legend.key.size = unit(0.8, "cm")) + 
  stat_ellipse(aes(PC1, PC2, group = paste0(pca12_m3$Dif_dis_treat, pca12_m3$Health), type = "norm")) + 
  scale_color_manual(values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3"))

ggplot(pca23_m3, aes(PC2, PC3, colour = Dif_dis_treat, shape = Health)) + 
  geom_point(size=3) +  
  xlab(paste0("PC2 13% variance")) + 
  ylab(paste0("PC3 10% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse() +
  scale_color_manual(values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3"))

ggplot(pca34_m3, aes(PC3, PC4, colour = Dif_dis_treat)) + 
  geom_point(size=3) +  
  xlab(paste0("PC3 10% variance")) + 
  ylab(paste0("PC4 8% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse() +
  scale_color_manual(values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3"))

ggplot(pca45_m3, aes(PC4, PC5, colour = Dif_dis_treat)) + 
  geom_point(size=3) +  
  xlab(paste0("PC4 8% variance")) + 
  ylab(paste0("PC5 6% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()+
  scale_color_manual(values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3"))

ggplot(pca56_m3, aes(PC5, PC6, colour = Dif_dis_treat)) + 
  geom_point(size=3) +  
  xlab(paste0("PC5 6% variance")) + 
  ylab(paste0("PC6 5% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8)) + 
  stat_ellipse()+
  scale_color_manual(values = c("dodgerblue3", "aquamarine3", "gold3", "plum1", "lightsteelblue3"))
```


## Genet LRT, DEGpatterns, KEGG, and GO analyses
### DeSeq2 LRT Analysis

```{r Differential Exprssion analysis for genet using LRT methodology, include = F}
ddslrt <- DESeqDataSetFromMatrix(countData = cccall, 
                                 colData = seq_tfall_final, 
                                 design = ~ Genotype)

ddslrt <- DESeq(ddslrt, 
                test="LRT", 
                reduced = ~ 1)

lrt_results <- results(ddslrt, 
                       alpha = 0.001)
```

```{r LRT results at alpha 0.001, echo = F}
summary(lrt_results)
```

```{r Getting results with annotation, echo = F}
as.data.frame(lrt_results) %>%
  na.exclude() %>%
  dplyr::filter(padj <= 0.001) %>% 
  rownames_to_column(var = "Count_ID") %>% 
  inner_join(annot_filtered_2_15 %>% 
               dplyr::select(Count_ID, Gene.Annotation, 6:21)) -> LRT_geno_0.001

# write.csv(LRT_geno_0.001,
#           file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/LRT_geno_0.001.csv")

LRT_geno_0.001 %>%
  rownames() -> filt_lrt
```


### Genet KEGG analysis with LRT genes as input

```{r running KEGG analysis, include = T}
LRT_geno_0.001_kegg <- enrichKEGG(gene = LRT_geno_0.001$KEGG.ID..threshold.30, 
                                  organism = 'ko', 
                                  pvalueCutoff = 0.01)
```

```{r making kegg term and genes in kegg terms lists, include = F}
LRT_geno_0.001_kegg %>% 
  as.data.frame() -> LRT_geno_0.001_kegg_df

as.data.frame(LRT_geno_0.001_kegg) %>%
  separate_rows(geneID, sep = '\\/', convert = T) %>%
  dplyr::rename(., pvalue_KEGG = pvalue) %>%
  inner_join(LRT_geno_0.001,
             by = c("geneID" = "KEGG.ID..threshold.30")) -> LRT_geno_0.001_kegg_genes

# write.csv(LRT_geno_0.001_kegg_df ,
#           file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/LRT_geno_0.01_KEGG.csv")
# write.csv(LRT_geno_0.001_kegg_genes,
#           file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/LRT_geno_0.01_KEGG_genes.csv")
```



### Genet GO Visulaisation of Results

GO enrichment analysis was run in cytoscape and the plugin app Bingo. 
Parameters
- FDR correction
- alpha 0.05


```{r reading in bingo results for genet go enrichment, include = F}
## for the skip need to change for all bingo files when reading in to r
## probably a way to automate but I am lazy so ye. 
read.table(
  file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/bingo/lrt_geno_0.01_apal.bgo", 
  skip = 51,  
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>% 
#  dplyr::select(GO.ID, Description, Genes.in.test.set) %>% View()
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::inner_join(annot_filtered_2_15 %>% 
               dplyr::select(Count_ID, Gene.Annotation) %>% 
                 dplyr::mutate(Count_ID = toupper(Count_ID)), 
             by = c("Genes.in.test.set" = "Count_ID")) %>% 
  dplyr::rename(., pvalue_GO = p.value) %>%
  inner_join(LRT_geno_0.001 %>%
               mutate(Genes.in.test.set = toupper(Count_ID))) %>%
  dplyr::select(Genes.in.test.set, Gene.Annotation, GO.ID_FULL, 2:8, 19, 13:18) -> LRT_geno_go_res_gene_0.01
# View(LRT_geno_go_res_gene_0.01)
```

```{r writing files for Supplementary}
# read.table(
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/bingo/lrt_geno_0.01_apal.bgo", 
#   skip = 51,  
#   stringsAsFactors = F,
#   sep = "\t",
#   header = T
# ) %>% 
#   write.csv(., 
#             file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/LRT_genet_go_res_0.01.csv")
# 
# LRT_geno_go_res_gene_0.01 %>% 
#   write.csv(., 
#             file = "/Users/benyoung/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/genet_go_res_genes_0.01.csv")
```

From the genet GO enrichment analysis immune system terms are enriched. So plotting some heatmaps for these to include in figures and supplementary. 

```{r making immune gene datframes for supp and for heatmap plotting, include = F}
LRT_geno_go_res_gene_0.01 %>% 
  dplyr::filter(GO.ID_FULL %in% c("GO:0050776", "GO:0006955", "GO:0045087")) %>% 
  dplyr::select(Gene.Annotation, Description, Genes.in.test.set) %>% 
  dplyr::mutate(membership = 1) %>%
  tidyr::spread(Description, membership, fill = 0) %>% 
  mutate(Count_ID = Genes.in.test.set) %>%
  inner_join(LRT_geno_0.001 %>% 
               mutate(Count_ID = toupper(Count_ID)), 
             by = c("Count_ID" = "Count_ID")) %>% 
  arrange(padj) -> LRT_geno_immune_results

LRT_geno_immune_results %>% 
  slice_min(padj, n = 75) -> Immune_for_heat_map
```


#### Immune Gene Genet Heatmap All genes

```{r GO Terms We Want, include = F}
wantGO <- c("GO:0050776", "GO:0006955", "GO:0045087")

LRT_geno_go_res_gene_0.01 %>% 
  dplyr::filter(Genes.in.test.set %in% LRT_geno_immune_results$Genes.in.test.set) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) %>% 
  dplyr::rename(Count_ID = Genes.in.test.set) -> intGOterms 
forheatmap <- as.character(intGOterms$Count_ID)

intGOterms %>% 
  dplyr::select(Count_ID, Gene.Annotation) -> LRT_immune_genes
```

```{r Genet immune heatmap prep work, include = F}
assay(vsdall) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "genes") %>% 
  mutate(genes = toupper(genes)) %>% 
  column_to_rownames(var = "genes") %>% 
  as.matrix() -> assayed_vsdall

assayed_vsdall[forheatmap,] -> matmatnorder
#View(matmatnorder)

# Colours for the GENETS
ccann <- data.frame(seq_tfall_final$Genotype)
colnames(ccann) <- c("Genotype")
colcol <- list("Genotype" = c("ML2" = "#C77CFF", "CN2" = "#F8766D", "HS1" = "#00BFC4", "CN4" = "#7CAE00"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(0.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 10),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

annot_filtered_2_15 %>% 
  dplyr::select(Count_ID, Gene.Annotation) %>% 
  mutate(Count_ID = toupper(Count_ID)) %>% 
  right_join(matmatnorder %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Count_ID")) %>% 
  column_to_rownames(var="Count_ID") %>%
  tidyr::drop_na() -> matmatnorder
```

```{r genet GO term heatmap prep, include = FALSE}
# GO matrix (yes no) for genes matching to a term
intGOterms %>%
  dplyr::select(Gene.Annotation, Description, Count_ID) %>%
  dplyr::mutate(membership = 1) %>%
  tidyr::spread(Description, membership, fill = 0) -> ML2_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
matmatnorder %>%
  rownames_to_column(var="Count_ID") %>%
  inner_join(ML2_GO2gene_matrix) %>%
  column_to_rownames(var="Count_ID") -> matmatnorder
#View(matmatnorder)

# Changing names to a character
matmatnorder$Gene.Annotation <- as.character(matmatnorder$Gene.Annotation)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% 
                                         as.data.frame() %>% 
                                         dplyr::select(Gene.Annotation)))
```

```{r PC1 Variance Heatmaps, fig.height = 20, fig.width = 8}
# View(matmatnorder)
# ncol(matmatnorder)

# NB manual specify of the GO columns, here it is 86:93. First heatmap we remove, 2nd heatmap we select. 
HM1 <-
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        "Gene.Annotation", 284:286
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 4,
    row_km = 10,
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    column_gap = unit(0.1, "cm"),
    row_gap = unit(0.1, "cm"),
    column_dend_height = unit(2, "cm"),
    row_dend_width = unit(8, "cm"),
    show_row_names = F,
    row_title_gp = gpar(fontsize = 7),
    column_title_gp = gpar(fontsize = 7),
    show_column_names = F
  )

#GO yes no heatmap
HM2 <-
  Heatmap(
    t(t(
      matmatnorder %>% as.data.frame() %>% dplyr::select(c(284:286))
    )),
    show_row_dend = F,
    show_column_dend = F,
    show_column_names = T,
    show_row_names = F,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$Gene.Annotation,  gp = gpar(fontsize = 5, font = 10)
    )),
    column_names_gp = gpar(fontsize = 7, font = 1),
    col = c("white", "black"),
    column_names_rot = 30, 
    heatmap_width = unit(8, "cm"), 
    heatmap_height = unit(28, "cm")
  )

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(0.3, "cm"))
```

### Immune gene LRT genet top 75 

```{r GO Terms We Want}
wantGO <- c("GO:0050776", "GO:0006955", "GO:0045087")

LRT_geno_go_res_gene_0.01 %>% 
  dplyr::filter(Genes.in.test.set %in% Immune_for_heat_map$Genes.in.test.set) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) %>% 
  dplyr::rename(Count_ID = Genes.in.test.set) -> intGOterms # need to do this to just get each gene name once, doesnt really matter LFC for heatmap. 
forheatmap <- as.character(intGOterms$Count_ID)

intGOterms %>% 
  dplyr::select(Count_ID, Gene.Annotation) -> LRT_immune_genes
```

```{r ML2 Main Heatmap Prep, include = F}
assay(vsdall) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "genes") %>% 
  mutate(genes = toupper(genes)) %>% 
  column_to_rownames(var = "genes") %>% 
  as.matrix() -> assayed_vsdall

assayed_vsdall[forheatmap,] -> matmatnorder
#View(matmatnorder)

# Colours for the GENETS
ccann <- data.frame(seq_tfall_final$Genotype)
colnames(ccann) <- c("Genotype")
colcol <- list("Genotype" = c("ML2" = "grey60", "CN2" = "wheat3", "HS1" = "springgreen3", "CN4" = "navy"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(0.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 10),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

annot_filtered_2_15 %>% 
  dplyr::select(Count_ID, Gene.Annotation) %>% 
  mutate(Count_ID = toupper(Count_ID)) %>% 
  right_join(matmatnorder %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Count_ID")) %>% 
  column_to_rownames(var="Count_ID") %>%
  tidyr::drop_na() -> matmatnorder
```

```{r PC1 variance complexheatmap prep}
# GO matrix (yes no) for genes matching to a term
intGOterms %>%
  dplyr::select(Gene.Annotation, Description, Count_ID) %>%
  dplyr::mutate(membership = 1) %>%
  tidyr::spread(Description, membership, fill = 0) -> ML2_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
matmatnorder %>%
  rownames_to_column(var="Count_ID") %>%
  inner_join(ML2_GO2gene_matrix) %>%
  column_to_rownames(var="Count_ID") -> matmatnorder
#View(matmatnorder)

# Changing names to a character
matmatnorder$Gene.Annotation <- as.character(matmatnorder$Gene.Annotation)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% 
                                         as.data.frame() %>% 
                                         dplyr::select(Gene.Annotation)))
```

```{r PC1 Variance Heatmaps, fig.height = 15, fig.width = 10}
# View(matmatnorder)
# ncol(matmatnorder)

# NB manual specify of the GO columns, here it is 86:93. First heatmap we remove, 2nd heatmap we select. 
HM1 <-
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        "Gene.Annotation", 284:286
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 4,
    row_km = 4,
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    column_gap = unit(0.2, "cm"),
    row_gap = unit(0.2, "cm"),
    column_dend_height = unit(3, "cm"),
    row_dend_width = unit(8, "cm"),
    show_row_names = F,
    row_title_gp = gpar(fontsize = 7),
    column_title_gp = gpar(fontsize = 7),
    show_column_names = F
  )

#GO yes no heatmap
HM2 <-
  Heatmap(
    t(t(
      matmatnorder %>% as.data.frame() %>% dplyr::select(c(284:286))
    )),
    show_row_dend = F,
    show_column_dend = F,
    show_column_names = T,
    show_row_names = F,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$Gene.Annotation,  gp = gpar(fontsize = 8, font = 10)
    )),
    column_names_gp = gpar(fontsize = 7, font = 1),
    col = c("white", "black"),
    column_names_rot = 30, 
    heatmap_width = unit(12, "cm"), 
    heatmap_height = unit(28, "cm")
  )

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(0.5, "cm"))
```


### Biosynthetic and metabolic Genes LRT Geno 

```{r GO Terms We Want for metabolic processes, include = F}
wantGO <- c("GO:0009311", "GO:0005984", "GO:0016137", "GO:0010126", "GO:0010125", "GO:0016138")

LRT_geno_go_res_gene_0.01 %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) %>% 
  dplyr::rename(Count_ID = Genes.in.test.set) -> intGOterms # need to do this to just get each gene name once, doesnt really matter LFC for heatmap. 
forheatmap <- as.character(intGOterms$Count_ID)

intGOterms %>% 
  dplyr::select(Count_ID, Gene.Annotation) -> LRT_metabolic_genes
```

```{r ML2 Main Heatmap Prep for metbolic, include = F}
assay(vsdall) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "genes") %>% 
  mutate(genes = toupper(genes)) %>% 
  column_to_rownames(var = "genes") %>% 
  as.matrix() -> assayed_vsdall

assayed_vsdall[forheatmap,] -> matmatnorder
#View(matmatnorder)

# Colours for the GENETS
ccann <- data.frame(seq_tfall_final$Genotype)
colnames(ccann) <- c("Genotype")
colcol <- list("Genotype" = c("ML2" = "#C77CFF", "CN2" = "#F8766D", "HS1" = "#00BFC4", "CN4" = "#7CAE00"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(0.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 10),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

annot_filtered_2_15 %>% 
  dplyr::select(Count_ID, Gene.Annotation) %>% 
  mutate(Count_ID = toupper(Count_ID)) %>% 
  right_join(matmatnorder %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Count_ID")) %>% 
  column_to_rownames(var="Count_ID") %>%
  tidyr::drop_na() -> matmatnorder
```

```{r PC1 variance complexheatmap prep metabolic pathways, include = F}
# GO matrix (yes no) for genes matching to a term
intGOterms %>%
  dplyr::select(Gene.Annotation, Description, Count_ID) %>%
  dplyr::mutate(membership = 1) %>%
  tidyr::spread(Description, membership, fill = 0) -> ML2_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
matmatnorder %>%
  rownames_to_column(var="Count_ID") %>%
  inner_join(ML2_GO2gene_matrix) %>%
  column_to_rownames(var="Count_ID") -> matmatnorder
# View(matmatnorder)

# Changing names to a character
matmatnorder$Gene.Annotation <- as.character(matmatnorder$Gene.Annotation)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% 
                                         as.data.frame() %>% 
                                         dplyr::select(Gene.Annotation)))
```

```{r PC1 Variance Heatmaps, fig.height = 18, fig.width = 10}
# View(matmatnorder)
# ncol(matmatnorder)

# NB manual specify of the GO columns, here it is 86:93. First heatmap we remove, 2nd heatmap we select. 
HM1 <-
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        "Gene.Annotation", 284:289
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 4,
    row_km = 4,
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    column_gap = unit(0.2, "cm"),
    row_gap = unit(0.2, "cm"),
    column_dend_height = unit(3, "cm"),
    row_dend_width = unit(8, "cm"),
    show_row_names = F,
    row_title_gp = gpar(fontsize = 7),
    column_title_gp = gpar(fontsize = 7),
    show_column_names = F
  )

#GO yes no heatmap
HM2 <-
  Heatmap(
    t(t(
      matmatnorder %>% as.data.frame() %>% dplyr::select(c(284:289))
    )),
    show_row_dend = F,
    show_column_dend = F,
    show_column_names = T,
    show_row_names = F,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$Gene.Annotation,  gp = gpar(fontsize = 6, font = 10)
    )),
    column_names_gp = gpar(fontsize = 7, font = 1),
    col = c("white", "black"),
    column_names_rot = 30, 
    heatmap_width = unit(11, "cm"), 
    heatmap_height = unit(60, "cm")
  )

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(0.5, "cm"))
```

### DeSeq2 Visual Health Status

```{r Differential Exprssion analysis Disease Response}
dds_disease <- DESeqDataSetFromMatrix(countData = cccall, 
                                      colData = seq_tfall_final, 
                                      design = ~ Genotype + Health)

dds_disease <- DESeq(dds_disease, 
                     test="Wald")

resultsNames(dds_disease)

c_vs_vu <- results(dds_disease, 
                   contrast = c("Health", "visually_healthy", "pre_exposure"), 
                   alpha = 0.01, 
                   test = "Wald")

c_vs_d <- results(dds_disease, 
                   contrast = c("Health", "diseased", "pre_exposure"), 
                   alpha = 0.01, 
                   test = "Wald")

vu_vs_d <- results(dds_disease, 
                   contrast = c("Health", "diseased", "visually_healthy"), 
                   alpha = 0.01, 
                   test = "Wald")

summary(c_vs_vu)
summary(c_vs_d)
summary(vu_vs_d)
```

```{r}
as.data.frame(c_vs_vu) %>% 
  dplyr::filter(padj < 0.01) %>% 
  rownames_to_column(var = "Count_ID") %>%
  inner_join(annot_4_analysis %>%
               dplyr::select(Count_ID, Gene.Annotation)) %>% 
  mutate(DeSeq_contrast = "c_vs_vu") %>%
  column_to_rownames(var = "Count_ID") -> c_vs_vu_nolfc

as.data.frame(c_vs_d) %>% 
  dplyr::filter(padj < 0.01) %>% 
  rownames_to_column(var = "Count_ID") %>%
  inner_join(annot_4_analysis %>%
               dplyr::select(Count_ID, Gene.Annotation)) %>% 
  mutate(DeSeq_contrast = "c_vs_d") %>%
  column_to_rownames(var = "Count_ID") -> c_vs_d_nolfc

as.data.frame(vu_vs_d) %>% 
  dplyr::filter(padj < 0.01) %>% 
  rownames_to_column(var = "Count_ID") %>%
  inner_join(annot_4_analysis %>%
               dplyr::select(Count_ID, Gene.Annotation)) %>% 
  mutate(DeSeq_contrast = "vu_vs_d") %>% 
  column_to_rownames(var = "Count_ID") -> vu_vs_d_nolfc

# write.csv(c_vs_vu_nolfc, file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/c_vs_vh_nolfc.csv")
# write.csv(c_vs_d_nolfc, file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/c_vs_d_nolfc.csv")
# write.csv(vu_vs_d_nolfc, file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/vu_vs_d_nolfc.csv")
```

```{r Rersults Data Frames LCC1 and 0.01, include = F}
as.data.frame(c_vs_vu) %>% 
  dplyr::filter(padj < 0.01) %>% 
  dplyr::filter(log2FoldChange >= 1 | log2FoldChange <= -1) %>% 
  rownames_to_column(var = "Count_ID") %>%
  inner_join(annot_4_analysis %>%
               dplyr::select(Count_ID, Gene.Annotation)) %>% 
  mutate(DeSeq_contrast = "c_vs_vu") %>%
  column_to_rownames(var = "Count_ID") -> c_vs_vu_lfc1

as.data.frame(c_vs_d) %>% 
  dplyr::filter(padj < 0.01) %>% 
  dplyr::filter(log2FoldChange >= 1 | log2FoldChange <= -1)%>% 
  rownames_to_column(var = "Count_ID") %>%
  inner_join(annot_4_analysis %>%
               dplyr::select(Count_ID, Gene.Annotation)) %>% 
  mutate(DeSeq_contrast = "c_vs_d") %>%
  column_to_rownames(var = "Count_ID") -> c_vs_d_lfc1

as.data.frame(vu_vs_d) %>% 
  dplyr::filter(padj < 0.01) %>% 
  dplyr::filter(log2FoldChange >= 1 | log2FoldChange <= -1)%>% 
  rownames_to_column(var = "Count_ID") %>%
  inner_join(annot_4_analysis %>%
               dplyr::select(Count_ID, Gene.Annotation)) %>% 
  mutate(DeSeq_contrast = "vu_vs_d") %>% 
  column_to_rownames(var = "Count_ID") -> vu_vs_d_lfc1

# write.csv(c_vs_vu_lfc1, file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/c_vs_vh_lfc1.csv")
# write.csv(c_vs_d_lfc1, file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/c_vs_d_lfc1.csv")
# write.csv(vu_vs_d_lfc1, file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/vu_vs_d_lfc1.csv")
```

```{r}
c_vs_vu_lfc1 %>% nrow()
c_vs_vu_lfc1 %>% 
  dplyr::filter(log2FoldChange >= 1) %>% nrow()
c_vs_vu_lfc1 %>% 
  dplyr::filter(log2FoldChange <= -1) %>% nrow()

c_vs_d_lfc1 %>% nrow()
c_vs_d_lfc1 %>% 
  dplyr::filter(log2FoldChange >= 1) %>% nrow()
c_vs_d_lfc1 %>% 
  dplyr::filter(log2FoldChange <= -1) %>% nrow()

vu_vs_d_lfc1 %>% nrow()
vu_vs_d_lfc1 %>% 
  dplyr::filter(log2FoldChange >= 1) %>% nrow()
vu_vs_d_lfc1 %>% 
  dplyr::filter(log2FoldChange <= -1) %>% nrow()
```


```{r intersect lfc1 xvsvh and cvsd}
cvsvu_rn <- rownames(c_vs_vu_lfc1)
cvsd_rn <- rownames(c_vs_d_lfc1)
vuvsd_rn <- rownames(vu_vs_d_lfc1)

venn(list(cvsvu_rn, cvsd_rn))
```

```{r Log fold change numbers}
c_vs_vu_lfc1 %>% nrow()
c_vs_vu_lfc1 %>%
  dplyr::filter(log2FoldChange >= 1) %>% nrow()
c_vs_vu_lfc1 %>%
  dplyr::filter(log2FoldChange <= -1) %>% nrow()

c_vs_d_lfc1 %>% nrow()
c_vs_d_lfc1 %>%
  dplyr::filter(log2FoldChange >= 1) %>% nrow()
c_vs_d_lfc1 %>%
  dplyr::filter(log2FoldChange <= -1) %>% nrow()

vu_vs_d_lfc1 %>% nrow()
vu_vs_d_lfc1 %>%
  dplyr::filter(log2FoldChange >= 1) %>% nrow()
vu_vs_d_lfc1 %>%
  dplyr::filter(log2FoldChange <= -1) %>% nrow()
```

```{r Gene list prep for heatmap}
c_vs_d_lfc1 %>% 
  rownames_to_column(var = "Count_ID") %>% 
  rbind(c_vs_vu_lfc1 %>% 
          rownames_to_column(var = "Count_ID")) %>% 
  group_by(Count_ID, .drop = F) %>% 
  dplyr::summarise(sum = n()) %>% 
  dplyr::filter(sum %in% c("2")) %>%
  column_to_rownames(var = "Count_ID") %>% 
  rownames() -> disease_common

c_vs_d_lfc1 %>% 
  rownames_to_column(var = "Count_ID") %>% 
  rbind(c_vs_vu_lfc1 %>% 
          rownames_to_column(var = "Count_ID")) -> disease_all_results

disease_all_results %>% 
  dplyr::filter(Count_ID %in% disease_common) -> disease_common_res

# write.csv(disease_common_res, file = "~/Dropbox/PhD/Projects/DHE/NGS/tagseq/dissertation_files/raw_lists/disease_common_res.csv")
```

```{r All common 68 genes prep, include = F}
matmatnorder <- assay(vsdgeno)[disease_common_res$Count_ID,]

# Colours for the tank treatments
ccann <- data.frame(seq_tfall_final$Health)
colnames(ccann) <- c("Health")
colcol <- list("Health" = c("pre_exposure"="dodgerblue3", "visually_healthy"="darkgoldenrod2", "diseased" = "orangered3"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(1.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 40),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
annot_filtered_2_15 %>% 
  dplyr::select(Count_ID, Gene.Annotation) %>%
  right_join(matmatnorder %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Count_ID")) %>%
  column_to_rownames(var="Count_ID") %>%
  tidyr::drop_na() -> matmatnorder
#View(matmatnorder)

# Changing names to a character
matmatnorder$Gene.Annotation <- as.character(matmatnorder$Gene.Annotation)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(Gene.Annotation)))
```

```{r PC1 Variance Heatmaps, fig.height=50, fig.width=40}
# NB manual specify of the GO columns, here it is 86:93. First heatmap we remove, 2nd heatmap we select. 
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        "Gene.Annotation"
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 3,
    column_gap = unit(1, "cm"),
    row_km = 4,
    row_gap = unit(1, "cm"),
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$Gene.Annotation,  gp = gpar(fontsize = 40, font = 10))), 
    show_row_names = F, 
    show_column_names = F,
    column_dend_height = unit(10, "cm"))
```


##### KEGG Analysis For Disease Response

```{r KEGG IDs from Annot File, include = F}
filter(annot_filtered_2_15, Count_ID %in% rownames(c_vs_vu_lfc1)) %>%
  dplyr::select(KEGG.ID..threshold.30, Count_ID) -> cvu_kegg

filter(annot_filtered_2_15, Count_ID %in% rownames(c_vs_d_lfc1)) %>%
  dplyr::select(KEGG.ID..threshold.30, Count_ID) -> cd_kegg

filter(annot_filtered_2_15, Count_ID %in% rownames(vu_vs_d_lfc1)) %>%
  dplyr::select(KEGG.ID..threshold.30, Count_ID) -> vud_kegg
```

```{r KEGG Enrichment Analysis, include = F}
cvu_kegg$KEGG.ID..threshold.30 <- as.character(cvu_kegg$KEGG.ID..threshold.30)
cd_kegg$KEGG.ID..threshold.30 <- as.character(cd_kegg$KEGG.ID..threshold.30)
vud_kegg$KEGG.ID..threshold.30 <- as.character(vud_kegg$KEGG.ID..threshold.30)

cvu_enrichkegg <- enrichKEGG(gene = cvu_kegg$KEGG.ID..threshold.30, organism = 'ko', pvalueCutoff = 0.01)
cd_enrichkegg <- enrichKEGG(gene = cd_kegg$KEGG.ID..threshold.30, organism = 'ko', pvalueCutoff = 0.01)
vud_enrichkegg <- enrichKEGG(gene = vud_kegg$KEGG.ID..threshold.30, organism = 'ko', pvalueCutoff = 0.01)
```

```{r KEGG Enrichment barplots, echo = F, fig.height = 12}
barplot(cvu_enrichkegg, showCategory = 20)
barplot(cd_enrichkegg, showCategory = 45)
barplot(vud_enrichkegg, showCategory = 65)

# View(as.data.frame(cvu_enrichkegg))
# View(as.data.frame(cd_enrichkegg))
# View(as.data.frame(vud_enrichkegg))
```

```{r, fig.width=20, fig.height=14}
library(ggnewscale)

emapplot(pairwise_termsim(cd_enrichkegg), 
         showCategory = 100, 
         cex_label_category = 1.2, 
         cex_category = 1.2, 
         min_edge = 0.2, 
         layout = "nicely", 
         cex_line = 1)

emapplot(pairwise_termsim(vud_enrichkegg), 
         showCategory = 100, 
         cex_label_category = 1.2, 
         cex_category = 1.2, 
         min_edge = 0.2, 
         layout = "nicely", 
         cex_line = 1)
```

```{r writing KEGG results, include = F}
# cvu_enrichkegg %>%
#   as.data.frame() %>%
#   write.csv(., file = "~/Dropbox/PhD/Projects/DHE/NGS/tagseq/dissertation_files/raw_lists/c_vs_vu_kegg_res.csv")
# 
# cd_enrichkegg %>%
#   as.data.frame() %>%
#   write.csv(., file = "~/Dropbox/PhD/Projects/DHE/NGS/tagseq/dissertation_files/raw_lists/c_vs_d_kegg_res.csv")
# 
# vud_enrichkegg %>%
#   as.data.frame() %>%
#   write.csv(., file = "~/Dropbox/PhD/Projects/DHE/NGS/tagseq/dissertation_files/raw_lists/vu_vs_d_kegg_res.csv")
```

```{r KEGG results with genes, include = F}
as.data.frame(cvu_enrichkegg) %>%
  separate_rows(geneID, sep = '\\/', convert = T) %>%
  dplyr::rename(., pvalue_KEGG = pvalue) %>%
  inner_join(cvu_kegg,
             by = c("geneID" = "KEGG.ID..threshold.30")) %>%
  inner_join(
    c_vs_vu_lfc1 %>%
      rownames_to_column(var = "Count_ID")
  ) %>%
  dplyr::select(Count_ID, Gene.Annotation, 1:7, 11:16) -> c_vs_vu_kegg_genes

as.data.frame(cd_enrichkegg) %>%
  separate_rows(geneID, sep = '\\/', convert = T) %>%
  dplyr::rename(., pvalue_KEGG = pvalue) %>%
  inner_join(cd_kegg,
             by = c("geneID" = "KEGG.ID..threshold.30")) %>%
  inner_join(
    c_vs_d_lfc1 %>%
      rownames_to_column(var = "Count_ID")
  ) %>%
  dplyr::select(Count_ID, Gene.Annotation, 1:7, 11:16) -> c_vs_d_kegg_genes

as.data.frame(vud_enrichkegg) %>%
  separate_rows(geneID, sep = '\\/', convert = T) %>%
  dplyr::rename(., pvalue_KEGG = pvalue) %>%
  inner_join(vud_kegg,
             by = c("geneID" = "KEGG.ID..threshold.30")) %>%
  inner_join(
    vu_vs_d_lfc1 %>%
      rownames_to_column(var = "Count_ID")
  ) %>%
  dplyr::select(Count_ID, Gene.Annotation, 1:7, 11:16) -> vu_vs_d_kegg_genes

# write.csv(c_vs_vu_kegg_genes, 
#           file = "~/Dropbox/PhD/Projects/DHE/NGS/tagseq/dissertation_files/raw_lists/c_vs_vu_kegg_genes.csv")
# write.csv(c_vs_d_kegg_genes, 
#           file = "~/Dropbox/PhD/Projects/DHE/NGS/tagseq/dissertation_files/raw_lists/c_vs_d_kegg_genes.csv")
# write.csv(vu_vs_d_kegg_genes, 
#           file = "~/Dropbox/PhD/Projects/DHE/NGS/tagseq/dissertation_files/raw_lists/vu_vs_d_kegg_genes.csv")
```


## WGCNA Analysis with Genet Removed

So LRT was used for genet but for looking at the disease response and the difference between the different inoculations to reduce number of different tests WGCNA is a betetr option. Additionally to this WGCNA has more biological "meaning" as it is looking at gene expresion profiles and clustering them accordingly. 

So lets do that and if a reviewer doesn't like it I will fight them. 

### WGCNA Pipeline

```{r SUPER DUPER IMPORTANT FOR WGCNA TO DO THIS, include = F}
options(stringsAsFactors = FALSE)
```

```{r adding in z score for health identified from global 16s ancombc for health, include = F}
seq_tfall_final %>%
  mutate(
    cluster_1_16s = case_when(
      Health == "pre_exposure" ~ 0.40670558,
      Health == "visually_healthy" ~ -0.30834068,
      Health == "diseased" ~ 0.10020529
    ),
    cluster_2_16s = case_when(
      Health == "pre_exposure" ~ 0.14000085,
      Health == "visually_healthy" ~ 1.52813914,
      Health == "diseased" ~ 0.02263729
    ),
    cluster_3_16s = case_when(
      Health == "pre_exposure" ~ 0.87825261,
      Health == "visually_healthy" ~ -0.05047150,
      Health == "diseased" ~ -0.20151005
    ),
    cluster_4_16s = case_when(
      Health == "pre_exposure" ~ -0.23691806,
      Health == "visually_healthy" ~ 1.34422021,
      Health == "diseased" ~ 1.14771038
    )
  ) -> seq_tfall_final
```

```{r Making the all trait file for correlatvie analysis, include = F}
seq_tfall_final %>% 
  rownames_to_column(var = "samps") %>%
  dplyr::select(samps, Overall_treat, Health_disease, Health, cluster_1_16s, 
                cluster_2_16s, cluster_3_16s, cluster_4_16s) %>%
  column_to_rownames(var = "samps") -> WGCNA_treat_cato

model.matrix(~Overall_treat-1, WGCNA_treat_cato) %>% 
  as.data.frame() %>%
  rownames_to_column(var = "Sample_numbers") %>% 
  full_join(model.matrix(~Health_disease-1, WGCNA_treat_cato) %>% 
              as.data.frame() %>% 
              rownames_to_column(var = "Sample_numbers")) %>% 
  full_join(model.matrix(~Health-1, WGCNA_treat_cato) %>% 
              as.data.frame() %>% 
              rownames_to_column(var = "Sample_numbers")) %>% 
    inner_join(WGCNA_treat_cato %>% 
                 dplyr::select(4, 5, 6, 7) %>%
                 rownames_to_column(var = "Sample_numbers")) %>%
  dplyr::select(1, 2, 3, 12, 13, 11, 4, 7, 6, 5, 10, 9, 8, 14, 15, 16, 17) -> allTraits
```

```{r Removin g genet variance for the input VST for WGCNA, include = F}
ddsall <- DESeqDataSetFromMatrix(countData = cccall, 
                                 colData = seq_tfall_final, 
                                 design = ~ Genotype + Health)
vsdgeno <- vst(ddsall)
assay(vsdgeno) <- limma::removeBatchEffect(assay(vsdgeno), vsdgeno$Genotype)
```

```{r Making WGCNA gene objects and checking everything matches up, echo = F}
genes4wgcna <- assay(vsdgeno)
genes4wgcna <- t(genes4wgcna)
genes4wgcna <- as.data.frame(genes4wgcna)
nrow(genes4wgcna)
dim(genes4wgcna)

gsg = goodSamplesGenes(genes4wgcna, verbose = 3)
gsg$allOK
```

```{r Sample clustering to identify outliers, fig.height=20, fig.width=20, echo = F}
sampleTree = hclust(dist(genes4wgcna), method = "ward.D2");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
sizeGrWindow(12,9)
#pdf(file = "Plots/sampleClustering.pdf", width = 12, height = 9);
par(cex = 1);
par(mar = c(0,4,2,0))
plot(sampleTree, 
     main = "Sample clustering to detect outliers", 
     sub="", 
     xlab="", 
     cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
# Plot a line to show the cut
abline(h = 1200, col = "red");
# Determine cluster under the line
clust = cutreeStatic(sampleTree, cutHeight = 1800, minSize = 10)
table(clust)
# clust 1 contains the samples we want to keep.
keepSamples = (clust==1)
datExpr = genes4wgcna[keepSamples, ]
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
```

There are no crazy outliers so going to include everything for the downstream WGCNA analysis. 

```{r Correlative data matching with WGCNA gene objects, echo = F}
#4. Create an object with the treatment file
#Create an object called "datTraits" that contains your trait data
head(allTraits)
str(allTraits)
#form a data frame analogous to expression data that will hold the clinical traits.
rownames(allTraits) = allTraits$Sample_numbers
allTraits$Sample_numbers = NULL
table(rownames(allTraits)==rownames(genes4wgcna)) #should return TRUE if datasets align correctly, otherwise your names are out of order
```

Everything matches which is fantastic news between WGCNA gene object and the treatment file. 

```{r Identyfying Soft Power, echo = F}
# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5, dataIsExpr = T)
# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```
Soft power identified as 7, curves look good so no need for additional data filtering. 

```{r Signed adjacency matrix using soft power from last step, include = F}
softPower = 7;
adjacency = adjacency(datExpr, type="signed", power = softPower);
```

```{r Generation of topological overlap, include = F}
# Turn adjacency into topological overlap
TOM = TOMsimilarity(adjacency);
dissTOM = 1-TOM
```

```{r Gene clustering with TOM-based dissimilarity plot, fig.width=10, fig.height=4}
# Call the hierarchical clustering function
geneTree = hclust(as.dist(dissTOM), 
                  method = "average");
# Plot the resulting clustering tree (dendrogram)
sizeGrWindow(12,9)
plot(geneTree, 
     xlab="", 
     sub="", 
     main = "Gene clustering on TOM-based dissimilarity",
labels = FALSE, hang = 0.04);
```

```{r Module identification, echo = F}
# We like large modules, so we set the minimum module size relatively high:
minModuleSize = 30;
# Module identification using dynamic tree cut:
dynamicMods = cutreeDynamic(dendro = geneTree, 
                            distM = dissTOM,
                            deepSplit = 2,
                            pamRespectsDendro = FALSE, 
                            minClusterSize = minModuleSize);
table(dynamicMods)
```

```{r Gene dendogram and module colours plot, echo = F, fig.width = 7, fig.height= 2}
# Convert numeric lables into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
# Plot the dendrogram and colors underneath
sizeGrWindow(8,6)
plotDendroAndColors(geneTree, 
                    dynamicColors, 
                    "Dynamic Tree Cut", 
                    dendroLabels = FALSE, 
                    hang = 0.03,
                    addGuide = TRUE, 
                    guideHang = 0.05, 
                    main = "Gene dendrogram and module colors")
```

```{r Merging similar modules at 0.40, fig.height=6, fig.width=10, echo = F}
# Calculate eigengenes
MEList = moduleEigengenes(datExpr, 
                          colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), 
                method = "average");
# Plot the result
sizeGrWindow(7, 6)

plot(METree, 
     main = "Clustering of module eigengenes",
     xlab = "", 
     sub = "")
MEDissThres = 0.25
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")
```

```{r Merging similar modules, include = F}
# Call an automatic merging function
merge = mergeCloseModules(datExpr, 
                          dynamicColors, 
                          cutHeight = MEDissThres, 
                          verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs;
```

```{r Cluster dendogram with original and merged modules, fig.height=2.5, fig.width=13, echo = F}
sizeGrWindow(12, 5)
#pdf(file = "Plots/geneDendro-3.pdf", wi = 9, he = 6)
plotDendroAndColors(geneTree, 
                    cbind(dynamicColors, 
                          mergedColors), 
                    c("Dynamic Tree Cut", "Merged dynamic"), 
                    dendroLabels = FALSE, 
                    hang = 0.03, 
                    addGuide = TRUE, 
                    guideHang = 0.05)
#dev.off()
```

```{r Eigengenes for each sample to each module, echo = F}
# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", 
               standardColors(50));
moduleLabels = match(moduleColors, 
                     colorOrder)-1
MEs = mergedMEs
# View(MEs)
```

```{r Significance of eigengenes to correlative data, inclue = F}
# Define numbers of genes and samples
nGenes = ncol(datExpr);
nSamples = nrow(datExpr);
# Recalculate MEs with color labels
MEs0 = moduleEigengenes(datExpr, 
                        moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, 
                     allTraits, 
                     use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, 
                                     nSamples);
```

```{r writing transcritpomic eigengenes for cor analysis with 16s, include = F}
# write.csv(MEs0, 
#           file = "~/Documents/projects/DHE/manuscript/r_saved_files/correlative_analysis/dhe_tran_eigengenes.csv")
```

```{r Hub gene for each module identified form analysis, echo = F}
chooseTopHubInEachModule(genes4wgcna,
                                   moduleColors,
                                   omitColors = "grey",
                                   power = 10) %>% 
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>% 
  rownames_to_column(var = "module") %>% 
  inner_join(annot_filtered_2_15 %>% 
               dplyr::select(Count_ID, Gene.Annotation)) -> tophub_annotated
View(tophub_annotated)

# write.csv(tophub_annotated,
#          file = "~/Dropbox/NOAA_postdoc/projects/phd_chapters/DHE/manuscript/r_saved_files/tophub_annotated.csv",
#          row.names = F)
```

Pipeline all run, now for visualization and analysis of the modules. 


### WGCNA Visulaisation 
#### Initial Plots to Look at Correlations

```{r All modules module-trait relationship plot}
sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
labeledHeatmap(
  Matrix = moduleTraitCor,
  xLabels = names(allTraits),
  yLabels = names(MEs),
  ySymbols = names(MEs),
  colorLabels = FALSE,
  colors = blueWhiteRed(50),
  textMatrix = textMatrix,
  setStdMargins = FALSE,
  cex.text = 0.5,
  zlim = c(-1, 1),
  cex.lab = 0.5,
  main = paste("Module-trait relationships")
)
```


#### Sexy Plot All Modules

I want to use complex heatmap for this and flip the input data so that 
- module is columns
- Trait correlation is rows 

```{r complex heatmap dataframes with correct columns, include = F}
textMatrix %>%
  as.data.frame() %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) -> tmatrix_ch

# 3, 4, 7, 8, 10, 11, 5, 9, 12, 13, 14, 15, 16

moduleTraitCor %>%
  as.data.frame() %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) %>%
  t() %>%
  as.data.frame() %>%
  mutate(
    Row_annotations = c(
      "Pre Exposure",
      "Visually Healthy",
      "Diseased",
      "Health Slurry (VH)",
      "Disease Slurry (VH)",
      "Serratia Placebo (VH)",
      "Serratia (VH)",
      "Disease Slurry (D)",
      "Serratia (D)"
    )
  ) %>% 
  rownames_to_column(var = "remove") %>% 
  dplyr::select(-remove) %>%
  column_to_rownames(var = "Row_annotations") %>%
  t() %>%
  as.data.frame() -> modtraitcor_ch

allTraits %>% 
  as.data.frame() %>% 
dplyr::select(3:12) -> alltrait_ch
```

```{r Complex heatmap prep for WGCNA analysis, include =}
ccann <- data.frame(rownames(moduleTraitCor))
colnames(ccann) <- c("Modules")
colcol <- list("Modules" = c("MEdarkred" = "Dark Red", "MEmidnightblue" = "Midnight Blue", "MEblack" = "Black", 
                             "MEdarkturquoise" = "Dark Turquoise", "MEcyan" = "Cyan", "MEdarkgreen" = "Dark Green", 
                             "MEyellow" = "Yellow", "MEdarkgrey" = "Dark Grey", "MEpink" = "Pink", 
                             "MEpurple" = "Purple", "MEgreenyellow" = "Green Yellow", "MEgrey60" = "Grey 60", "MEblue" = "Blue", 
                             "MEbrown" = "Brown", "MElightcyan" = "Light Cyan", "MElightyellow" = "Light Yellow"))

samname <- HeatmapAnnotation(df = ccann, 
                             which = "row", 
                             col = colcol, 
                             simple_anno_size = unit(0.5, "cm"), 
                             annotation_name_gp = gpar(fontsize = 10), 
                             show_annotation_name = F)

col_fun = colorRamp2(c(-1, 0, 1), c("royalblue", "white", "red"))
```

```{r WGCNA complex heatmap, echo = F, fig.width = 11, fig.height=7}
Heatmap(
  modtraitcor_ch %>% as.matrix(),
  cluster_rows = F,
  cluster_columns = F,
  left_annotation = samname,
  show_row_names = T, 
  row_names_side = "left", 
  column_names_rot = 45,
  show_column_names = T,
  col = col_fun,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(sprintf(tmatrix_ch[i, j]), x, y, gp = gpar(fontsize = 9))
  }, 
  column_split = data.frame(rep(c("a", "b", "c"), c(3, 4, 2))), 
  column_gap = unit(0.3, "cm"))
```

#### Sexy Plot JUST PAPER MODULES

I want to use complex heatmap for this and flip the input data so that 
- module is columns
- Trait correlation is rows 

```{r complex heatmap dataframes with correct columns, include = F}
textMatrix %>%
  as.data.frame() %>% 
  dplyr::slice(13, 3, 6, 15, 4, 12, 1, 16) %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) -> tmatrix_ch

# 3, 4, 7, 8, 10, 11, 5, 9, 12, 13, 14, 15, 16

moduleTraitCor %>%
  as.data.frame() %>%
  dplyr::select(3, 4, 5, 7, 8, 10, 11, 9, 12) %>%
  t() %>%
  as.data.frame() %>%
  mutate(
    Row_annotations = c(
      "Pre Exposure",
      "Visually Healthy",
      "Diseased",
      "Health Slurry (VH)",
      "Disease Slurry (VH)",
      "Serratia Placebo (VH)",
      "Serratia (VH)",
      "Disease Slurry (D)",
      "Serratia (D)"
    )
  ) %>% 
  rownames_to_column(var = "remove") %>% 
  dplyr::select(-remove) %>%
  column_to_rownames(var = "Row_annotations") %>% 
  dplyr::select(MEblue, MEblack, MEdarkgreen, MElightcyan, MEdarkturquoise, MEgrey60, MEdarkred, MElightyellow) %>% 
  t() %>%
  as.data.frame() -> modtraitcor_ch

allTraits %>% 
  as.data.frame() %>% 
dplyr::select(3:12) -> alltrait_ch
```

```{r Complex heatmap prep for WGCNA analysis, include =}
ccann <- data.frame(rownames(modtraitcor_ch))
colnames(ccann) <- c("Modules")
colcol <- list(
  "Modules" = c(
    "MEblue" = "Blue",
    "MEblack" = "Black",
    "MEdarkgreen" = "Dark Green",
    "MElightcyan" = "Light Cyan",
    "MEdarkturquoise" = "Dark Turquoise",
    "MEgrey60" = "Grey 60",
    "MEdarkred" = "Dark Red",
    "MElightyellow" = "Light Yellow"
  )
)

samname <- HeatmapAnnotation(df = ccann, 
                             which = "row", 
                             col = colcol, 
                             simple_anno_size = unit(0.5, "cm"), 
                             annotation_name_gp = gpar(fontsize = 10), 
                             show_annotation_name = F)

col_fun = colorRamp2(c(-1, 0, 1), c("royalblue", "white", "red"))
```

```{r WGCNA complex heatmap, echo = F, fig.width = 10, fig.height=9}
Heatmap(
  modtraitcor_ch %>% as.matrix(),
  cluster_rows = F,
  cluster_columns = F,
  show_row_names = F,
  column_names_rot = 45,
  show_column_names = T,
  left_annotation = samname, 
  row_names_side = "left",
  col = col_fun,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(sprintf(tmatrix_ch[i, j]), x, y, gp = gpar(fontsize = 12))
  }, 
  column_split = data.frame(rep(c("a", "b", "c"), c(3, 4, 2))), 
  row_split = data.frame(rep(c("1", "2", "3", "4", "5"), c(1, 2, 2, 1, 2))),
  column_gap = unit(0.3, "cm"), 
  row_gap = unit(0.3, "cm"))
```


### Module Genes, Number of genes, and GO analysis
#### Annotated module genelists

```{r dataframes of genes within each module from WGCNA analysis, include = F}
 names(genes4wgcna)[moduleColors == "darkred"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> darkred_annotated

names(genes4wgcna)[moduleColors == "midnightblue"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> midnightblue_annotated

names(genes4wgcna)[moduleColors == "black"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> black_annotated

names(genes4wgcna)[moduleColors == "darkturquoise"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> darkturquoise_annotated

 names(genes4wgcna)[moduleColors == "cyan"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> cyan_annotated

names(genes4wgcna)[moduleColors == "darkgreen"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> darkgreen_annotated

names(genes4wgcna)[moduleColors == "yellow"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> yellow_annotated

names(genes4wgcna)[moduleColors == "darkgrey"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> darkgrey_annotated

names(genes4wgcna)[moduleColors == "pink"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> pink_annotated

names(genes4wgcna)[moduleColors == "purple"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> purple_annotated


names(genes4wgcna)[moduleColors == "greenyellow"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> greenyellow_annotated

names(genes4wgcna)[moduleColors == "grey60"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> grey60_annotated

names(genes4wgcna)[moduleColors == "blue"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> blue_annotated

names(genes4wgcna)[moduleColors == "brown"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> brown_annotated

names(genes4wgcna)[moduleColors == "lightcyan"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> lightcyan_annotated

names(genes4wgcna)[moduleColors == "lightyellow"] %>%
  as.data.frame() %>%
  dplyr::rename(., Count_ID = .) %>%
  inner_join(annot_filtered_2_15 %>%
               dplyr::select(Count_ID, Gene.Annotation, 6:21, 25)) -> lightyellow_annotated
```

```{r writing annotataed module gene lists, include = F}
# write.csv(
#   darkred_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/darkred_annotated.csv")
# 
# write.csv(
#   midnightblue_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/midnightblue_annotated.csv")
# 
# write.csv(
#   black_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/black_annotated.csv")
# 
# write.csv(
#   darkturquoise_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/darkturquoise_annotated.csv")
# 
# write.csv(
#   cyan_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/cyan_annotated.csv")
# 
# write.csv(
#   darkgreen_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/darkgreen_annotated.csv")
# 
# write.csv(
#   yellow_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/yellow_annotated.csv")
# 
# write.csv(
#   darkgrey_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/darkgrey_annotated.csv")
# 
# write.csv(
#   pink_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/pink_annotated.csv")
# 
# write.csv(
#   purple_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/purple_annotated.csv")
# 
# write.csv(
#   greenyellow_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/greenyellow_annotated.csv")
# 
# write.csv(
#   grey60_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/grey60_annotated.csv")
# 
# write.csv(
#   blue_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/blue_annotated.csv")
# 
# write.csv(
#   brown_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/brown_annotated.csv")
# 
# write.csv(
#   lightcyan_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/lightcyan_annotated.csv")
# 
# write.csv(
#   lightyellow_annotated,
#   file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/wgcna_genelists/lightyellow_annotated.csv")
```


#### Barplot for Number of Genes in ALL Modules

```{r Number of genes in each module from WGCNA analysis, echo = F}
nrow(darkred_annotated)
nrow(midnightblue_annotated)
nrow(black_annotated)
nrow(darkturquoise_annotated)
nrow(cyan_annotated)
nrow(darkgreen_annotated)
nrow(yellow_annotated)
nrow(darkgrey_annotated)
nrow(pink_annotated)
nrow(purple_annotated)
nrow(greenyellow_annotated)
nrow(grey60_annotated)
nrow(blue_annotated)
nrow(brown_annotated)
nrow(lightcyan_annotated)
nrow(lightyellow_annotated)
```

```{r gene count heatmap prep, include = F}
gene_bg <-
  cbind(c(
    nrow(lightyellow_annotated), 
    nrow(darkred_annotated), 
    nrow(grey60_annotated), 
    nrow(darkturquoise_annotated), 
    nrow(lightcyan_annotated),
    nrow(darkgreen_annotated), 
    nrow(black_annotated), 
    nrow(blue_annotated)))

colnames(gene_bg) <- c("genes")
rownames(gene_bg) <-
  c(
    "Light Yellow",
    "Dark Red",
    "Grey 60",
    "Dark Turquoise",
    "Light Cyan",
    "Dark Green", 
    "Black", 
    "Blue" 
  )

gene_bg %>%
  as.data.frame() %>%
  rownames_to_column(var = "module") -> gene_bg
gene_bg$module <-
  factor(
    gene_bg$module,
    c(
    "Light Yellow",
    "Dark Red",
    "Grey 60",
    "Dark Turquoise",
    "Light Cyan",
    "Dark Green", 
    "Black", 
    "Blue"
    )
  )
str(gene_bg)
```

```{r fucntion for reverse log transformation, include = F}
reverselog_trans <- function(base = exp(1)) {
    trans <- function(x) -log(x, base)
    inv <- function(x) base^(-x)
    trans_new(paste0("reverselog-", format(base)), trans, inv, 
              log_breaks(base = base), 
              domain = c(1e-100, Inf))
}
```

```{r bargraph of genes for MM heatmap, fig.height=15, fig.width=5}
ggplot(data = gene_bg, (aes(module, genes, fill = module))) +
  geom_bar(
    stat = "identity",
    fill = c(
      "Light Yellow",
    "Dark Red",
    "Grey 60",
    "Dark Turquoise",
    "Light Cyan",
    "Dark Green", 
    "Black", 
    "Blue"
    ),
    position = "dodge"
  ) +
  theme_classic() +
  theme(
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    axis.line.x = element_blank(),
    axis.line = element_line(),
    axis.title.y = element_blank(),
    axis.line.y = element_blank(),
    legend.position = "none",
    text = element_text(size = 15)
  )  +
  scale_y_continuous(position = "left",
                     trans = "log10") +
  coord_flip() +
  geom_text(aes(label = genes),
            hjust = -1.5,
            label.size = 8)
```

#### Barplot for Number of Genes in Paper Modules

```{r Number of genes in each module from WGCNA analysis, echo = F}
nrow(black_annotated)
nrow(darkturquoise_annotated)
nrow(darkgreen_annotated)
nrow(grey60_annotated)
nrow(blue_annotated)
nrow(lightcyan_annotated)
nrow(brown_annotated)
nrow(darkred_annotated)
nrow(lightyellow_annotated)
```

```{r gene count heatmap prep, include = F}
gene_bg <-
  cbind(c(
nrow(black_annotated),
nrow(darkturquoise_annotated),
nrow(darkgreen_annotated),
nrow(grey60_annotated),
nrow(blue_annotated),
nrow(lightcyan_annotated),
nrow(brown_annotated),
nrow(darkred_annotated),
nrow(lightyellow_annotated)
  ))
colnames(gene_bg) <- c("genes")
rownames(gene_bg) <-
  c(
    "Black",
    "Dark Turquoise",
    "Dark Green",
    "Grey 60", 
    "Blue", 
    "Light Cyan", 
    "Brown", 
    "Dark Red",
    "Light Yellow"
  )
gene_bg %>%
  as.data.frame() %>%
  rownames_to_column(var = "module") -> gene_bg
gene_bg$module <-
  factor(
    gene_bg$module,
    c(
    "Black",
    "Dark Turquoise",
    "Dark Green",
    "Grey 60", 
    "Blue", 
    "Light Cyan", 
    "Brown", 
    "Dark Red",
    "Light Yellow"
    )
  )
str(gene_bg)
```

```{r fucntion for reverse log transformation, include = F}
reverselog_trans <- function(base = exp(1)) {
    trans <- function(x) -log(x, base)
    inv <- function(x) base^(-x)
    trans_new(paste0("reverselog-", format(base)), trans, inv, 
              log_breaks(base = base), 
              domain = c(1e-100, Inf))
}
```

```{r bargraph of genes for MM heatmap, fig.height=10, fig.width=5}
ggplot(data = gene_bg, (aes(module, genes, fill = module))) +
  geom_bar(
    stat = "identity",
    fill = c(
    "Black",
    "Dark Turquoise",
    "Dark Green",
    "Grey 60", 
    "Blue", 
    "Light Cyan", 
    "Brown", 
    "Dark Red",
    "Light Yellow"
    ),
    position = "dodge"
  ) +
  theme_classic() +
  theme(
    strip.text.x = element_blank(),
    panel.border = element_blank(),
    axis.line.x = element_blank(),
    axis.line = element_line(),
    axis.title.y = element_blank(),
    axis.line.y = element_blank(),
    legend.position = "none",
    text = element_text(size = 15)
  )  +
  scale_y_continuous(position = "left",
                     trans = "log10")
  # coord_flip(90)
  # geom_text(aes(label = genes),
  #           hjust = -1.5,
  #           label.size = 8)
```


#### Overlap with Modules from 2020 Paper

Brown module was the immune gene module form the 2020 paper, thus is there any overlap between this study and that. 

```{r reading in brown module from 2020 paper, include = F}
read.csv(file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/brown_mod_2020.csv") %>% 
  mutate(Gene.Name.v2 = Gene.ID, 
         Gene.Annotation.v2 = Gene.Annotation) %>% 
  mutate(Gene.Name.v2 = str_replace_all(Gene.Name.v2, "evm.model", "gene.model")) %>%
  dplyr::select(-1, -2) -> brown_module_2020
```

```{r intersect of 2020 brown module and current modules, echo = F}
venn(list(brown_module_2020$Gene.Name.v2, darkred_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, midnightblue_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, black_annotated$Gene.Name.v2))

venn(list(brown_module_2020$Gene.Name.v2, darkturquoise_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, cyan_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, darkgreen_annotated$Gene.Name.v2))

venn(list(brown_module_2020$Gene.Name.v2, yellow_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, darkgrey_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, pink_annotated$Gene.Name.v2))

venn(list(brown_module_2020$Gene.Name.v2, purple_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, greenyellow_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, grey60_annotated$Gene.Name.v2))

venn(list(brown_module_2020$Gene.Name.v2, blue_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, brown_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, lightcyan_annotated$Gene.Name.v2))
venn(list(brown_module_2020$Gene.Name.v2, lightyellow_annotated$Gene.Name.v2))
```

```{r}
blue_annotated %>%
  dplyr::filter(Gene.Name.v2 %in% intersect(brown_module_2020$Gene.Name.v2, blue_annotated$Gene.Name.v2)) -> blue_brown_2020_intersect
View(blue_brown_2020_intersect)

write.csv(blue_brown_2020_intersect, 
          file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/r_saved_files/blue_brown_intersect.csv")

brown_annotated %>%
  dplyr::filter(Gene.Name.v2 %in% intersect(brown_module_2020$Gene.Name.v2, brown_annotated$Gene.Name.v2)) %>% View()
```


### KEGG Analysis WGCNA Modules

```{r}
library(clusterProfiler)
```

```{r running enrich kegg for modules, include = F}
kegg_darkred <- enrichKEGG(
  gene = darkred_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_midnightblue <- enrichKEGG(
  gene = midnightblue_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_black <- enrichKEGG(
  gene = black_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_darkturquoise <- enrichKEGG(
  gene = darkturquoise_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_cyan <- enrichKEGG(
  gene = cyan_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_darkgreen <- enrichKEGG(
  gene = darkgreen_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_yellow <- enrichKEGG(
  gene = yellow_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_darkgrey <- enrichKEGG(
  gene = darkgrey_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_pink <- enrichKEGG(
  gene = pink_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_purple <- enrichKEGG(
  gene = purple_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_greenyellow <- enrichKEGG(
  gene = greenyellow_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_grey60 <- enrichKEGG(
  gene = grey60_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_blue <- enrichKEGG(
  gene = blue_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_brown <- enrichKEGG(
  gene = brown_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_lightcyan <- enrichKEGG(
  gene = lightcyan_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)

kegg_lightyellow <- enrichKEGG(
  gene = lightyellow_annotated$KEGG.ID..threshold.30,
  organism = 'ko',
  pvalueCutoff = 0.05
)
```

```{r KEGG result dataframes from WGCNA and clusterprofiler, include = F}
kegg_darkred %>% 
  as.data.frame() -> kegg_darkred_res

kegg_midnightblue %>% 
  as.data.frame() -> kegg_midnightblue_res

kegg_black %>% 
  as.data.frame() -> kegg_black_res

kegg_darkturquoise %>% 
  as.data.frame() -> kegg_darkturquoise_res

kegg_cyan %>% 
  as.data.frame() -> kegg_cyan_res

kegg_darkgreen %>% 
  as.data.frame() -> kegg_darkgreen_res

kegg_yellow %>% 
  as.data.frame() -> kegg_yellow_res

kegg_darkgrey %>% 
  as.data.frame() -> kegg_darkgrey_res

kegg_pink %>% 
  as.data.frame() -> kegg_pink_res

kegg_purple %>% 
  as.data.frame() -> kegg_purple_res

kegg_greenyellow %>% 
  as.data.frame() -> kegg_greenyellow_res

kegg_grey60 %>% 
  as.data.frame() -> kegg_grey60_res

kegg_blue %>% 
  as.data.frame() -> kegg_blue_res

kegg_brown %>% 
  as.data.frame() -> kegg_brown_res

kegg_lightcyan %>% 
  as.data.frame() -> kegg_lightcyan_res

kegg_lightyellow %>% 
  as.data.frame() -> kegg_lightyellow_res
```

```{r}
# View(kegg_blue_res)
#View(kegg_lightyellow_res)
# View(kegg_pink_res)
# View(kegg_darkgrey_res)
# View(kegg_purple_res)
View(kegg_darkgreen_res)
# View(kegg_darkred_res)
View(kegg_lightcyan_res)
# View(kegg_cyan_res)
#View(kegg_darkturquoise_res)
# View(kegg_midnightblue_res)
View(kegg_brown_res)
View(kegg_blue_res)
#View(kegg_black_res)
#View(kegg_grey60_res)
```

```{r}
# barplot(kegg_darkred, showCategory = 100)
barplot(kegg_midnightblue, showCategory = 100)
barplot(kegg_black, showCategory = 100)
barplot(kegg_darkturquoise, showCategory = 100)
barplot(kegg_cyan, showCategory = 100)
# barplot(kegg_darkgreen, showCategory = 100)
barplot(kegg_yellow, showCategory = 100)
# barplot(kegg_darkgrey, showCategory = 100)
barplot(kegg_pink, showCategory = 100)
barplot(kegg_purple, showCategory = 100)
barplot(kegg_greenyellow, showCategory = 100)
barplot(kegg_grey60, showCategory = 100)
barplot(kegg_blue, showCategory = 100)
barplot(kegg_brown, showCategory = 100)
# barplot(kegg_lightcyan, showCategory = 100)
barplot(kegg_lightyellow, showCategory = 100)
```

```{r}
view(darkgreen_annotated)
```


### GO Gene Lists for INteresting Modules 

```{r}
pad27 <- function(x){
  str_x = as.character(x)
  digits = nchar(str_x)
  toAdd = 7-digits
  name = paste0('GO:',
                strrep(0,toAdd), 
                x)
  name
}
```

```{r reading in the brown, blue, dark green and light cyan GO analysis, include = F}
read.table(
    file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/bingo/wgcna_0.01_brown.bgo", 
  skip = 29,  
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
#  dplyr::select(GO.ID, Description, Genes.in.test.set) %>% View()
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>% 
  dplyr::inner_join(annot_filtered_2_15 %>% 
               dplyr::select(Count_ID, Gene.Annotation) %>% 
                 dplyr::mutate(Count_ID = toupper(Count_ID)), 
             by = c("Genes.in.test.set" = "Count_ID")) %>% 
  dplyr::rename(., pvalue_GO = p.value) -> brown_wgcna_GO_res

read.table(
    file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/bingo/wgcna_0.01_blue.bgo", 
  skip = 31,  
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
#  dplyr::select(GO.ID, Description, Genes.in.test.set) %>% View()
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::inner_join(annot_filtered_2_15 %>% 
               dplyr::select(Count_ID, Gene.Annotation) %>% 
                 dplyr::mutate(Count_ID = toupper(Count_ID)), 
             by = c("Genes.in.test.set" = "Count_ID")) %>% 
  dplyr::rename(., pvalue_GO = p.value) -> blue_wgcna_go_Res

read.table(
    file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/bingo/wgcna_0.01_darkgreen.bgo", 
  skip = 22,  
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
#  dplyr::select(GO.ID, Description, Genes.in.test.set) %>% View()
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::inner_join(annot_filtered_2_15 %>% 
               dplyr::select(Count_ID, Gene.Annotation) %>% 
                 dplyr::mutate(Count_ID = toupper(Count_ID)), 
             by = c("Genes.in.test.set" = "Count_ID")) %>% 
  dplyr::rename(., pvalue_GO = p.value) -> darkgreen_wgcna_go_res


read.table(
    file = "/Users/benjamin.d.young/Documents/projects/DHE/manuscript/bingo/wgcna_0.01_lightcyan.bgo", 
  skip = 19,  
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
#  dplyr::select(GO.ID, Description, Genes.in.test.set) %>% View()
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::inner_join(annot_filtered_2_15 %>% 
               dplyr::select(Count_ID, Gene.Annotation) %>% 
                 dplyr::mutate(Count_ID = toupper(Count_ID)), 
             by = c("Genes.in.test.set" = "Count_ID")) %>% 
  dplyr::rename(., pvalue_GO = p.value) -> lightcyan_wgcna_go_res

View(blue_wgcna_go_Res)
View(darkgreen_wgcna_go_res)
View(lightcyan_wgcna_go_res)
View(brown_wgcna_GO_res)
```


